--- oleo-1.99.16.orig/po/oleo.pot
+++ oleo-1.99.16/po/oleo.pot
@@ -6,27 +6,29 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
-"POT-Creation-Date: 2001-03-10 15:32+0100\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2005-09-04 15:52+0200\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=CHARSET\n"
-"Content-Transfer-Encoding: ENCODING\n"
+"Content-Transfer-Encoding: 8bit\n"
 
-#: src/io-term.c:815
+#: src/io-term.c:817
 #, c-format
 msgid ""
 "This is %s %s\n"
 "\n"
 msgstr ""
 
-#: src/io-term.c:817
+#: src/io-term.c:819
 #, c-format
 msgid "Usage: %s [OPTION]... [FILE]...\n"
 msgstr ""
 
-#: src/io-term.c:820
+#: src/io-term.c:822
+#, c-format
 msgid ""
 "\n"
 "  -h, --help               display this help and exit\n"
@@ -43,56 +46,74 @@
 "Report bugs to <bug-oleo@gnu.org>.\n"
 msgstr ""
 
-#: src/io-term.c:1025
+#: src/io-term.c:1027
 #, c-format
 msgid "%s %s\n"
 msgstr ""
 
-#: src/io-term.c:1026
-msgid "Copyright © 1992-2000 Free Software Foundation, Inc.\n"
+#: src/io-term.c:1028
+#, c-format
+msgid "Copyright (C) 1992-2000 Free Software Foundation, Inc.\n"
 msgstr ""
 
-#: src/io-term.c:1027
+#: src/io-term.c:1029
 #, c-format
 msgid "%s comes with ABSOLUTELY NO WARRANTY.\n"
 msgstr ""
 
-#: src/io-term.c:1028
+#: src/io-term.c:1030
 #, c-format
 msgid "You may redistribute copies of %s\n"
 msgstr ""
 
-#: src/io-term.c:1029
+#: src/io-term.c:1031
+#, c-format
 msgid "under the terms of the GNU General Public License.\n"
 msgstr ""
 
-#: src/io-term.c:1030
+#: src/io-term.c:1032
+#, c-format
 msgid "For more information about these matters, "
 msgstr ""
 
-#: src/io-term.c:1031
+#: src/io-term.c:1033
+#, c-format
 msgid "see the files named COPYING.\n"
 msgstr ""
 
-#: src/io-term.c:1174
+#: src/io-term.c:1176
+#, c-format
 msgid "Error in the builtin init scripts (a bug!).\n"
 msgstr ""
 
-#: src/io-term.c:1194 src/io-term.c:1196
+#: src/io-term.c:1196 src/io-term.c:1198
 #, c-format
 msgid "   error occured in init file %s near line %d."
 msgstr ""
 
-#: src/io-term.c:1215 src/io-term.c:1216
+#: src/io-term.c:1217 src/io-term.c:1218
 #, c-format
 msgid ", error occured reading '%s'\n"
 msgstr ""
 
-#: src/io-term.c:1223 src/io-term.c:1224
+#: src/io-term.c:1225 src/io-term.c:1226
 #, c-format
 msgid "Can't open %s: %s\n"
 msgstr ""
 
+#.
+#. * This is used in two places.
+#.
+#: src/io-motif.c:150
+msgid ""
+"\n"
+"GNU Oleo is free software, you are welcome to\n"
+"distribute copies of it. See the file COPYING for the\n"
+"conditions.\n"
+"\n"
+"GNU Oleo comes with ABSOLUTELY NO WARRANTY."
+msgstr ""
+
 #: src/io-motif.c:722
 msgid "Fit Print Area"
 msgstr ""
@@ -330,5 +351,6 @@
 msgstr ""
 
 #: src/io-motif.c:4791 src/io-motif.c:5906
+#, c-format
 msgid "XmProcessTraversal failed\n"
 msgstr ""
--- oleo-1.99.16.orig/doc/expressions.texi
+++ oleo-1.99.16/doc/expressions.texi
@@ -1,4 +1,4 @@
-@node   Expressions, , .oleorc file, Basics
+@node   Expressions, , ~/.oleorc file, Basics
 @section Expressions
 @cindex Expressions
 
--- oleo-1.99.16.orig/doc/basics.texi
+++ oleo-1.99.16/doc/basics.texi
@@ -46,7 +46,7 @@
 * Getting Help::                Getting help
 * Saving and reading::          Saving and reading Spreadsheets
 * Command Line Options::        Command Line Options
-* .oleorc file::                The initialization file
+* ~/.oleorc file::              The initialization file
 * Expressions::                 Types of expressions
 @end menu
 
@@ -109,7 +109,7 @@
 M-x set-option @key{RET} input -1
 @end example
 
-This can be made the default.  (@pxref{.oleorc file})
+This can be made the default.  (@pxref{~/.oleorc file})
 
 @node Status Line , Work Area, Minibuffer, The Screen
 @unnumberedsubsec The status line
@@ -638,7 +638,7 @@
 
 
 
-@node Command Line Options, .oleorc file, Saving and reading, Basics
+@node Command Line Options, ~/.oleorc file, Saving and reading, Basics
 @section Command Line Options
 @comment  node-name,  next,  previous,  up
 @cindex Command Line Options
@@ -683,8 +683,8 @@
 @end table
 
 
-@node .oleorc file, Expressions, Command Line Options, Basics
-@section The @file{.oleorc} file
+@node ~/.oleorc file, Expressions, Command Line Options, Basics
+@section The @file{~/.oleorc} file
 @cindex .oleorc
 @comment  node-name,  next,  previous,  up
 
--- oleo-1.99.16.orig/doc/oleo.info-1
+++ oleo-1.99.16/doc/oleo.info-1
@@ -1,4 +1,4 @@
-This is oleo.info, produced by makeinfo version 4.0 from oleo.texi.
+This is oleo.info, produced by makeinfo version 4.3 from oleo.texi.
 
    This file documents Oleo, the GNU spreadsheet.
 
@@ -98,7 +98,7 @@
 * Getting Help::                Getting help
 * Saving and reading::          Saving and reading Spreadsheets
 * Command Line Options::        Command Line Options
-* .oleorc file::                The initialization file
+* ~/.oleorc file::              The initialization file
 * Expressions::                 Types of expressions
 
    ---------- Footnotes ----------
@@ -275,7 +275,7 @@
 
      M-x set-option <RET> input -1
 
-   This can be made the default.  (*note .oleorc file::)
+   This can be made the default.  (*note ~/.oleorc file::)
 
 
 File: oleo.info,  Node: Status Line,  Next: Work Area,  Prev: Minibuffer,  Up: The Screen
@@ -714,7 +714,7 @@
    (2) what is the default
 
 
-File: oleo.info,  Node: Command Line Options,  Next: .oleorc file,  Prev: Saving and reading,  Up: Basics
+File: oleo.info,  Node: Command Line Options,  Next: ~/.oleorc file,  Prev: Saving and reading,  Up: Basics
 
 Command Line Options
 ====================
@@ -767,7 +767,7 @@
    (1) But what does this really do?
 
 
-File: oleo.info,  Node: .oleorc file,  Next: Expressions,  Prev: Command Line Options,  Up: Basics
+File: oleo.info,  Node: ~/.oleorc file,  Next: Expressions,  Prev: Command Line Options,  Up: Basics
 
 The `.oleorc' file
 ==================
@@ -786,7 +786,7 @@
    (1) what else can go here? can I set the default font here?
 
 
-File: oleo.info,  Node: Expressions,  Prev: .oleorc file,  Up: Basics
+File: oleo.info,  Node: Expressions,  Prev: ~/.oleorc file,  Up: Basics
 
 Expressions
 ===========
--- oleo-1.99.16.orig/doc/oleo.info-2
+++ oleo-1.99.16/doc/oleo.info-2
@@ -1,4 +1,4 @@
-This is oleo.info, produced by makeinfo version 4.0 from oleo.texi.
+This is oleo.info, produced by makeinfo version 4.3 from oleo.texi.
 
    This file documents Oleo, the GNU spreadsheet.
 
--- oleo-1.99.16.orig/doc/oleo.info-3
+++ oleo-1.99.16/doc/oleo.info-3
@@ -1,4 +1,4 @@
-This is oleo.info, produced by makeinfo version 4.0 from oleo.texi.
+This is oleo.info, produced by makeinfo version 4.3 from oleo.texi.
 
    This file documents Oleo, the GNU spreadsheet.
 
--- oleo-1.99.16.orig/doc/oleo.info-4
+++ oleo-1.99.16/doc/oleo.info-4
@@ -1,4 +1,4 @@
-This is oleo.info, produced by makeinfo version 4.0 from oleo.texi.
+This is oleo.info, produced by makeinfo version 4.3 from oleo.texi.
 
    This file documents Oleo, the GNU spreadsheet.
 
@@ -28,7 +28,7 @@
 
 * Menu:
 
-* .oleorc:                               .oleorc file.
+* .oleorc:                               ~/.oleorc file.
 * a0 mode:                               a0 mode.
 * Absolute references:                   Cell referencing.
 * Addressing cells:                      Cell referencing.
--- oleo-1.99.16.orig/doc/oleo.info
+++ oleo-1.99.16/doc/oleo.info
@@ -1,4 +1,4 @@
-This is oleo.info, produced by makeinfo version 4.0 from oleo.texi.
+This is oleo.info, produced by makeinfo version 4.3 from oleo.texi.
 
    This file documents Oleo, the GNU spreadsheet.
 
@@ -23,9 +23,9 @@
 
 Indirect:
 oleo.info-1: 888
-oleo.info-2: 50331
-oleo.info-3: 99293
-oleo.info-4: 148190
+oleo.info-2: 50339
+oleo.info-3: 99301
+oleo.info-4: 148198
 
 Tag Table:
 (Indirect)
@@ -39,142 +39,142 @@
 Node: Comparison8414
 Node: The Screen8882
 Node: Minibuffer9268
-Node: Status Line9727
-Ref: Status Line-Footnote-110778
-Node: Work Area10958
-Node: Cursor and Mark11553
-Node: Recalculation12630
-Node: Movement13557
-Node: Single Cell Movement14000
-Node: Scanning14680
-Ref: Scanning-Footnote-115646
-Node: Large Scale Movement15773
-Ref: Large Scale Movement-Footnote-116907
-Ref: Large Scale Movement-Footnote-217007
-Node: Movement examples17023
-Node: Screen Moving18543
-Node: Mouse19287
-Ref: Mouse-Footnote-119901
-Node: Regions and Variables19976
-Node: Getting Help20973
-Node: Saving and reading22872
-Ref: Saving and reading-Footnote-124051
-Ref: Saving and reading-Footnote-224090
-Node: Command Line Options24118
-Ref: Command Line Options-Footnote-125114
-Node: .oleorc file25152
-Ref: .oleorc file-Footnote-125686
-Node: Expressions25750
-Node: Infix expressions26103
-Node: Examples27761
-Node: Error values28720
-Node: How to enter data30483
-Node: Oleo values30823
-Node: Entering31759
-Node: Input Editing32314
-Ref: Input Editing-Footnote-134612
-Node: Cell Motion34657
-Node: Making changes35834
-Node: Copying36171
-Node: Moving38610
-Ref: Moving-Footnote-139966
-Node: Deleting40025
-Node: Insertion40858
-Node: Style41439
-Node: Alignment42228
-Node: Formats for numbers42923
-Ref: Formats for numbers-Footnote-147773
-Ref: Formats for numbers-Footnote-247796
-Ref: Formats for numbers-Footnote-347843
-Node: Cell Height47916
-Ref: Cell Height-Footnote-148657
-Node: Width of Cells48692
-Ref: Width of Cells-Footnote-149391
-Node: Hidden Rows and Columns49406
-Node: Protection50331
-Node: Fonts (in X11)51310
-Node: Multiple Windows52912
-Node: Windows53519
-Node: Split Window54233
-Node: Other Window54973
-Node: Change Window56041
-Node: Options56765
-Node: Appearance57228
-Ref: Appearance-Footnote-158600
-Node: Window Options58650
-Node: Other Options60324
-Ref: Other Options-Footnote-162446
-Ref: Other Options-Footnote-262461
-Ref: Other Options-Footnote-362493
-Node: Printing and Plotting62530
-Node: Writing Files62804
-Ref: Writing Files-Footnote-167475
-Node: Graphing67552
-Node: Clearing Plots69144
-Node: Data sets69574
-Ref: Data sets-Footnote-171558
-Node: The axes71590
-Ref: The axes-Footnote-172365
-Ref: The axes-Footnote-272379
-Node: Viewing Choices72396
-Node: Output Type73225
-Node: Style Options74270
-Ref: Style Options-Footnote-174991
-Node: Name Options75044
-Ref: Name Options-Footnote-175640
-Node: Macros75674
-Node: Keyboard Macros75849
-Ref: Keyboard Macros-Footnote-177584
-Node: Basic Kbd Macro77616
-Ref: Basic Kbd Macro-Footnote-179083
-Ref: Basic Kbd Macro-Footnote-279111
-Node: Save Kbd Macro79136
-Ref: Save Kbd Macro-Footnote-179632
-Node: Programs79658
-Node: One Line Programs79996
-Node: Long One Liners80940
-Node: Writing programs81299
-Ref: Writing programs-Footnote-183026
-Node: Commands in Programs83082
-Node: Keymaps83404
-Node: Functions87048
-Node: Boolean functions87869
-Node: Math functions88900
-Node: General math functions89270
-Ref: General math functions-Footnote-191230
-Node: Trigonometric functions91275
-Node: Statistical functions92329
-Node: String functions95420
-Ref: String functions-Footnote-197475
-Ref: String functions-Footnote-297551
-Node: Structural functions97679
-Node: Search functions99293
-Ref: Search functions-Footnote-1101667
-Node: Business functions101834
-Node: Date functions106123
-Node: Gsl functions112039
-Node: Extending Oleo113571
-Node: New at functions113783
-Node: Reporting Bugs118229
-Node: Motif119592
-Node: Motif Introduction119983
-Node: LessTif120361
-Node: Advantages120995
-Node: What you need121358
-Node: Alternatives122116
-Node: Using the mouse122422
-Node: Using the keyboard shortcuts123448
-Node: Using Help124264
-Node: Fallback124391
-Node: Motif Buttons in a Spreadsheet Cell124900
-Node: Database Access125498
-Node: Database Access Introduction125669
-Node: MySQL126294
-Node: Xbase126565
-Node: Key Index126647
-Node: Command Index131982
-Node: Function Index138818
-Node: Options Index145616
-Node: Concept Index148190
+Node: Status Line9729
+Ref: Status Line-Footnote-110780
+Node: Work Area10960
+Node: Cursor and Mark11555
+Node: Recalculation12632
+Node: Movement13559
+Node: Single Cell Movement14002
+Node: Scanning14682
+Ref: Scanning-Footnote-115648
+Node: Large Scale Movement15775
+Ref: Large Scale Movement-Footnote-116909
+Ref: Large Scale Movement-Footnote-217009
+Node: Movement examples17025
+Node: Screen Moving18545
+Node: Mouse19289
+Ref: Mouse-Footnote-119903
+Node: Regions and Variables19978
+Node: Getting Help20975
+Node: Saving and reading22874
+Ref: Saving and reading-Footnote-124053
+Ref: Saving and reading-Footnote-224092
+Node: Command Line Options24120
+Ref: Command Line Options-Footnote-125118
+Node: ~/.oleorc file25156
+Ref: ~/.oleorc file-Footnote-125692
+Node: Expressions25756
+Node: Infix expressions26111
+Node: Examples27769
+Node: Error values28728
+Node: How to enter data30491
+Node: Oleo values30831
+Node: Entering31767
+Node: Input Editing32322
+Ref: Input Editing-Footnote-134620
+Node: Cell Motion34665
+Node: Making changes35842
+Node: Copying36179
+Node: Moving38618
+Ref: Moving-Footnote-139974
+Node: Deleting40033
+Node: Insertion40866
+Node: Style41447
+Node: Alignment42236
+Node: Formats for numbers42931
+Ref: Formats for numbers-Footnote-147781
+Ref: Formats for numbers-Footnote-247804
+Ref: Formats for numbers-Footnote-347851
+Node: Cell Height47924
+Ref: Cell Height-Footnote-148665
+Node: Width of Cells48700
+Ref: Width of Cells-Footnote-149399
+Node: Hidden Rows and Columns49414
+Node: Protection50339
+Node: Fonts (in X11)51318
+Node: Multiple Windows52920
+Node: Windows53527
+Node: Split Window54241
+Node: Other Window54981
+Node: Change Window56049
+Node: Options56773
+Node: Appearance57236
+Ref: Appearance-Footnote-158608
+Node: Window Options58658
+Node: Other Options60332
+Ref: Other Options-Footnote-162454
+Ref: Other Options-Footnote-262469
+Ref: Other Options-Footnote-362501
+Node: Printing and Plotting62538
+Node: Writing Files62812
+Ref: Writing Files-Footnote-167483
+Node: Graphing67560
+Node: Clearing Plots69152
+Node: Data sets69582
+Ref: Data sets-Footnote-171566
+Node: The axes71598
+Ref: The axes-Footnote-172373
+Ref: The axes-Footnote-272387
+Node: Viewing Choices72404
+Node: Output Type73233
+Node: Style Options74278
+Ref: Style Options-Footnote-174999
+Node: Name Options75052
+Ref: Name Options-Footnote-175648
+Node: Macros75682
+Node: Keyboard Macros75857
+Ref: Keyboard Macros-Footnote-177592
+Node: Basic Kbd Macro77624
+Ref: Basic Kbd Macro-Footnote-179091
+Ref: Basic Kbd Macro-Footnote-279119
+Node: Save Kbd Macro79144
+Ref: Save Kbd Macro-Footnote-179640
+Node: Programs79666
+Node: One Line Programs80004
+Node: Long One Liners80948
+Node: Writing programs81307
+Ref: Writing programs-Footnote-183034
+Node: Commands in Programs83090
+Node: Keymaps83412
+Node: Functions87056
+Node: Boolean functions87877
+Node: Math functions88908
+Node: General math functions89278
+Ref: General math functions-Footnote-191238
+Node: Trigonometric functions91283
+Node: Statistical functions92337
+Node: String functions95428
+Ref: String functions-Footnote-197483
+Ref: String functions-Footnote-297559
+Node: Structural functions97687
+Node: Search functions99301
+Ref: Search functions-Footnote-1101675
+Node: Business functions101842
+Node: Date functions106131
+Node: Gsl functions112047
+Node: Extending Oleo113579
+Node: New at functions113791
+Node: Reporting Bugs118237
+Node: Motif119600
+Node: Motif Introduction119991
+Node: LessTif120369
+Node: Advantages121003
+Node: What you need121366
+Node: Alternatives122124
+Node: Using the mouse122430
+Node: Using the keyboard shortcuts123456
+Node: Using Help124272
+Node: Fallback124399
+Node: Motif Buttons in a Spreadsheet Cell124908
+Node: Database Access125506
+Node: Database Access Introduction125677
+Node: MySQL126302
+Node: Xbase126573
+Node: Key Index126655
+Node: Command Index131990
+Node: Function Index138826
+Node: Options Index145624
+Node: Concept Index148198
 
 End Tag Table
--- oleo-1.99.16.orig/src/io-utils.c
+++ oleo-1.99.16/src/io-utils.c
@@ -71,7 +71,7 @@
 
 double __plinf;
 double __neinf;
-double __nan;
+double oleo__nan;
 
 char nname[] = "#NOT_A_NUMBER";
 char iname[] = "#INFINITY";
@@ -125,7 +125,7 @@
   __neinf = divide (-1., 0.);
   (void) signal (SIGFPE, ignore_sig);
 #endif
-  __nan = __plinf + __neinf;
+  oleo__nan = __plinf + __neinf;
 }
 
 
--- oleo-1.99.16.orig/src/ref.c
+++ oleo-1.99.16/src/ref.c
@@ -263,7 +263,7 @@
 read_new_value (CELLREF row, CELLREF col, char *form, char *val)
 {
   unsigned char *new_bytes;
-  extern double __plinf, __neinf, __nan;
+  extern double __plinf, __neinf, oleo__nan;
 
   cur_row = row;
   cur_col = col;
@@ -348,7 +348,7 @@
 	  else if (!stricmp (nname, val))
 	    {
 	      SET_TYP (my_cell, TYP_FLT);
-	      my_cell->cell_flt = __nan;
+	      my_cell->cell_flt = oleo__nan;
 	    }
 	  else
 	    {
--- oleo-1.99.16.orig/src/io-x11.c
+++ oleo-1.99.16/src/io-x11.c
@@ -34,7 +34,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <sys/param.h>
-#define NeedFunctionPrototypes 0
+/* #define NeedFunctionPrototypes 0 */
 #include <X11/X.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
--- oleo-1.99.16.orig/src/plotter.c
+++ oleo-1.99.16/src/plotter.c
@@ -2639,8 +2639,6 @@
 	case SP_PLOT_PIE:
 		sp_pie_plot_point(mg, point);
 		break;
-	default:
-		/* ?? */
 	}
 }
 
@@ -2770,8 +2768,6 @@
 	case SP_PLOT_PIE:
 		sp_pie_begin_graph(mg, scale, trans_x, trans_y);
 		break;
-	default:
-		/* ??? */
 	}
 
 	sp_first_dataset(mg);
@@ -2790,8 +2786,6 @@
 	case SP_PLOT_PIE:
 		sp_pie_end_graph(mg);
 		break;
-	default:
-		/* ??? */
 	}
 }
 
--- oleo-1.99.16.orig/src/io-term.c
+++ oleo-1.99.16/src/io-term.c
@@ -234,7 +234,7 @@
 
 		if (Preferences[i].copynext) {
 			ptr += strlen(Preferences[i].name) + 1;
-			((char *)Preferences[i].var) = strdup(ptr);
+			Preferences[i].var = strdup(ptr);
 		} else if (Preferences[i].var)
 			*((int *)Preferences[i].var) = Preferences[i].value;
 
@@ -770,6 +770,8 @@
     }
 }
 
+extern struct cmd_func cmd_funcs[];
+
 static void
 init_maps (void)
 {
@@ -1023,7 +1025,7 @@
 	  case 'v':
 	  case 'V':
 	    printf(_("%s %s\n"), GNU_PACKAGE, VERSION);
-            printf(_("Copyright © 1992-2000 Free Software Foundation, Inc.\n"));
+            printf(_("Copyright (C) 1992-2000 Free Software Foundation, Inc.\n"));
             printf(_("%s comes with ABSOLUTELY NO WARRANTY.\n"), GNU_PACKAGE);
             printf(_("You may redistribute copies of %s\n"), PACKAGE);
             printf(_("under the terms of the GNU General Public License.\n"));
--- oleo-1.99.16.orig/src/io-term.h
+++ oleo-1.99.16/src/io-term.h
@@ -44,8 +44,6 @@
 extern int default_lock;
 
 
-extern struct cmd_func cmd_funcs[];
-
 /* When printing ascii output, this controls page width. */
 extern int modified;
 
--- oleo-1.99.16.orig/src/global.h
+++ oleo-1.99.16/src/global.h
@@ -208,7 +208,7 @@
 extern char nname[];
 
 extern VOIDSTAR parse_hash;
-extern double __plinf, __neinf, __nan;
+extern double __plinf, __neinf, oleo__nan;
 
 /* These have two uses.  During parsing, these contain the 
  * base address of all relative references.  During evaluation,
@@ -231,7 +231,7 @@
 
 extern unsigned short current_cycle;
 extern int ioerror;
-extern int errno;
+#include <errno.h>
 extern const char oleo_version_string[];
 
 extern double astof (char **);
--- oleo-1.99.16.orig/src/eval.c
+++ oleo-1.99.16/src/eval.c
@@ -565,7 +565,7 @@
 	case CONST_NINF:
 	case CONST_NAN:
 	  p->type = TYP_FLT;
-	  p->Float = (byte == CONST_INF) ? __plinf : ((byte == CONST_NINF) ? __neinf : __nan);
+	  p->Float = (byte == CONST_INF) ? __plinf : ((byte == CONST_NINF) ? __neinf : oleo__nan);
 	  break;
 
 	case VAR:
--- oleo-1.99.16.orig/src/posixtm.y
+++ oleo-1.99.16/src/posixtm.y
@@ -85,6 +85,7 @@
 		   YYABORT;
 		 }
 	       }
+		;
 
 year : digitpair {
                    t.tm_year = $1;
--- oleo-1.99.16.orig/src/sysdef.h
+++ oleo-1.99.16/src/sysdef.h
@@ -119,8 +119,6 @@
 
 #endif /* !defined(__STDC__) */
 
-extern char *getenv ();
-
 #ifndef RETSIGTYPE
 #define RETSIGTYPE void
 #endif /* RETSIGTYPE */
--- oleo-1.99.16.orig/src/basic.h
+++ oleo-1.99.16/src/basic.h
@@ -36,9 +36,6 @@
 
 extern const int colmagic[9];
 extern  const int rowmagic[9];
-extern char * motion_name[9];
-extern enum motion_magic complementary_motion[9];
-extern enum motion_magic opposite_motion[9];
 
 extern int run_load_hooks;
 
--- oleo-1.99.16.orig/src/xrdb.c
+++ oleo-1.99.16/src/xrdb.c
@@ -57,7 +57,7 @@
 /* This should be included before the X include files; otherwise, we get
    warnings about redefining NULL under BSD 4.3.  */
 #include <sys/param.h>
-#define NeedFunctionPrototypes 0
+/* #define NeedFunctionPrototypes 0 */
 #include <X11/X.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
