--- collatinus-9.3.orig/collatinus.pro
+++ collatinus-9.3/collatinus.pro
@@ -3,7 +3,7 @@
 ######################################################################
 
 TEMPLATE = app
-TARGET = 
+TARGET =
 DEPENDPATH += .
 INCLUDEPATH += .
 
@@ -15,8 +15,10 @@
 
 CONFIG += release_binary
 
-install.files = collatinus lemmata.* expressions.fr config lucretia.txt INSTALL COPYING 
-install.path = /usr/share/collatinus 
-install.commands = cp -a doc /usr/share/collatinus/; $(SYMLINK) /usr/share/collatinus/collatinus /usr/bin/collatinus 
+install.files = collatinus lemmata.* expressions.fr config lucretia.txt INSTALL COPYING
+install.path = /usr/share/collatinus
+documentation.path = /usr/share/collatinus/doc
+documentation.files = doc/*
+install.commands = $(SYMLINK) ../share/collatinus/collatinus $(INSTALL_ROOT)/usr/bin/collatinus
 INSTALLS += install
-
+INSTALLS += documentation
