--- downloader.py.old	2015-01-02 19:25:00.061764104 -0500
+++ downloader.py	2015-01-02 19:25:47.551170405 -0500
@@ -157,6 +157,8 @@
     if passed_defaultsini:
         configuration.readfp(passed_defaultsini)
 
+    if isfile('/etc/fanficdownloader/defaults.ini'):
+  	conflist.append('/etc/fanficdownloader/defaults.ini')
     if isfile(join(homepath, 'defaults.ini')):
         conflist.append(join(homepath, 'defaults.ini'))
     if isfile('defaults.ini'):
