diff -u fastdep-0.16.old/FileCache.cc fastdep-0.16/FileCache.cc
--- fastdep-0.16.old/FileCache.cc	2010-11-17 21:17:10.000000000 -0500
+++ fastdep-0.16/FileCache.cc	2010-11-17 21:18:46.000000000 -0500
@@ -17,6 +17,7 @@
 #include <fstream>
 
 #include <algorithm>
+#include <memory>
 
 #ifdef WIN32
 extern char *realpath(const char *path, char resolved_path[]);
diff -u fastdep-0.16.old/MappedFile.cc fastdep-0.16/MappedFile.cc
--- fastdep-0.16.old/MappedFile.cc	2010-11-17 21:25:20.000000000 -0500
+++ fastdep-0.16/MappedFile.cc	2010-11-17 21:25:50.000000000 -0500
@@ -7,6 +7,10 @@
 #include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <cstdio>
+#include <cerrno>
 
 #endif
 
