From a4af30a55acc7f5e39ed8f22308187f0cff05772 Mon Sep 17 00:00:00 2001
From: Aptrug <60358026+Aptrug@users.noreply.github.com>
Date: Sat, 10 Oct 2020 16:02:51 +0100
Subject: [PATCH 18/40] fix grammatical error in parseobject.cpp

---
 src/parseobject.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/parseobject.cpp b/src/parseobject.cpp
index 3b8832da..6f85daea 100755
--- a/src/parseobject.cpp
+++ b/src/parseobject.cpp
@@ -1098,7 +1098,7 @@ void ParseObject::runUserFilter(int feedId, int filterId)
           case 0: // condition -> contains
             qStr1.append(QString("UPPER(title) LIKE '%%1%' ").arg(content.toUpper()));
             break;
-          case 1: // condition -> doesn't contains
+          case 1: // condition -> doesn't contain
             qStr1.append(QString("UPPER(title) NOT LIKE '%%1%' ").arg(content.toUpper()));
             break;
           case 2: // condition -> is
@@ -1123,7 +1123,7 @@ void ParseObject::runUserFilter(int feedId, int filterId)
           case 0: // condition -> contains
             qStr1.append(QString("UPPER(description) LIKE '%%1%' ").arg(content.toUpper()));
             break;
-          case 1: // condition -> doesn't contains
+          case 1: // condition -> doesn't contain
             qStr1.append(QString("UPPER(description) NOT LIKE '%%1%' ").arg(content.toUpper()));
             break;
           case 2: // condition -> regExp
@@ -1136,7 +1136,7 @@ void ParseObject::runUserFilter(int feedId, int filterId)
           case 0: // condition -> contains
             qStr1.append(QString("UPPER(author_name) LIKE '%%1%' ").arg(content.toUpper()));
             break;
-          case 1: // condition -> doesn't contains
+          case 1: // condition -> doesn't contain
             qStr1.append(QString("UPPER(author_name) NOT LIKE '%%1%' ").arg(content.toUpper()));
             break;
           case 2: // condition -> is
@@ -1155,7 +1155,7 @@ void ParseObject::runUserFilter(int feedId, int filterId)
           case 0: // condition -> contains
             qStr1.append(QString("UPPER(category) LIKE '%%1%' ").arg(content.toUpper()));
             break;
-          case 1: // condition -> doesn't contains
+          case 1: // condition -> doesn't contain
             qStr1.append(QString("UPPER(category) NOT LIKE '%%1%' ").arg(content.toUpper()));
             break;
           case 2: // condition -> is
@@ -1207,7 +1207,7 @@ void ParseObject::runUserFilter(int feedId, int filterId)
           case 0: // condition -> contains
             qStr1.append(QString("link_href LIKE '%%1%' ").arg(content));
             break;
-          case 1: // condition -> doesn't contains
+          case 1: // condition -> doesn't contain
             qStr1.append(QString("link_href NOT LIKE '%%1%' ").arg(content));
             break;
           case 2: // condition -> is
@@ -1232,7 +1232,7 @@ void ParseObject::runUserFilter(int feedId, int filterId)
           case 0: // condition -> contains
             qStr1.append(QString("(UPPER(title) LIKE '%%1%' OR UPPER(description) LIKE '%%1%') ").arg(content.toUpper()));
             break;
-          case 1: // condition -> doesn't contains
+          case 1: // condition -> doesn't contain
             qStr1.append(QString("(UPPER(title) NOT LIKE '%%1%' OR UPPER(description) NOT LIKE '%%1%') ").arg(content.toUpper()));
             break;
           case 2: // condition -> regExp
-- 
2.43.0

