From a930c75b6d2ab3c31a280da032b6b1d0fdaeaa46 Mon Sep 17 00:00:00 2001
From: Zlatan <16328515+darcosion@users.noreply.github.com>
Date: Tue, 9 Feb 2021 03:26:17 +0100
Subject: [PATCH 25/40] Update INSTALL, add webkitwidgets seem necessary

---
 INSTALL | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/INSTALL b/INSTALL
index 52879eac..f3211363 100644
--- a/INSTALL
+++ b/INSTALL
@@ -15,7 +15,7 @@ Build depends:
     
  Linux Mint 20 (others?):
  sudo apt install qt5-default qttools5-dev-tools qttools5-dev libsqlite3-dev
- sudo apt-get install qtmultimedia5-dev libqt5multimediawidgets5 libqt5multimedia5-plugins libqt5multimedia5
+ sudo apt-get install qtmultimedia5-dev libqt5multimediawidgets5 libqt5multimedia5-plugins libqt5multimedia5 libqt5webkit5-dev
 
 
 Instructions for Linux and other UNIX-like systems:
-- 
2.43.0

