--- a/usr/share/ecub/ecub.desktop	2013-01-09 13:08:59.000000000 -0500
+++ b/usr/share/ecub/ecub.desktop	2013-01-09 13:11:48.000000000 -0500
@@ -1,16 +1,14 @@
 [Desktop Entry]
-Encoding=UTF-8
 Version=1.0
 Type=Application
 Terminal=false
 Exec=/usr/share/ecub/ecub
 TryExec=/usr/share/ecub/ecub
-Icon[en_GB]=ecub.png
-Icon=ecub.png
+Icon[en_GB]=ecub
+Icon=ecub
 Name[en_GB]=eCub
 Name=eCub
 Comment=ePub creation tool
-MimeType=application/x-ecubebkproj
-Patterns=*.ebkproj;
+MimeType=application/x-ecubebkproj;
 X-MultipleArgs=false
-Categories=Office;WordProcessor;Literature;GTK;
+Categories=Office;WordProcessor;GTK;
