--- blockattack-1.4.1.orig/debian/copyright
+++ blockattack-1.4.1/debian/copyright
@@ -0,0 +1,61 @@
+This package was debianized by Gonéri Le Bouder <goneri@rulezlan.org> on
+Fri, 19 May 2006 17:38:23 +0200.
+
+It was downloaded from http://blockattack.sf.net
+
+The actual maintainer is the Debian Games Team:
+http://wiki.debian.org/Games/Development
+
+
+
+Copyright 2006 Poul Sander <poul@poulsander.com>
+
+SCons scripts and some minor changes around SHAREDIR were done by
+Gonéri Le Bouder
+Copyright 2006 Gonéri Le Bouder <goneri@rulezlan.org>
+
+License:
+
+   This package is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This package is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this package; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+
+On Debian systems, the complete text of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL'.
+
+SFont files are Copyright (C) 2003 Karl Bartel
+License: GPL or LGPL (at your choice)
+
+On Debian systems, the complete text of the GNU Lesser General
+Public License can be found in `/usr/share/common-licenses/LGPL'.
+
+The enet directorys:
+These files were downloaded from http://enet.cubik.org/
+Copyright 2002-2006 Lee Salzman
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+sell copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
--- blockattack-1.4.1.orig/debian/rules
+++ blockattack-1.4.1/debian/rules
@@ -0,0 +1,56 @@
+#!/usr/bin/make -f
+# -*- makefile -*-
+#export DH_VERBOSE=1
+
+
+build: build-stamp
+build-stamp:
+	dh_testdir
+	QUILT_PATCHES=debian/patches quilt push -a || test $$? = 2
+	scons prefix=/usr sharedir=/usr/share/games/blockattack
+	touch build-stamp
+
+clean:
+	dh_testdir
+	scons -c
+	rm -f .sconsign.dblite build-stamp
+	- QUILT_PATCHES=debian/patches quilt pop -a -R || test $$? = 2
+
+	rm -rf .pc
+	rm -f enet-1.0/config.status
+	rm -f enet-1.0/config.log
+
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k 
+	dh_installdirs
+	find Game -type f -exec chmod 644 {} \;
+	scons prefix=/usr bindir=/usr/games mandir=/usr/share/man sharedir=/usr/share/games/blockattack destdir=$(CURDIR)/debian/blockattack install
+
+	#Remove Windows Thumbnail database.
+	rm -f $(CURDIR)/debian/blockattack/usr/share/games/blockattack/gfx/Thumbs.db
+
+binary-indep: build install
+
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_installchangelogs changelog*
+	dh_installdocs README AUTHORS block_attack_manual.pdf
+	dh_installman
+	dh_installmenu
+	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install
--- blockattack-1.4.1.orig/debian/compat
+++ blockattack-1.4.1/debian/compat
@@ -0,0 +1 @@
+5
--- blockattack-1.4.1.orig/debian/control
+++ blockattack-1.4.1/debian/control
@@ -0,0 +1,41 @@
+Source: blockattack
+Section: games
+Priority: optional
+Maintainer: Debian Games Team <pkg-games-devel@lists.alioth.debian.org>
+Uploaders: Gonéri Le Bouder <goneri@rulezlan.org>, Cyril Brulebois <kibi@debian.org>, Barry deFreese <bdefreese@debian.org>
+Build-Depends: debhelper (>= 5.0.0), docbook-to-man, libsdl-mixer1.2-dev,
+ libsdl-image1.2-dev, libsdl1.2-dev, libenet-dev (>= 1.2-3), quilt, scons
+Standards-Version: 3.8.2
+Homepage: http://blockattack.sf.net
+Vcs-Svn: svn://svn.debian.org/svn/pkg-games/packages/trunk/blockattack/
+Vcs-Browser: http://svn.debian.org/wsvn/pkg-games/packages/trunk/blockattack/?op=log
+
+Package: blockattack
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: a puzzle game inspired by Tetris
+ Block Attack is a puzzle/blockfall game inspired by Nintendo's Tetris Attack
+ for the Super Nintendo. The game is pretty action packed for a puzzle game :-)
+ .
+ Features:
+  * Based on the classic "Tetris Attack" for the SNES 
+  * 5 single player modes: Endless, Time Trial, Puzzle Mode, Stage Clear and
+  Vs. Mode 
+  * 2 two player options: Time Trial and Vs. Mode
+  * Puzzle mode has 20+ puzzles
+  * Players can use a custom key setup
+  * Handicap in multiplayer
+  * Highscores are saved 
+  * Joypad support
+  * Music and sound effects 
+  * Animations
+  * Puzzle level editor (writen in Java)
+  * Select puzzle file from a list in the game 
+  * Runs under Linux and Windows + more
+ .
+ System requirements:
+  * Screen resolution: 1024x768 (cannot be changed, but it can run in a window)
+  * Keyboard
+  * Mouse (optional)
+  * Processor: Decent (tested on 733 MHz)
+  * Memory: at least 64 MB
--- blockattack-1.4.1.orig/debian/changelog
+++ blockattack-1.4.1/debian/changelog
@@ -0,0 +1,58 @@
+blockattack (1.3.1-4) unstable; urgency=low
+
+  * Build depend on libenet-dev instead of libenet1-dev, thank you Andreas
+    Moog (Closes: #541067)
+  * Standard version 3.8.2, no changes
+
+ -- Gonéri Le Bouder <goneri@rulezlan.org>  Tue, 11 Aug 2009 16:16:29 +0200
+
+blockattack (1.3.1-3) unstable; urgency=low
+
+  [ Gonéri Le Bouder ]
+  * remove the Xs- prefix from the Vcs entrys
+  * remove the homepage from the package description since the Homepage: field
+    do the job now
+
+  [ Eddy Petrișor ]
+  * fix Vcs-Svn URL to point to svn, not the incomplete ssh location
+
+  [ Cyril Brulebois ]
+  * update my mail address
+
+  [ Bary deFreese ]
+  * Add myself to uploaders.
+  * gcc44.diff - Build with gcc-4.4. (Closes: #505026).
+    + Thanks to Martin Michlmayr for the hint and patch.
+  * Remove quilt .pc dir on clean.
+  * Remove autogenerated enet-1.0/config.{status, log} on clean.
+  * Remove Windows Thumbs.db on install so it's not in the .deb.
+  * Bump Standards Version to 3.8.1. (No changes needed).
+
+ -- Barry deFreese <bdefreese@debian.org>  Mon, 30 Mar 2009 16:50:08 -0400
+
+blockattack (1.3.1-2) unstable; urgency=low
+
+  [ Gonéri Le Bouder ]
+  * fix spelling error in blockattack's long description (Closes: #432653)
+  * add desktop_file.diff to remove the Application category from the
+    desktop file
+
+  [ Jon Dowland ]
+  * update menu section to "Games/Action" for menu transition
+    (thanks Linas Žvirblis)
+
+  [ Jordà Polo ]
+  * debian/patches:
+    + scons_args.diff: New patch to fix scons command-line argument handling.
+      (Closes: #444536)
+    + replay_init.diff: New patch to initialize the Replay class correctly.
+  * debian/control: Added Homepage and XS-Vcs-Browser fields.
+
+ -- Gonéri Le Bouder <goneri@rulezlan.org>  Mon, 01 Oct 2007 22:16:15 +0000
+
+blockattack (1.3.1-1) unstable; urgency=low
+
+  * Initial release Closes: #368395
+
+ -- Gonéri Le Bouder <goneri@rulezlan.org>  Mon, 14 May 2007 00:12:30 +0200
+
--- blockattack-1.4.1.orig/debian/watch
+++ blockattack-1.4.1/debian/watch
@@ -0,0 +1,8 @@
+# Watch control file for uscan
+# See uscan(1) for format
+
+# Compulsory line, this is a version 3 file
+version=3
+
+# Find new files on sourceforge, for debscripts >= 2.9
+http://sf.net/blockattack/blockattack-(.*)\.tar\.gz
--- blockattack-1.4.1.orig/debian/menu
+++ blockattack-1.4.1/debian/menu
@@ -0,0 +1,4 @@
+?package(blockattack):needs="X11" section="Games/Action"\
+  title="Blockattack" command="/usr/games/blockattack" \
+  longtitle="BlockAttack - Rise of the Blocks" \
+  icon="blockattack32.xpm"
--- blockattack-1.4.1.orig/debian/patches/replay_init.diff
+++ blockattack-1.4.1/debian/patches/replay_init.diff
@@ -0,0 +1,22 @@
+--- blockattack-1.4.1.orig/source/code/replay.cpp
++++ blockattack-1.4.1/source/code/replay.cpp
+@@ -29,7 +29,7 @@
+ 
+ #include "replay.h"
+ 
+-void Replay::newReplay()
++Replay::Replay()
+ {
+ 	nrOfFrames=0;
+ }
+--- blockattack-1.4.1.orig/source/code/replay.h
++++ blockattack-1.4.1/source/code/replay.h
+@@ -66,7 +66,7 @@
+       enum { gameOver=0, winner, looser, draw } theResult;
+       
+       public:     
+-      void newReplay(); //Constructor
++      Replay(); //Constructor
+       Uint32 getNumberOfFrames(); //Returns number of frames
+       void setFrameSecTo(Uint32,boardPackage); //Sets frame at a given time to the package
+       void setFinalFrame(boardPackage,int); //Sets the final package
--- blockattack-1.4.1.orig/debian/patches/scons_args.diff
+++ blockattack-1.4.1/debian/patches/scons_args.diff
@@ -0,0 +1,9 @@
+--- blockattack-1.4.1.orig/SConstruct
++++ blockattack-1.4.1/SConstruct
+@@ -1,5 +1,5 @@
+ # Option      #
+-opts = Options(ARGUMENTS)
++opts = Options(args = ARGUMENTS)
+ opts.Add('destdir', 'Staging area to install BlockAttack to.  Useful for packagers.', '')
+ opts.Add('prefix', 'Destination directory', '/usr/local')
+ opts.Add('sharedir', 'Directory to use to store data file', '$prefix/share/blockattack')
--- blockattack-1.4.1.orig/debian/patches/series
+++ blockattack-1.4.1/debian/patches/series
@@ -0,0 +1,6 @@
+scons_args.diff
+fix_scons.diff
+fix_spelling.diff
+replay_init.diff
+desktop_file.diff
+gcc44.diff
--- blockattack-1.4.1.orig/debian/patches/desktop_file.diff
+++ blockattack-1.4.1/debian/patches/desktop_file.diff
@@ -0,0 +1,10 @@
+Index: blockattack-1.4.1/blockattack.desktop
+===================================================================
+--- blockattack-1.4.1.orig/blockattack.desktop	2007-10-01 22:15:33.000000000 +0000
++++ blockattack-1.4.1/blockattack.desktop	2007-10-01 22:16:00.000000000 +0000
+@@ -11,4 +11,4 @@
+ Icon=blockattack32
+ Terminal=false
+ Type=Application
+-Categories=Application;Game;ArcadeGame
++Categories=Game;ArcadeGame
--- blockattack-1.4.1.orig/debian/patches/fix_spelling.diff
+++ blockattack-1.4.1/debian/patches/fix_spelling.diff
@@ -0,0 +1,13 @@
+Index: blockattack-1.4.1/main.cpp
+===================================================================
+--- blockattack-1.4.1.orig/main.cpp	2007-03-28 00:34:07.000000000 +0200
++++ blockattack-1.4.1/main.cpp	2007-03-28 00:34:14.000000000 +0200
+@@ -2927,7 +2927,7 @@
+           endl << "-priority  Starts game in high priority" << endl <<
+           "-forceredraw  Redraw the whole screen every frame, prevents garbage" << endl <<
+           "-forcepartdraw  Only draw what is changed, sometimes cause garbage" << endl <<
+-	  "-nosound  No sound will be played at all, and sound hardware want be loaded (use this if game crashes because of sound)" << endl;
++	  "-nosound  No sound will be played at all, and sound hardware won't be loaded (use this if game crashes because of sound)" << endl;
+ #ifdef WIN32
+ 		  system("Pause");
+ #endif
--- blockattack-1.4.1.orig/debian/patches/fix_scons.diff
+++ blockattack-1.4.1/debian/patches/fix_scons.diff
@@ -0,0 +1,17 @@
+Index: blockattack-1.4.1/Game/SConscript
+===================================================================
+--- blockattack-1.4.1.orig/Game/SConscript	2007-03-28 00:27:40.000000000 +0200
++++ blockattack-1.4.1/Game/SConscript	2007-03-28 00:29:31.000000000 +0200
+@@ -125,9 +125,9 @@
+ 'gfx/bChange.png',
+ 'gfx/Thumbs.db',
+ 'gfx/bHost.png',
+-'gfx/b1600.png',
+-'gfx/b1400.png',
+-'gfx/b1280.png',
++#'gfx/b1600.png',
++#'gfx/b1400.png',
++#'gfx/b1280.png',
+ 'gfx/iCheckBoxArea2.png',
+ 'gfx/blackLine.png',
+ 'gfx/b1024.png',
--- blockattack-1.4.1.orig/debian/patches/gcc44.diff
+++ blockattack-1.4.1/debian/patches/gcc44.diff
@@ -0,0 +1,12 @@
+Index: blockattack-1.4.1/source/code/listFiles.h
+===================================================================
+--- blockattack-1.4.1.orig/source/code/listFiles.h	2009-03-30 16:07:00.000000000 -0400
++++ blockattack-1.4.1/source/code/listFiles.h	2009-03-30 16:07:34.000000000 -0400
+@@ -25,6 +25,7 @@
+ 
+ //listFiles.h - List files in a given dictory, 10 files at a time, at most 250 files
+ #include <string.h>
++#include <cstdio>
+ #include <iostream>
+ #include <stdlib.h>
+ #if defined(_WIN32)
