diff -ru a/Makefile b/Makefile
--- a/Makefile	2023-09-15 21:26:06.475039969 -0400
+++ b/Makefile	2023-09-15 21:28:28.911038061 -0400
@@ -5,10 +5,9 @@
 ZOPFLI_OBJ      := lib/zopfli/hash.o lib/zopfli/squeeze.o lib/zopfli/gzip_container.o lib/zopfli/katajainen.o lib/zopfli/zopfli_lib.o lib/zopfli/cache.o lib/zopfli/zlib_container.o lib/zopfli/util.o lib/zopfli/tree.o lib/zopfli/deflate.o lib/zopfli/blocksplitter.o lib/zopfli/lz77.o
 ZOPFLIPNG_OBJ   := lib/zopflipng/lodepng/lodepng.o lib/zopflipng/lodepng/lodepng_util.o lib/zopflipng/zopflipng_lib.o
 
-CFLAGS      += -Wall -Werror -O3 -flto
 CPPFLAGS    += -I./lib
-CXXFLAGS    += $(CFLAGS) -std=c++17 -fno-rtti
-LDFLAGS     += -flto -lpthread
+CXXFLAGS    += -std=c++17 -fno-rtti
+LDFLAGS     += -lpthread
 
 ifeq ($(OS), Windows_NT)
     SYSTEM  := Windows
Only in a: .Makefile.swp
