Disable checks for libs if --disable-client is given

--- configure.ac.orig	2009-10-01 07:39:34.214302673 +0200
+++ configure.ac	2009-10-01 09:13:20.163741753 +0200
@@ -29,18 +29,24 @@
 AM_CONDITIONAL(ENABLE_SERVER, test x$enable_server = xyes)
 AM_CONDITIONAL(ENABLE_CLIENT, test x$enable_client = xyes)
 
-PKG_CHECK_MODULES(OPROFILEUI, [glib-2.0 libglade-2.0 gtk+-2.0 libxml-2.0 gnome-vfs-2.0 gconf-2.0])
-AC_SUBST(OPROFILEUI_CFLAGS)
-AC_SUBST(OPROFILEUI_LIBS)
-
-PKG_CHECK_MODULES(OPROFILE_SERVER, [glib-2.0])
-AC_SUBST(OPROFILE_SERVER_CFLAGS)
-AC_SUBST(OPROFILE_SERVER_LIBS)
+PKG_PROG_PKG_CONFIG()
+
+if test "x$enable_client" = "xyes"; then
+        PKG_CHECK_MODULES(OPROFILEUI, [glib-2.0 libglade-2.0 gtk+-2.0 libxml-2.0 gnome-vfs-2.0 gconf-2.0])
+        AC_SUBST(OPROFILEUI_CFLAGS)
+        AC_SUBST(OPROFILEUI_LIBS)
+fi
+
+if test "x$enable_server" = "xyes"; then
+        PKG_CHECK_MODULES(OPROFILE_SERVER, [glib-2.0])
+        AC_SUBST(OPROFILE_SERVER_CFLAGS)
+        AC_SUBST(OPROFILE_SERVER_LIBS)
+fi
 
 AC_ARG_WITH(avahi,
         [AC_HELP_STRING([--with-avahi], [Use Avahi to announce and search for OProfile servers])],
         with_avahi="$withval", with_avahi=no)
-if test $with_avahi = "yes"; then
+if test "x$with_avahi" = "xyes"; then
         PKG_CHECK_MODULES(AVAHI, [avahi-client avahi-glib])
         AC_DEFINE_UNQUOTED(WITH_AVAHI, [1], [Using Avahi])
 fi
