--- Makefile.old	2017-04-21 19:02:19.268404333 -0400
+++ Makefile	2017-04-21 19:02:43.241405100 -0400
@@ -27,6 +27,8 @@
 	install -d ${PREFIX}/share/icons/hicolor/32x32/apps
 	install -d ${PREFIX}/share/icons/hicolor/48x48/apps
 	install -d ${PREFIX}/share/icons/hicolor/64x64/apps
+	install -d ${PREFIX}/share/icons/hicolor/96x96/apps
+	install -d ${PREFIX}/share/icons/hicolor/128x128/apps
 	install -d ${PREFIX}/share/icons/hicolor/scalable/apps
 	install -d ${PREFIX}/share/applications
 	install -d ${PPUBDIR}/styles
