--- a/buttons.h	Wed Mar 19 08:40:07 1997
+++ b/buttons.h	Sun Jun 28 20:08:05 2009
@@ -58,7 +58,7 @@
      
 
      button (win  *b_w,
-             char b_label [],
+             const char b_label [],
              int  b_x,
              int  b_y,
              bool b_with_repeat = false,
@@ -76,8 +76,8 @@
      void press (bool mode);
      bool eval  (int  &button);
      bool eval  ();
-     void write (char label_string []);
-     void write (char label_string [], bool is_pressed);
+     void write (const char label_string []);
+     void write (const char label_string [], bool is_pressed);
 
   };
 
--- a/cmap.h	Fri Feb 21 06:48:24 1997
+++ b/cmap.h	Sun Jun 28 18:53:02 2009
@@ -23,7 +23,7 @@
    int         rot;
 
    cmap      (win  *w_i,
-              char name_i [],
+              const char name_i [],
               bool preload = true,
               bool smart_load = true);
    ~cmap     ();
--- a/cmap_selector.h	Fri Feb 21 06:48:24 1997
+++ b/cmap_selector.h	Sun Jun 28 21:24:38 2009
@@ -40,7 +40,7 @@
    int      case_dy;   
    int      case_dx;
    
-   cmap_selector     (char name [], 
+   cmap_selector     (const char name [], 
                       win  *w,
                       int  num_cases,
                       char case_string [max_cmap_selector_cases][128],
@@ -65,8 +65,8 @@
 
   }; 
 
-bool cmap_sel      (char name [], char pattern []);
-void get_file_list (char file_pattern [],
+bool cmap_sel      (const char name [], const char pattern []);
+void get_file_list (const char file_pattern [],
                     int  &num_files,
                     char file_list [max_cmap_selector_cases][128]);
 
--- a/compress.h	Fri Feb 21 06:48:58 1997
+++ b/compress.h	Sun Jun 28 19:44:59 2009
@@ -26,7 +26,7 @@
      compress  ();
      ~compress ();
 
-     char *frame_name (char postfix []);
+     char *frame_name (const char postfix []);
      void encode      (char name    []);
      void decode      (char name    []);
  
--- a/cry.hc	Fri Feb 21 06:50:13 1997
+++ b/cry.hc	Sun Jun 28 19:48:00 2009
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include "cry.h"
 
 void cry (char *func_name)
--- a/dial.h	Fri Feb 21 06:50:13 1997
+++ b/dial.h	Sun Jun 28 18:54:56 2009
@@ -35,7 +35,7 @@
 
      int  val_dx;
 
-     getline *g_value;
+     class getline *g_value;
      button  *up;
      button  *down;
      button  *up2;
@@ -46,7 +46,7 @@
      bool    with_edit;
 
    dial  (win  *w, 
-          char title [],
+          const char title [],
           int  title_dx,
           int  x, 
           int  y,
@@ -60,7 +60,7 @@
           int  incr2       = 0,
           bool auto_repeat = false);
    dial  (win  *w, 
-          char title [],
+          const char title [],
           int  title_dx,
           int  x, 
           int  y,
--- a/errorhandling.h	Wed Mar 19 08:41:06 1997
+++ b/errorhandling.h	Sun Jun 28 19:02:39 2009
@@ -13,26 +13,26 @@
 #include "stdio.h"
 
 void errorstop (int  error_no,
-                char c_class [],
-                char msg     []);
+                const char c_class [],
+                const char msg     []);
 
 void errorstop (int  error_no,
-                char c_class [],
-                char msg     [],
-                char info    []);
+                const char c_class [],
+                const char msg     [],
+                const char info    []);
 
 void errorstop (int  error_no,
-                char c_class [],
-                char msg     [],
-                char info    [],
-		char param1  []);
+                const char c_class [],
+                const char msg     [],
+                const char info    [],
+		const char param1  []);
 
 void errorstop (int  error_no,
-                char c_class [],
-                char msg     [],
-                char info    [],
-		char param1  [],
-                char param2  []);
+                const char c_class [],
+                const char msg     [],
+                const char info    [],
+		const char param1  [],
+                const char param2  []);
 
 #endif
 
--- a/file_selector.h	Fri Feb 21 06:50:14 1997
+++ b/file_selector.h	Sun Jun 28 19:41:06 2009
@@ -16,8 +16,8 @@
    win      *w;
 
    selector *sel;
-   getline  *get_name;
-   getline  *get_pattern;
+   class getline  *get_name;
+   class getline  *get_pattern;
 
    int      num_files;
    char     file_list [max_selector_cases][128];
@@ -32,17 +32,17 @@
 
    button   *cancel;
   
-   file_selector      (char name [], 
+   file_selector      (const char name [], 
                        int  x,
                        int  y,
-                       char file_name    [],
-                       char file_pattern [],
+                       const char file_name    [],
+                       const char file_pattern [],
                        bool must_exists = true);
 
-   file_selector      (char name [],
+   file_selector      (const char name [],
                        int  x,
                        int  y,
-                       char full_file_name [],
+                       const char full_file_name [],
                        bool must_exist = true);
 
    ~file_selector     ();
--- a/getline.h	Fri Feb 21 06:50:14 1997
+++ b/getline.h	Sun Jun 28 18:56:35 2009
@@ -37,7 +37,7 @@
    bool    was_deactive;
    bool    is_single_char;
    
-   getline      (char name [],
+   getline      (const char name [],
                  win  *w,
                  char string [],
                  int  x,
--- a/hcc/hcc.cc	Fri Feb 21 06:50:16 1997
+++ b/hcc/hcc.cc	Sun Jun 28 18:51:50 2009
@@ -1,8 +1,8 @@
-#include "bool.h"
-#include "stdio.h"
-#include "std.h"
-#include "string.h"
-#include "ctype.h"
+#include "../bool.h"
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <cctype>
 
 #define max_refinements 10000
 #define max_src_lines   10000
@@ -39,7 +39,7 @@
 
 
 
-void f_open (FILE *&f, char name [], char mode [])
+void f_open (FILE *&f, const char name [], const char mode [])
   {f = fopen (name, mode);
     /* check_error */      {if (mode [0] == 'r' && f == NULL)
          {printf ("1, FILE, file '%s' not found\n", name);
@@ -91,7 +91,7 @@
 
 void changeall (char s           [], 
                 int  max_length_of_s,
-                char tmplate     [],
+                const char tmplate     [],
                 char replacement [])
 
   {char *pos;
@@ -175,7 +175,7 @@
      };
   }
 
-void pass_1_error (int line_no, char msg [], int l)
+void pass_1_error (int line_no, const char msg [], int l)
   {char err_msg [256];
    char ii      [128];
 
@@ -187,7 +187,7 @@
    exit      (1);
   }
 
-void pass_1_error (int line_no, char msg [])
+void pass_1_error (int line_no, const char msg [])
   {fclose (src);
    printf ("error : %d : %s\n", line_no, msg);
    exit   (1);
--- a/install	Thu Jun 12 08:30:18 1997
+++ b/install	Sun Jun 28 18:42:20 2009
@@ -1,75 +1,75 @@
-#!/bin/sh 
+#!/bin/bash 
 
 #---------------------------------------------------------------------
 #
 # The following assigns must be changed according to your system
 #
 
-GCC=gcc
-GPP="g++"
+GCC=${CC:-gcc}
+GPP=${CXX:-g++}
 
-XLIB=/usr/X11R6/lib
+#XLIB=/usr/X11R6/lib
 
 #
 #--------------------------------------------------------------------
 
-CFLAGS="-O2 -fexpensive-optimizations -finline-functions"
+#CFLAGS="-O2 -fexpensive-optimizations -finline-functions"
 HCC=hcc/hcc
 CMP="$HCC -compiler $GCC $CFLAGS -c"
 
-LFLAGS="-lX11 -lm -L$XLIB"
+LFLAGS="${LDFLAGS} -lX11 -lm"
 
 
-echo hcc           ; gcc  -o hcc/hcc hcc/hcc.cc
+echo hcc           ; ${GPP}  -o ${HCC} hcc/hcc.cc || exit 1
 
-echo color_trans   ; $CMP color_trans
-echo xtimer        ; $CMP xtimer        
-echo option_menu   ; $CMP option_menu   
-echo setupedit     ; $CMP setupedit     
-echo robot         ; $CMP robot         
-echo land          ; $CMP land          
-echo ship          ; $CMP ship          
-echo building      ; $CMP building      
-echo dir           ; $CMP dir           
-echo object_handler; $CMP object_handler 
-echo pic           ; $CMP pic           
-echo player        ; $CMP player        
-echo field         ; $CMP field         
-echo ilfield       ; $CMP ilfield       
-echo xbm           ; $CMP xbm           
-echo cmap_edit     ; $CMP cmap_edit     
-echo cmap          ; $CMP cmap          
-echo ppm           ; $CMP ppm           
-echo compress      ; $CMP compress      
-echo table         ; $CMP table         
-echo xmath         ; $CMP xmath         
-echo io            ; $CMP io            
-echo cry           ; $CMP cry           
-echo win           ; $CMP win           
-echo dial          ; $CMP dial          
-echo multi_dial    ; $CMP multi_dial    
-echo menu          ; $CMP menu          
-echo xsystem       ; $CMP xsystem       
-echo xfile         ; $CMP xfile         
-echo ppm_handler   ; $CMP ppm_handler   
-echo errorhandling ; $CMP errorhandling 
-echo paramreader   ; $CMP paramreader   
-echo pattern_match ; $CMP pattern_match 
-echo objects       ; $CMP objects       
-echo xstring       ; $CMP xstring       
-echo id_handler    ; $CMP id_handler    
-echo masks         ; $CMP masks         
-echo buttons       ; $CMP buttons       
-echo lru           ; $CMP lru           
-echo scroller      ; $CMP scroller      
-echo selector      ; $CMP selector      
-echo cmap_selector ; $CMP cmap_selector 
-echo getline       ; $CMP getline       
-echo file_selector ; $CMP file_selector 
-echo history       ; $CMP history       
-echo craft         ; $CMP craft            
+echo color_trans   ; $CMP color_trans || exit 1
+echo xtimer        ; $CMP xtimer         || exit 1
+echo option_menu   ; $CMP option_menu    || exit 1
+echo setupedit     ; $CMP setupedit      || exit 1
+echo robot         ; $CMP robot          || exit 1
+echo land          ; $CMP land           || exit 1
+echo ship          ; $CMP ship           || exit 1
+echo building      ; $CMP building       || exit 1
+echo dir           ; $CMP dir            || exit 1
+echo object_handler; $CMP object_handler  || exit 1
+echo pic           ; $CMP pic            || exit 1
+echo player        ; $CMP player         || exit 1
+echo field         ; $CMP field          || exit 1
+echo ilfield       ; $CMP ilfield        || exit 1
+echo xbm           ; $CMP xbm            || exit 1
+echo cmap_edit     ; $CMP cmap_edit      || exit 1
+echo cmap          ; $CMP cmap           || exit 1
+echo ppm           ; $CMP ppm            || exit 1
+echo compress      ; $CMP compress       || exit 1
+echo table         ; $CMP table          || exit 1
+echo xmath         ; $CMP xmath          || exit 1
+echo io            ; $CMP io             || exit 1
+echo cry           ; $CMP cry            || exit 1
+echo win           ; $CMP win            || exit 1
+echo dial          ; $CMP dial           || exit 1
+echo multi_dial    ; $CMP multi_dial     || exit 1
+echo menu          ; $CMP menu           || exit 1
+echo xsystem       ; $CMP xsystem        || exit 1
+echo xfile         ; $CMP xfile          || exit 1
+echo ppm_handler   ; $CMP ppm_handler    || exit 1
+echo errorhandling ; $CMP errorhandling  || exit 1
+echo paramreader   ; $CMP paramreader    || exit 1
+echo pattern_match ; $CMP pattern_match  || exit 1
+echo objects       ; $CMP objects        || exit 1
+echo xstring       ; $CMP xstring        || exit 1
+echo id_handler    ; $CMP id_handler     || exit 1
+echo masks         ; $CMP masks          || exit 1
+echo buttons       ; $CMP buttons        || exit 1
+echo lru           ; $CMP lru            || exit 1
+echo scroller      ; $CMP scroller       || exit 1
+echo selector      ; $CMP selector       || exit 1
+echo cmap_selector ; $CMP cmap_selector  || exit 1
+echo getline       ; $CMP getline        || exit 1
+echo file_selector ; $CMP file_selector  || exit 1
+echo history       ; $CMP history        || exit 1
+echo craft         ; $CMP craft             || exit 1
 
-$GPP xtimer.o option_menu.o setupedit.o robot.o land.o ship.o building.o dir.o object_handler.o pic.o player.o field.o ilfield.o xbm.o cmap_edit.o cmap.o ppm.o compress.o table.o xmath.o io.o cry.o win.o dial.o multi_dial.o menu.o xsystem.o xfile.o ppm_handler.o errorhandling.o paramreader.o pattern_match.o objects.o xstring.o id_handler.o color_trans.o masks.o buttons.o lru.o scroller.o selector.o cmap_selector.o getline.o file_selector.o history.o craft.o $LFLAGS -o craft
+$GPP xtimer.o option_menu.o setupedit.o robot.o land.o ship.o building.o dir.o object_handler.o pic.o player.o field.o ilfield.o xbm.o cmap_edit.o cmap.o ppm.o compress.o table.o xmath.o io.o cry.o win.o dial.o multi_dial.o menu.o xsystem.o xfile.o ppm_handler.o errorhandling.o paramreader.o pattern_match.o objects.o xstring.o id_handler.o color_trans.o masks.o buttons.o lru.o scroller.o selector.o cmap_selector.o getline.o file_selector.o history.o craft.o $LFLAGS -o craft || exit 1
 
 
 
