--- a/masks.h	Fri Feb 21 06:52:06 1997
+++ b/masks.h	Sun Jun 28 21:11:01 2009
@@ -25,20 +25,20 @@
                  int dx, int dy, 
                  int color_1, int color_2);
 
-bool yes        (win *w, char host [], char question []);
-bool yes        (char host [], char question []);
-bool yes        (char question []);
-void ack        (char host [], char msg []);
-void ack        (char message []);
+bool yes        (win *w, const char host [], const char question []);
+bool yes        (const char host [], const char question []);
+bool yes        (const char question []);
+void ack        (const char host [], const char msg []);
+void ack        (const char message []);
 void tell       (win  *&w,
-                 char message []);
-void tell       (char host [],
+                 const char message []);
+void tell       (const char host [],
                  win  *&w,
-                 char message []);
-void tell       (char host [],
+                 const char message []);
+void tell       (const char host [],
                  win  *parent,
                  win  *&w,
-                 char message []);
+                 const char message []);
 
 bool get_line   (char s    [],
                  char name [],
--- a/menu.h	Wed Mar 19 08:41:59 1997
+++ b/menu.h	Sun Jun 28 20:10:00 2009
@@ -93,10 +93,10 @@
   
 
       menu ();
-      menu (char name [],
+      menu (const char name [],
             int  x,
             int  y,
-            char cmds [],
+            const char cmds [],
             int  wmode = m_pop_up,
             int  menu_group = std_menu_group);
 
@@ -112,7 +112,7 @@
      void release      ();
      void release      (int no);
      void write        (int m_no, char string [], bool pressed = false);
-     bool get_cmd      (char cmds [], int &p, char cmd []);
+     bool get_cmd      (const char cmds [], int &p, char cmd []);
      
   };
 
--- a/object_handler.h	Thu Apr 17 09:07:38 1997
+++ b/object_handler.h	Sun Jun 28 19:17:30 2009
@@ -297,7 +297,7 @@
   bool can_built             (int id, int x, int y, int cmd,
                               bool is_robot = false);
   bool attack_possible       (int id, int x, int y);
-  void write                 (int color, char msg []);
+  void write                 (int color, const char msg []);
   bool direct_move           (int id, int range, int &dx, int &dy);
   bool max_diff              (int x, int y, int h);
   bool min_diff              (int x, int y, int h);
--- a/option_menu.hc	Fri Mar 21 07:20:47 1997
+++ b/option_menu.hc	Sun Jun 28 18:57:14 2009
@@ -185,7 +185,7 @@
   {dial    *d_is_self;
    dial    *d_is_suny;
    dial    *d_water_world;
-   getline *seed;
+   class getline *seed;
    char    seed_s [128];
 
    init;
@@ -217,7 +217,7 @@
       d_is_suny     = new dial (w, "", 0,200, 140,is_suny);
       d_water_world = new dial (w, "", 0,200, 180,is_water_world);
       strcpy (seed_s, "");
-      seed = new getline ("", w, seed_s, 200, 250, 90, 24);
+      seed = new class getline ("", w, seed_s, 200, 250, 90, 24);
      }.
 
 .  finish
@@ -270,8 +270,8 @@
   }
 
 void player_menu (int act_menu, int &tic)
-  {getline  *g_host;
-   getline  *g_name;
+  {class getline  *g_host;
+   class getline  *g_name;
 
    init;
    session;
@@ -298,8 +298,8 @@
      }.
 
 .  show_buttons
-     {g_host  = new getline  ("",w, host [pno],60,160,180, 20);
-      g_name  = new getline  ("",w, name [pno],60,230,180, 20);
+     {g_host  = new class getline  ("",w, host [pno],60,160,180, 20);
+      g_name  = new class getline  ("",w, name [pno],60,230,180, 20);
      }.
 
 .  finish
--- a/paramreader.h	Wed Mar 19 08:42:26 1997
+++ b/paramreader.h	Sun Jun 28 21:18:22 2009
@@ -56,13 +56,13 @@
      int  num_includes;
 
 
-          paramreader (char *param_file_name);
+          paramreader (const char *param_file_name);
 
    void   dump       ();
-   char * s_param    (char name []);
+   char * s_param    (const char name []);
    double d_param    (char name []);
-   int    i_param    (char name []);
-   int    param_no   (char name []);
+   int    i_param    (const char name []);
+   int    param_no   (const char name []);
    void   set        (char name [], char value []);
    void   read_sym   (char sym  [], bool &is_eof);
    int    max_i_name ();
--- a/player.h	Fri Feb 21 06:52:14 1997
+++ b/player.h	Sun Jun 28 19:31:30 2009
@@ -24,7 +24,7 @@
 
    color_trans *ct [max_players];
 
-   getline *g_msg;
+   class getline *g_msg;
  
    int    wdx;
    int    wdy;
@@ -149,7 +149,7 @@
    void edit                ();
 
    void inform              (char msg []);
-   void write               (char msg []);
+   void write               (const char msg []);
 
    void load_pics           ();
 
@@ -184,17 +184,17 @@
    int  x_center            (int lx);
    int  y_center            (int ly);
 
-   void show_percent        (int x,int y,int dx,int dy,int p,char title []);
+   void show_percent        (int x,int y,int dx,int dy,int p,const char title []);
    void clear_percent       (int x,int y,int dx,int dy);
    void clear_status        ();
    void show_status         (int id, bool is_first_time);
 
-   void push_cmd            (char name [], 
+   void push_cmd            (const char name [], 
                              int  code, 
                              char c, 
                              int  price, 
                              int  wood);
-   void adjust_cmd          (int cmd, char c, char name []);
+   void adjust_cmd          (int cmd, char c, const char name []);
    void adjust_cmds         ();
 
    void talk                (char from [], char msg []);
--- a/ppm_handler.hc	Fri Feb 21 06:52:17 1997
+++ b/ppm_handler.hc	Sun Jun 28 20:27:49 2009
@@ -4,6 +4,7 @@
 /*                                                                      */
 /*======================================================================*/
 
+#include <cstring>
 #include "ppm_handler.h"
 #include "xfile.h"
 
--- a/scroller.h	Fri Feb 21 06:52:19 1997
+++ b/scroller.h	Sun Jun 28 18:57:40 2009
@@ -39,7 +39,7 @@
      button *button_incr;
      button *button_decr;
 
-     scroller  (char name [],
+     scroller  (const char name [],
                 win  *w,
                 int  x,
                 int  y,
--- a/selector.h	Fri Feb 21 06:52:20 1997
+++ b/selector.h	Sun Jun 28 21:31:55 2009
@@ -38,7 +38,7 @@
    bool     mark         [max_selector_cases];
    int      case_dy;   
    
-   selector (char name [], 
+   selector (const char name [], 
              win  *w,
              int  num_cases,
              char case_string [max_selector_cases][128],
--- a/table.h	Wed Mar 19 08:43:26 1997
+++ b/table.h	Sun Jun 28 19:59:03 2009
@@ -68,7 +68,7 @@
      table  (char dir [], char name [], bool &is_new, int lru_size = 0);
      ~table ();
 
-     void  check_bound (int adr, int row_no, int column_no, char action []);
+     void  check_bound (int adr, int row_no, int column_no, const char action []);
      void  save        ();
      void  load        ();
 
@@ -76,7 +76,7 @@
      void  rename      (char  new_name []);
      void  copy        (table *destination); 
 
-     int   add_column  (char name [], int type);
+     int   add_column  (const char name [], int type);
      int   column_no   (char name []);
      int   num_columns (); 
 
--- a/win.h	Wed Mar 19 08:43:52 1997
+++ b/win.h	Sun Jun 28 21:15:21 2009
@@ -129,7 +129,7 @@
 /* ROUTINES text size information (deklarations)                        */
 /*----------------------------------------------------------------------*/
 
-void text_size (char s [], char font [], int &dx, int &dy, int &num_of_lines);
+void text_size (const char s [], const char font [], int &dx, int &dy, int &num_of_lines);
 
 /*----------------------------------------------------------------------*/
 /* Default Handler (deklarations)                                       */
@@ -140,11 +140,11 @@
 extern bool        win_handler_init;
 extern paramreader *win_defaults;
 
-int   win_default_i (int param, char default_name []);
-int   win_default_i (char default_name []);
-int   win_default_c (int param, char default_name []);
-int   win_default_c (char default_name []);
-char *win_default_s (char default_name []);
+int   win_default_i (int param, const char default_name []);
+int   win_default_i (const char default_name []);
+int   win_default_c (int param, const char default_name []);
+int   win_default_c (const char default_name []);
+char *win_default_s (const char default_name []);
 int   win_color     (char color_name   []);
 
 /*----------------------------------------------------------------------*/
@@ -241,10 +241,10 @@
      int           event_mark;
      int           event_id;
 
-          win  (char title []);
+          win  (const char title []);
 
-          win  (char title [],
-                char host  [],
+          win  (const char title [],
+                const char host  [],
                 int  x,
                 int  y, 
                 int  dx,
@@ -252,8 +252,8 @@
                 bool enable        = true,
                 bool resize_enable = false);
           win  (win  *parent,
-                char title [],
-                char host  [],
+                const char title [],
+                const char host  [],
                 int  x,
                 int  y, 
                 int  dx,
@@ -289,9 +289,9 @@
      void   mark_mouse       ();
      void   scratch_mouse    ();
      void   clear            ();
-     void   set_font         (char name []);
-     void   text_size        (char string [], int &dx, int &dy);
-     void   write            (int x, int y, char   string []);
+     void   set_font         (const char name []);
+     void   text_size        (const char string [], int &dx, int &dy);
+     void   write            (int x, int y, const char   string []);
      void   write            (int x, int y, double d);
      void   write            (int x, int y, int    i);
      void   set_clip         (int x, int y, int dx, int dy);
@@ -332,13 +332,13 @@
      int    dy               ();
      void   tick             (bool just_raised);
      void   tick             ();
-     void   alloc_color      (char name [], int no);
+     void   alloc_color      (const char name [], int no);
      void   create_color_map ();
      void   draw             (polyline *p);
      void   fill             (polyline *p);
      void   xsync            ();
 
-     Window grab             (Display *dsp, Window wnd, char name []);
+     Window grab             (Display *dsp, Window wnd, const char name []);
 
      void   get_image        (XImage *&i, int x, int y, int dx, int dy);
      void   put_image        (XImage *i, int x, int y, int dx, int dy);
--- a/xfile.h	Wed Mar 19 08:44:01 1997
+++ b/xfile.h	Sun Jun 28 21:30:16 2009
@@ -16,15 +16,15 @@
 
 double d_get              (FILE *f, bool &is_eof);
 time_t f_date             (char f_name []);
-bool   f_exists           (char f_name []);
-void   f_open             (FILE *&f, char name [], char mode []);
+bool   f_exists           (const char f_name []);
+void   f_open             (FILE *&f, const char name [], const char mode []);
 void   f_open_with_extend (FILE *&f, char name [], char mode [], int max_no);
 char   *f_getline         (FILE *f,  char line [], int  max_length); 
 
 char   *f_postfix         (char full_path []);
-char   *f_name            (char full_path []);
+char   *f_name            (const char full_path []);
 char   *f_tail            (char full_path []);
-char   *f_path            (char full_path []);
+char   *f_path            (const char full_path []);
 char   *f_home_dir        (char full_path []);
 bool   f_is_pattern       (char f_name    []);
 bool   f_is_dir           (char f_name    []);
@@ -35,8 +35,8 @@
                            char f_pat  [],
                            char mode   []);
 
-char   *complete          (char name [],
-                           char tail []);
+char   *complete          (const char name [],
+                           const char tail []);
 
 void   bprintf            (FILE *f, int num_bytes, int i);
 bool   bscanf             (FILE *f, int num_bytes, int &i);
--- a/xstring.h	Fri Feb 21 06:52:27 1997
+++ b/xstring.h	Sun Jun 28 20:26:26 2009
@@ -14,8 +14,8 @@
                  int  max_length_of_s,
                  char tmpalte     [],
                  char replacement []);
-char *substring (char s [], int from, int to);
-char *substring (char s [], int from); 
+char *substring (const char s [], int from, int to);
+char *substring (const char s [], int from); 
 int  submatch   (char s [], char p [], int &pos);
 void strcat     (char s [], int len, char app []);
 void delchar    (char s [], int pos);
