--- leksbot-1.2.orig/kataxwr.c
+++ leksbot-1.2/kataxwr.c
@@ -4,7 +4,7 @@
 #define SPACEPIN 500
 #define SIZE 21
 #define SIZE1 1000
-#define FILEO "Index"
+#define FILEO "/var/lib/leksbot/Index"
 
 int
 main ()
@@ -12,7 +12,8 @@
 
   FILE *fp;
   char leksi[SIZE], simasia[SIZE1], h, l =
-    0, space1[SPACEPIN][SIZE], noncase ();
+    0, space1[SPACEPIN][SIZE];
+  int noncase( char[SIZE] );
   static char space[] = " ";
   int v, m, e, r, a = 0;
 
@@ -27,7 +28,7 @@
     }
 
   l++;
-  gets (leksi);
+  fgets(leksi,sizeof(leksi),stdin);
   *leksi = noncase (leksi);
   m = strlen (leksi);
   r = 0;
@@ -43,7 +44,7 @@
   strcat (leksi, space1[a]);
   a++;
   puts ("dwsete tin simasia");
-  gets (simasia);
+  fgets(simasia,sizeof(simasia),stdin);
   fprintf (fp, "#%s\n%s\n", leksi, simasia);
   printf ("Do you want to add an other term?(y-n)\n");
   h = getchar ();
--- leksbot-1.2.orig/lekshead.h
+++ leksbot-1.2/lekshead.h
@@ -5,33 +5,33 @@
 #define MAX 1000
 #define CLS "clear"
 
-#define VER "   This is the LEKSBOT program Version 1,2
-		Coded and designed by Nick Apostolakis.
-		Compiled with the GNU's GCC compiler.
-		Please remember that LEKSBOT is 
-		under GNU GPL but the copyright
-		belongs to Nick Apostolakis.
-		It is distributed as is with no
-		warranty of any kind for any damage this
-		program may cause to your computer
+#define VER "   This is the LEKSBOT program Version 1,2\n\
+		Coded and designed by Nick Apostolakis.\n\
+		Compiled with the GNU's GCC compiler.\n\
+		Please remember that LEKSBOT is \n\
+		under GNU GPL but the copyright\n\
+		belongs to Nick Apostolakis.\n\
+		It is distributed as is with no\n\
+		warranty of any kind for any damage this\n\
+		program may cause to your computer\n\
 		or fitness for any particular purpose."
 
 
-#define HEL "   LEKSBOT is an explanatory dictionary
-		of botanical and biological terms.
-		LEKSBOT accepts the following commands:
-		-v   Displays the current version of the program.
-		-h   Displays this help.
+#define HEL "   LEKSBOT is an explanatory dictionary\n\
+		of botanical and biological terms.\n\
+		LEKSBOT accepts the following commands:\n\
+		-v   Displays the current version of the program.\n\
+		-h   Displays this help.\n\
 		 -man Displays the manual of the program."
 		 
-#define MAN "    This is the manual of the LEKSBOT program.
-		 This is the rule that you must follow 
-		 to search for a term:
-		 The largest term that you can search for
-		 is one with 20 letters.
-		 Note: LEKSBOT is from now on case insensitive 
-		 so you don't have to worry for
-		 upper case and lower case letters in your term.
-		 Multiple term query is not supported yet. This means for example that
-		 you can't search for all the terms that have 
+#define MAN "    This is the manual of the LEKSBOT program.\n\
+		 This is the rule that you must follow \n\
+		 to search for a term:\n\
+		 The largest term that you can search for\n\
+		 is one with 20 letters.\n\
+		 Note: LEKSBOT is from now on case insensitive \n\
+		 so you don't have to worry for\n\
+		 upper case and lower case letters in your term.\n\
+		 Multiple term query is not supported yet. This means for example that\n\
+		 you can't search for all the terms that have \n\
 		 the first 3 letters and exist in LEKSBOT's database."		 
--- leksbot-1.2.orig/leksiko.c
+++ leksbot-1.2/leksiko.c
@@ -64,7 +64,7 @@
       printf ("\t\t\tPlease insert the term\n\n\n\n");
       printf
 	("\t\t\t#____________________#\b\b\b\b\b\b\b\b\b\b\b\b\b\b\b\b\b\b\b\b\b");
-      gets (oros);
+      fgets(oros,sizeof(oros),stdin);
       *oros = fillup (oros);
       *oros = noncase (oros);	
       o = searchterm (oros);
--- leksbot-1.2.orig/debian/KATAXWR.1
+++ leksbot-1.2/debian/KATAXWR.1
@@ -0,0 +1,26 @@
+.TH KATAXWR 1 "February 10, 2001"
+.SH NAME
+KATAXWR \- LEKSBOT database editor
+.SH SYNOPSIS
+.B KATAXWR
+.SH DESCRIPTION
+This manual page documents briefly the
+.B KATAXWR
+command.
+This manual page was written for the Debian GNU/Linux distribution
+because the original program does not have a manual page.
+.PP
+\fBKATAXWR\fP is a simple utility with which you can add terms in your LEKSBOT database.
+Just type KATAXWR and you are prompted to enter the item to be added. Then you have to
+enter the brief explanation. In Debian the LEKSBOT database is installed in
+/var/lib/leksbot/Index, it is world writable so that all users can
+add to the database.
+.SH OPTIONS
+There are none. The program is interactive and selfexplaining.
+.SH SEE ALSO
+.BR LEKSBOT (1),
+.BR leksiko (1).
+.br
+.SH AUTHOR
+This manual page was written by Dr. Guenter Bechly <gbechly@debian.org>,
+for the Debian GNU/Linux system (but may be used by others).
--- leksbot-1.2.orig/debian/LEKSBOT.1
+++ leksbot-1.2/debian/LEKSBOT.1
@@ -0,0 +1,37 @@
+.TH LEKSBOT 1 "February 10, 2001"
+.SH NAME
+LEKSBOT \- An explanatory dictionary of botanic and biological terms
+.SH SYNOPSIS
+.B LEKSBOT
+.RI [ options ] " database-file" ...
+.SH DESCRIPTION
+This manual page documents briefly the
+.B LEKSBOT
+command.
+This manual page was written for the Debian GNU/Linux distribution
+because the original program does not have a manual page.
+.PP
+\fBLEKSBOT\fP is a LEKSBOT is an explanatory dictionary of botanic
+and biological terms. Currently it contains about 1500 terms but the
+number is growing up and will cover other sciences relative with
+biology (entomology, etc.). The LEKSBOT database can be edited with
+\fBKATAXWR\fP. You can call LEKSBOT with the correct path to the
+installed dictionary with the tiny script \fBleksiko\fP.
+.SH OPTIONS
+A summary of options is included below.
+.TP
+.B \-h
+Show this summary of options.
+.TP
+.B \-v
+Show version of program.
+.TP
+.B \-man
+Show brief manual.
+.SH SEE ALSO
+.BR leksiko (1),
+.BR KATAXWR (1).
+.br
+.SH AUTHOR
+This manual page was written by Dr. Guenter Bechly <gbechly@debian.org>,
+for the Debian GNU/Linux system (but may be used by others).
--- leksbot-1.2.orig/debian/changelog
+++ leksbot-1.2/debian/changelog
@@ -0,0 +1,83 @@
+leksbot (1.2-8) unstable; urgency=low
+
+  * Cleaned up debian/rules
+  * Moved leksbot from section text to more appropriate
+    section science
+  * Added 'dh_install' to install target of the debian/rules file
+    instead of using '$(MAKE) install DESTDIR=$$(pwd)/debian/leksbot'
+  * Patched kataxwr.c, conflicting types for 'noncase'; Closes: #259016
+
+ -- Brian R Furry <bfurry@nac.net>  Sun, 8 Aug 2004 03:37:19 +0000
+
+leksbot (1.2-7) unstable; urgency=low
+
+  * Fixed the multiline string problem; Closes: #194946
+  * updated to follow the latest version of 
+    Debian Policy (Standards-Version 3.5.10)
+
+ -- Brian R Furry <bfurry@nac.net>  Sun,  8 Jun 2003 20:42:54 -0400
+
+leksbot (1.2-6) unstable; urgency=low
+
+  * Correct bad maintainer field in changelog
+
+ -- Brian R Furry <bfurry@nac.net>  Sun,  6 Apr 2003 08:51:48 -0400
+
+leksbot (1.2-5) unstable; urgency=low
+
+  * Fixed insecure and unnecessary setuid-root binary; Closes: #186421
+  * Fixed modifiable database stored in /usr; Closes: #186422
+  * As leksbot has no merge function for its database, mark Index as a
+    conffile to allow users to merge by hand on upgrade
+
+ -- Camm Maguire <camm@wisdom.m.enhanced.com>  Sun,  6 Apr 2003 02:50:31 +0000
+
+leksbot (1.2-4) unstable; urgency=low
+
+  * Removed README.Debian file; Closes: #184047
+  * Fixed debian/copyright file to comply with policy
+  * cleaned emacs cruft in debian/changelog
+  * updated my current email address
+  * gets -> fgets in C source files
+  * DH_COMPAT version 3
+  * remove unused dh_installmenu
+  * Add clean target to main Makefile four clean source package builds
+
+ -- Brian R Furry <bfurry@nac.net>  Thu, 20 Mar 2003 09:42:36 -0500
+
+leksbot (1.2-3.2) unstable; urgency=low
+
+  * new maintainer
+
+ -- Brian R Furry <fbrian@nac.net>  Sat, 18 May 2002 22:04:16 -0400
+
+leksbot (1.2-3.1) unstable; urgency=low
+
+  * Don't have KATAXWR setuid, rather make the Index world writable.
+
+ -- David Kimdon <dwhedon@debian.org>  Fri,  9 Nov 2001 00:43:36 -0800
+
+leksbot (1.2-3) unstable; urgency=low
+
+  * Maintainer set to Debian QA Group <packages@qa.debian.org>.
+
+ -- Adrian Bunk <bunk@fs.tum.de>  Wed, 29 Aug 2001 21:27:43 +0200
+
+leksbot (1.2-2) unstable; urgency=low
+
+  * Corrected typo in copyright file. 
+  * Modified clean process in rules.
+
+ -- Dr. Guenter Bechly <gbechly@debian.org>  Thu, 29 Mar 2001 12:36:01 +0200
+
+leksbot (1.2-1) unstable; urgency=low
+
+  * Initial Release; closes: #85479
+  * Added a Makefile and two manpages for Debian
+  * Added script leksiko
+  * Set KATAXWR setuid so that every user can edit the lexikon Index
+  * Edited path to Index in kataxwr.c 
+
+ -- Dr. Guenter Bechly <gbechly@debian.org>  Sat, 10 Feb 2001 13:16:34 +0100
+
+
--- leksbot-1.2.orig/debian/compat
+++ leksbot-1.2/debian/compat
@@ -0,0 +1 @@
+4
--- leksbot-1.2.orig/debian/conffiles
+++ leksbot-1.2/debian/conffiles
@@ -0,0 +1 @@
+/var/lib/leksbot/Index
--- leksbot-1.2.orig/debian/leksbot.install
+++ leksbot-1.2/debian/leksbot.install
@@ -0,0 +1,4 @@
+KATAXWR /usr/bin  
+LEKSBOT /usr/bin
+leksiko /usr/bin
+Index /var/lib/leksbot/
--- leksbot-1.2.orig/debian/control
+++ leksbot-1.2/debian/control
@@ -0,0 +1,17 @@
+Source: leksbot
+Section: science
+Priority: optional
+Maintainer: Brian R Furry <bfurry@nac.net>
+Build-Depends: debhelper ( >= 4)
+Standards-Version: 3.6.1
+
+Package: leksbot
+Architecture: any
+Depends: ${shlibs:Depends}
+Description: An explanatory dictionary of botanic and biological terms
+ LEKSBOT is an explanatory dictionary of botanic and biological terms.
+ Currently it contains about 1500 terms but the number is growing up and
+ will cover other sciences relative with biology (entomology, etc.).
+ .
+ The included program KATAXWR is a simple utility with which you can add
+ terms to the database.
--- leksbot-1.2.orig/debian/rules
+++ leksbot-1.2/debian/rules
@@ -0,0 +1,53 @@
+#!/usr/bin/make -f
+
+build: build-stamp
+build-stamp:
+	dh_testdir
+
+	# Add here commands to compile the package.
+	$(MAKE)
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp configure-stamp
+
+	# Add here commands to clean up after the build process.
+	-$(MAKE) clean
+	-rm -f KATAXWR LEKSBOT
+
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs -v
+
+	# Add here commands to install the package into debian/leksbot.
+#	$(MAKE) install DESTDIR=$$(pwd)/debian/leksbot
+	chmod 0755 leksiko
+	dh_install -v
+
+# Build architecture-dependent files here.
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_installdocs
+	dh_installmenu
+	dh_installmanpages LEKSBOT KATAXWR leksiko
+	dh_installchangelogs 
+	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install configure
--- leksbot-1.2.orig/debian/copyright
+++ leksbot-1.2/debian/copyright
@@ -0,0 +1,15 @@
+This package was debianized by Brian R. Furry <bfurry@nac.net> on
+Sun, 16 Mar 2003 11:08:52 -0500.
+
+It was downloaded from http://agriroot.aua.gr/~nickapos/leksiko/gpl/leksbot.tgz
+
+Upstream Author: Nick Apostolakis <nickapos@agriroot.aua.gr>
+
+Copyright: GPL
+
+This software is copyright (c) by Nick Apostolakis
+
+You are free to distribute this software under the terms of
+the GNU General Public Licence.
+On Debian systems, the complete text of the GNU General Public
+Licence can be found in /usr/share/common-licenses/GPL file.
--- leksbot-1.2.orig/debian/dirs
+++ leksbot-1.2/debian/dirs
@@ -0,0 +1,3 @@
+usr/bin
+usr/share/doc/leksbot
+usr/share/man/man1
--- leksbot-1.2.orig/debian/docs
+++ leksbot-1.2/debian/docs
@@ -0,0 +1 @@
+readme
--- leksbot-1.2.orig/debian/menu
+++ leksbot-1.2/debian/menu
@@ -0,0 +1,5 @@
+?package(leksbot):\
+ needs="text"\
+ section="Apps/Science"\
+ title="leksbot"\
+ command="/usr/bin/LEKSBOT"
--- leksbot-1.2.orig/debian/leksiko.1
+++ leksbot-1.2/debian/leksiko.1
@@ -0,0 +1,22 @@
+.TH LEKSIKO 1 "Februar 10, 2001"
+.SH NAME
+leksiko \- An explanatory dictionary of botanic and biological terms
+.SH SYNOPSIS
+.B leksiko
+.SH DESCRIPTION
+This manual page documents briefly the
+.B leksiko
+command.
+.PP
+\fBleksiko\fP is a tiny script for Debian that calls LEKSBOT with the
+path to the Index file. The LEKSBOT database can be edited with
+\fBKATAXWR\fP.
+.SH OPTIONS
+There are none.
+.SH SEE ALSO
+.BR LEKSBOT (1),
+.BR KATAXWR (1).
+.br
+.SH AUTHOR
+This manual page was written by Dr. Guenter Bechly <gbechly@debian.org>,
+for the Debian GNU/Linux system (but may be used by others).
--- leksbot-1.2.orig/Makefile
+++ leksbot-1.2/Makefile
@@ -0,0 +1,25 @@
+# Makefile added for Debian by Dr. Guenter Bechly 
+
+LEKSBOT KATAXWR:
+	gcc leksiko.c -O2 -o LEKSBOT
+	gcc kataxwr.c -O2 -o KATAXWR
+
+DESTDIR   =
+BINDIR    = $(DESTDIR)/usr/bin
+INDEXDIR = $(DESTDIR)/var/lib/leksbot
+
+clean:
+	rm -f LEKSBOT KATAXWR
+
+install: 
+	install -d $(BINDIR)
+	install leksiko  $(BINDIR)
+	install LEKSBOT  $(BINDIR)
+	install KATAXWR $(BINDIR)
+	install -d $(INDEXDIR)
+	install Index  $(INDEXDIR)
+
+#!/bin/sh
+	chmod 644 $(INDEXDIR)/Index
+	chmod 755 $(BINDIR)/leksiko
+	chmod 755 $(BINDIR)/KATAXWR
--- leksbot-1.2.orig/leksiko
+++ leksbot-1.2/leksiko
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+/usr/bin/LEKSBOT /var/lib/leksbot/Index
