diff -ru src.old/analyzer/ALoc.cc src/analyzer/ALoc.cc
--- src.old/analyzer/ALoc.cc	2011-01-09 15:52:51.000000000 -0500
+++ src/analyzer/ALoc.cc	2011-01-09 15:54:28.000000000 -0500
@@ -23,7 +23,7 @@
 //ALocSet::ALocSet(C_Decl& d): Numbered(Numbered::ALOCSET) { insert(&d); }
 ALocSet::ALocSet(C_Decl* d): Numbered(Numbered::ALOCSET) { insert(d); }
 
-ostream& operator<<(ostream& s, Pset<C_Decl> &a) {
+std::ostream& operator<<(std::ostream& s, Pset<C_Decl> &a) {
   if ( a.empty() )
     return s << "(none)" ;
   bool first = true ;
@@ -35,7 +35,7 @@
   return s ;
 }
 
-ostream& operator<<(ostream& s, C_Decl &d) {
+std::ostream& operator<<(std::ostream& s, C_Decl &d) {
   switch(d.tag) {
   case ExtFun:
     return s << "extern " << d.get_name() ;
diff -ru src.old/analyzer/ALoc.h src/analyzer/ALoc.h
--- src.old/analyzer/ALoc.h	2011-01-09 15:52:51.000000000 -0500
+++ src/analyzer/ALoc.h	2011-01-09 15:54:28.000000000 -0500
@@ -22,8 +22,8 @@
   ALocSet(C_Decl* d);
 };
 
-ostream& operator<<(ostream& s, Pset<C_Decl> &a);
-ostream& operator<<(ostream& s, C_Decl &d);
+std::ostream& operator<<(std::ostream& s, Pset<C_Decl> &a);
+std::ostream& operator<<(std::ostream& s, C_Decl &d);
 
 class Outcore ;
 
