diff -ru src.old/analyzer/auxilary.cc src/analyzer/auxilary.cc
--- src.old/analyzer/auxilary.cc	2010-12-03 01:33:55.000000000 -0500
+++ src/analyzer/auxilary.cc	2010-12-03 01:37:16.000000000 -0500
@@ -135,10 +135,10 @@
 char* stringDup(const char* str)
 {
     if (str == NULL) return NULL;
-    int len = std::strlen(str);
+    int len = strlen(str);
     char* newStr = new char[len+1];
     assert(newStr!=NULL);
-    std::strcpy(newStr,str);
+    strcpy(newStr,str);
     return newStr;
 }
