diff -rNu comparator-2.12.old/setup.py comparator-2.12/setup.py
--- comparator-2.12.old/setup.py	2022-12-03 17:47:56.159962720 -0500
+++ comparator-2.12/setup.py	2022-12-03 18:32:44.675962720 -0500
@@ -1,6 +1,5 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
-setup(name="comparator",
-      version="2.0",
-      py_modules=["comparator"])
+import setuptools
+if __name__ == "__main__":
+    setuptools.setup()
