diff -ru a/src/editor/ThingDialog.java b/src/editor/ThingDialog.java
--- a/src/editor/ThingDialog.java	2012-02-14 02:08:48.000000000 -0500
+++ b/src/editor/ThingDialog.java	2012-02-14 02:09:19.000000000 -0500
@@ -156,7 +156,7 @@
      * Get the selected terrain type for this thing.
      */
     public short
-    getType() {
+    getSelectedType() {
         Terrain     t = (Terrain)type.getSelectedItem();
         return t.getId();
     }
@@ -200,7 +200,7 @@
 
     public Thing
     getThing() {
-        thing.setType(getType());
+        thing.setType(getSelectedType());
         thing.setName(getName());
         thing.setDescription(getDescription());
         thing.setFontSize(getFontSize());
