diff -r ecd4b4754511 -r f0654e02168b src/options.cc
--- a/src/options.c	Sat Sep 11 21:42:42 2010 +0200
+++ b/src/options.c	Sat Sep 11 22:08:29 2010 +0200
@@ -91,6 +91,6 @@
 			} else {
 				STRDUP(language[0], optarg);
-				if ((s = index(language[0], ':')) != NULL) {
+				if ((s = const_cast<char*>(index(language[0], ':'))) != NULL) {
 					*s++  = 0;
 					language[1] = s;
 				}
@@ -182,6 +182,6 @@
 			} else {
 				STRDUP(language[0], optarg);
-				if ((s = index(language[0], ':')) != NULL) {
+				if ((s = const_cast<char*>(index(language[0], ':'))) != NULL) {
 					*s++  = 0;
 					language[1] = s;
 				}
