From 33fb4b9b322d7607f8055bc23ba8cf2ee3b17d01 Mon Sep 17 00:00:00 2001
From: "Artur R. Czechowski" <arturcz@hell.pl>
Date: Sun, 18 Mar 2012 01:28:03 +0100
Subject: [PATCH] makes compilation verbose

---
 imms/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imms/Makefile b/imms/Makefile
index 8ea29c2..85b0d3c 100644
--- a/imms/Makefile
+++ b/imms/Makefile
@@ -52,4 +52,4 @@ dist: immsconf.h distclean
 vars.mk:;
 
 %:
-	@$(MAKE) -C build --no-print-directory $@
+	$(MAKE) -C build $@
