--- Makefile.am.old	2000-11-01 10:16:19.000000000 -0700
+++ Makefile.am	2005-07-29 11:08:24.000000000 -0600
@@ -21,9 +21,9 @@
 	make install_gamedata target=@GAME_INSTALLDIR@
 
 install_gamedata:
-	sh mkinstalldirs $(target)/
+	sh mkinstalldirs $(DESTDIR)$(target)/
 	cp -rv sounds \
-               $(target)/
+               $(DESTDIR)$(target)/
 
 # Rule to build tar-gzipped distribution package
 $(PACKAGE)-$(VERSION).tar.gz: dist
