--- megahal-9.1.1a.orig/Makefile
+++ megahal-9.1.1a/Makefile
@@ -18,7 +18,9 @@
 
 # DEBUG=-DDEBUG
 TCLVERSION=8.3
-TCLINCLUDE=-I/usr/include/tcl$(TCLVERSION)
+# /usr/include/tcl is provided by tcl-dev on Debian
+#TCLINCLUDE=-I/usr/include/tcl$(TCLVERSION)
+TCLINCLUDE=-I/usr/include/tcl
 CFLAGS=-g -Wall
 
 
--- megahal-9.1.1a.orig/quick-learn.pl
+++ megahal-9.1.1a/quick-learn.pl
@@ -0,0 +1,13 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use Megahal;
+
+Megahal::megahal_initialize();
+
+while (<STDIN>) {
+	Megahal::megahal_learn_no_reply($_, 1);
+}
+
+Megahal::megahal_cleanup();
--- megahal-9.1.1a.orig/python-interface.c
+++ megahal-9.1.1a/python-interface.c
@@ -7,7 +7,7 @@
 static PyObject *mhinitbrain(PyObject *self, PyObject *args)
 {
     megahal_initialize();
-    return Py_None;
+	return Py_BuildValue("");
 }
 
 static PyObject *mhdoreply(PyObject *self, PyObject *args)
@@ -32,13 +32,13 @@
 
     megahal_learn_no_reply(input, 1);
 
-    return Py_None;
+	return Py_BuildValue("");
 }
 
 static PyObject *mhcleanup(PyObject *self, PyObject *args)
 {
     megahal_cleanup();
-    return Py_None;
+	return Py_BuildValue("");
 }
 
 static struct PyMethodDef mh_methods[] = {
--- megahal-9.1.1a.orig/megahal.c
+++ megahal-9.1.1a/megahal.c
@@ -225,8 +225,13 @@
 static bool nowrap=FALSE;
 static bool nobanner=FALSE;
 
+/* WARNING : if you ever change the following name,
+ * please change the malloc below as well.
+ */
+
 static char *errorfilename = "megahal.log";
 static char *statusfilename = "megahal.txt";
+static char *dictionaryfilename = "megahal.dic";
 static DICTIONARY *words=NULL;
 static DICTIONARY *greets=NULL;
 static MODEL *model=NULL;
@@ -240,7 +245,7 @@
 static DICTIONARY *grt=NULL;
 static SWAP *swp=NULL;
 static bool used_key;
-static char *directory=NULL;
+static char *directory=".";
 static char *last=NULL;
 
 static COMMAND command[] = {
@@ -408,11 +413,19 @@
 
 void megahal_initialize(void)
 {
+    char *filenamebuff;
     errorfp = stderr;
     statusfp = stdout;
+    
+    filenamebuff = (char *) malloc (strlen (directory) + strlen(SEP) + 12);
+
+    sprintf(filenamebuff, "%s%s%s", directory, SEP, errorfilename);
+    initialize_error(filenamebuff);
+
+    sprintf(filenamebuff, "%s%s%s", directory, SEP, statusfilename);
+    initialize_status(filenamebuff);
+    free (filenamebuff);
 
-    initialize_error(errorfilename);
-    initialize_status(statusfilename);
     ignore(0);
 
 #ifdef AMIGA
@@ -1371,7 +1384,7 @@
 void load_dictionary(FILE *file, DICTIONARY *dictionary)
 {
     register int i;
-    int size;
+    BYTE4 size;
 
     fread(&size, sizeof(BYTE4), 1, file);
     progress("Loading dictionary", 0, 1);
@@ -1853,9 +1866,14 @@
     register unsigned int i;
     register unsigned int j;
     FILE *file;
+    char *filename;
 
-    file=fopen("megahal.dic", "w");
-    if(file==NULL) {
+    filename = (char *) malloc (sizeof(char)*(strlen(directory)+strlen(SEP)+12));
+
+    sprintf (filename, "%s%s%s", directory, SEP, dictionaryfilename);
+    file = fopen(filename, "w");
+    free (filename);
+    if(file == NULL) {
 	warn("show_dictionary", "Unable to open file");
 	return;
     }
--- megahal-9.1.1a.orig/quick-learn.py
+++ megahal-9.1.1a/quick-learn.py
@@ -0,0 +1,20 @@
+#!/usr/bin/python
+
+# MegaHAL quick learning script.
+# By Laurent Fousse <laurent@komite.net> GPL 2003
+#
+# Usage: quick-learn < text-file
+# reads text-file linewise and feeds it to megahal
+
+
+import mh_python
+import sys
+
+mh_python.initbrain()
+
+while 1:
+	ligne = sys.stdin.readline()
+	if not ligne: break
+	mh_python.learn(ligne)
+
+mh_python.cleanup()
--- megahal-9.1.1a.orig/docs/megahal.1
+++ megahal-9.1.1a/docs/megahal.1
@@ -12,8 +12,8 @@
 conversation. It is possible to teach MegaHAL to talk about new topics,
 and in different languages.
 .BR megahal-personal
-is a varient of megahal that saves its state files in ~/.megahal/ when it
-exits. This lets each user have thier own personal megahal.
+is a variant of megahal that saves its state files in ~/.megahal/ when it
+exits. This lets each user have their own personal megahal.
 .SH OPTIONS
 .IP "--no-prompt, -p"
 Do not display prompt when asking user for input. This is a debian-specific
@@ -24,9 +24,14 @@
 modification, and is useful mainly if you want to interface megahal to
 another program.
 .IP "--no-banner, -b"
-Supress the startup banner. This is a debian-specific
+Suppress the startup banner. This is a debian-specific
 modification, and is useful mainly if you want to interface megahal to
 another program.
+.SH WARNING
+MegaHAL allows multiple lines to form a single input. Thus a single
+newline character is parsed as any other token and MegaHAL can output
+replies consisting of multiple lines. To end an input, you need to
+type enter twice.
 .SH "SEE ALSO"
 /usr/share/doc/megahal/README.TXT.gz
 .SH AUTHOR
--- megahal-9.1.1a.orig/Megahal/const-c.inc
+++ megahal-9.1.1a/Megahal/const-c.inc
@@ -39,11 +39,11 @@
 my $types = {map {($_, 1)} qw(IV)};
 my @names = (qw(MEGAHAL_H));
 
-print constant_types(); # macro defs
+print constant_types(), "\n"; # macro defs
 foreach (C_constant ("Megahal", 'constant', 'IV', $types, undef, 3, @names) ) {
     print $_, "\n"; # C constant subs
 }
-print "#### XS Section:\n";
+print "\n#### XS Section:\n";
 print XS_constant ("Megahal", $types);
 __END__
    */
--- megahal-9.1.1a.orig/Megahal/const-xs.inc
+++ megahal-9.1.1a/Megahal/const-xs.inc
@@ -22,12 +22,14 @@
            Second, if present, is found value */
         switch (type) {
         case PERL_constant_NOTFOUND:
-          sv = sv_2mortal(newSVpvf("%s is not a valid Megahal macro", s));
+          sv =
+	    sv_2mortal(newSVpvf("%s is not a valid Megahal macro", s));
           PUSHs(sv);
           break;
         case PERL_constant_NOTDEF:
           sv = sv_2mortal(newSVpvf(
-	    "Your vendor has not defined Megahal macro %s, used", s));
+	    "Your vendor has not defined Megahal macro %s, used",
+				   s));
           PUSHs(sv);
           break;
         case PERL_constant_ISIV:
--- megahal-9.1.1a.orig/debian/control
+++ megahal-9.1.1a/debian/control
@@ -0,0 +1,26 @@
+Source: megahal
+Section: misc
+Priority: optional
+Build-Depends: debhelper (>= 7), tcl-dev, python-all-dev (>= 2.3.5-11), python-support (>= 0.3)
+Maintainer: Debian QA Group <packages@qa.debian.org>
+Standards-Version: 3.9.1
+Homepage: http://megahal.alioth.debian.org/
+
+Package: megahal
+Architecture: any
+Depends: ${shlibs:Depends}, ${perl:Depends}, ${misc:Depends}
+Provides: ${python:Provides}
+Suggests: ${python:Depends}
+Description: conversation simulator that can learn as you talk to it
+ Conversation simulators are computer programs which give the appearance of
+ conversing with a user in natural language. Such programs are effective
+ because they exploit the fact that human beings tend to read much more
+ meaning into what is said than is actually there; we are fooled into reading
+ structure into chaos, and we interpret non-sequitur as valid conversation.
+ .
+ MegaHAL differs from conversation simulators such as Eliza in that it uses
+ a Markov Model to learn how to hold a conversation. It is possible to
+ teach MegaHAL to talk about new topics, and in different languages.
+ .
+ This package contains the megahal binary, and for your scripting
+ pleasure modules for perl, Python and Tcl.
--- megahal-9.1.1a.orig/debian/clean
+++ megahal-9.1.1a/debian/clean
@@ -0,0 +1 @@
+Megahal/Makefile.old
--- megahal-9.1.1a.orig/debian/Hal.pm
+++ megahal-9.1.1a/debian/Hal.pm
@@ -0,0 +1,101 @@
+#!/usr/bin/perl -w
+#
+# Communicate 2-way with a megahal process. This is a mess because
+# I have to handle all the IO myself using a select loop.
+#
+# Joey Hess <joey@kitenet.net> GPL 1998
+
+package Hal;
+use strict;
+use IPC::Open2;
+use Fcntl;
+use IO::Select;
+
+my $rselect=IO::Select->new();
+my $wselect=IO::Select->new();
+
+# Start megahal running and return his first line of output.
+sub start {
+	my $progname=shift || "megahal-personal";
+	
+	my $pid=open2(\*READ, \*WRITE, $progname, 
+		"--no-prompt", "--no-wrap", "--no-banner");
+	_nonblock(\*READ);
+	_nonblock(\*WRITE);
+	$rselect->add(\*READ);
+	$wselect->add(\*WRITE);
+	# Now read the first line he outputs which we will return.
+	return Hal::read();
+}
+
+# Pass a line of input to hal, returns everything he says until he expects
+# input from you again. If the input is trusted, you must pass 1 as the
+# second parameter. Otherwise, #-commands are ignored.
+sub talk { 
+	my $line=shift;
+	my $trusted=shift;
+	
+	$line=~s/^\s*//;
+	$line=~tr/ 	/ /s;
+	
+	return if (!$trusted && $line=~m/^\s*#/);
+	
+	$wselect->can_write();
+	# TODO: partial write detection &etc.
+	syswrite(WRITE,"$line\n\n",length($line)+3) || print "$!\n";
+	
+	return Hal::read();
+}
+
+# Read any pending output. There _must_ be some, or it'll block.
+sub read {
+	my $done=0;
+	my $ret='';
+	until ($done) {
+		$rselect->can_read();
+		if (sysread(READ,$_,1024)) {
+			$done=1 if (/\n/);
+			$ret.=$_;
+		}
+	}
+	return $ret;
+}
+
+# Pass a line of input to hal, in quiet mode. He listens, but does not reply.
+# See above for trusted input docs.
+sub tell {
+	my $line=shift;
+	my $trusted=shift;
+
+	$line=~s/^\s*//;
+	$line=~tr/ 	/ /s;
+
+	return if (!$trusted && $line=~m/^\s*#/);
+	
+	$wselect->can_write();
+	# TODO: partial write detection &etc.
+	syswrite(WRITE,"#quiet\n\n$line\n\n#quiet\n\n",length($line)+20) || print "$!\n";
+}
+
+# Exit cleanly.
+sub quit {
+	$wselect->can_write();
+	syswrite(WRITE,"#quit\n\n",8);
+}
+
+# Save brain.
+sub save {
+	$wselect->can_write();
+	syswrite(WRITE,"#quiet\n\n#save\n\n#quiet\n\n");
+}
+
+# Set a socket into nonblocking mode.
+sub _nonblock {
+	my $socket=shift;
+	my $flags= fcntl($socket, F_GETFL, 0) 
+		or die "Can't get flags for socket: $!\n";
+	fcntl($socket, F_SETFL, $flags | O_NONBLOCK)
+		or die "Can't make socket nonblocking: $!\n";
+}
+
+1
--- megahal-9.1.1a.orig/debian/pyversions
+++ megahal-9.1.1a/debian/pyversions
@@ -0,0 +1 @@
+2.4-
--- megahal-9.1.1a.orig/debian/docs
+++ megahal-9.1.1a/debian/docs
@@ -0,0 +1,2 @@
+docs/paper.txt
+docs/README.TXT
--- megahal-9.1.1a.orig/debian/dirs
+++ megahal-9.1.1a/debian/dirs
@@ -0,0 +1,2 @@
+usr/bin
+usr/lib/megahal
--- megahal-9.1.1a.orig/debian/copyright
+++ megahal-9.1.1a/debian/copyright
@@ -0,0 +1,18 @@
+This package was originally debianized by Joey Hess <joeyh@debian.org>.
+
+The next debian maintainer was David N. Welton <davidw@debian.org>.
+
+It was downloaded from http://megahal.sourceforge.net/
+
+Upstreams Authors: Jason Hutchens <jason_huntchens@users.sourceforge.net>. 
+                   David N. Welton <davidw@debian.org>
+
+This software is Copyright (C) 2000-2004 Jason Hutchens, David N. Welton and
+others.
+
+You are free to distribute this software under the terms of
+the GNU General Public License, version 2, or (at your option)
+any later version.
+
+On Debian systems, the complete text of version 2 of the GNU General Public
+License can be found in the file `/usr/share/common-licenses/GPL-2'.
--- megahal-9.1.1a.orig/debian/menu
+++ megahal-9.1.1a/debian/menu
@@ -0,0 +1,3 @@
+?package(megahal):needs="text" section="Applications/Science/Social" \
+	title="MegaHal" longtitle="Talk with MegaHal" \
+	command="/usr/bin/megahal"
--- megahal-9.1.1a.orig/debian/megahal-personal
+++ megahal-9.1.1a/debian/megahal-personal
@@ -0,0 +1,27 @@
+#!/bin/sh
+# MegaHal wrapper that makes a ~/.megahal/ directory and populates it with
+# the megahal files. This lets each user on the system have thier own
+# personal megahal.
+
+DATADIR=/usr/lib/megahal/
+
+if [ ! -d "$HOME/.megahal" ]; then
+	mkdir $HOME/.megahal
+fi
+
+cd $HOME/.megahal
+
+
+for file in aux ban grt swp trn; do
+	if [ ! -e "megahal.$file" ]; then
+		cp $DATADIR/megahal.$file .
+	fi
+done
+
+# create the brain if it's the first time we're called
+
+if [ ! -e "megahal.brn" ]; then
+    printf "#quit\n\n\n" | megahal > /dev/null 2>&1
+fi
+
+exec megahal "$@"
--- megahal-9.1.1a.orig/debian/examples
+++ megahal-9.1.1a/debian/examples
@@ -0,0 +1,4 @@
+debian/hal.pl
+debian/Hal.pm
+quick-learn.py
+quick-learn.pl
--- megahal-9.1.1a.orig/debian/compat
+++ megahal-9.1.1a/debian/compat
@@ -0,0 +1 @@
+7
--- megahal-9.1.1a.orig/debian/changelog
+++ megahal-9.1.1a/debian/changelog
@@ -0,0 +1,300 @@
+megahal (9.1.1a-9) unstable; urgency=low
+
+  * QA upload.
+  * Replace build-dep on tcp8.3-dev with tcl-dev. (Closes: #590652)
+    + Change Makefile to use /usr/include/tcl.
+  * Use debhelper compat level 7.
+    + debian/rules: Use dh_prep instead of dh_clean -k.
+  * debian/copyright: Refer to /usr/share/common-licenses/GPL-2.
+  * debian/control: Add ${misc:Depends}.
+  * debian/rules: Remove Megahal/Makefile.old in clean target.
+  * Remove ./dirs: should be debian/dirs which already exists.
+  * Bump Standards-Version to 3.9.1.
+
+ -- Ansgar Burchardt <ansgar@43-1.org>  Thu, 26 Aug 2010 14:15:48 +0900
+
+megahal (9.1.1a-8) unstable; urgency=low
+
+  * QA upload.
+  * Drop Build-dependency on python2.5-dev.
+
+ -- Bernd Zeimetz <bzed@debian.org>  Sat, 22 May 2010 01:06:53 +0200
+
+megahal (9.1.1a-7) unstable; urgency=low
+
+  * QA upload.
+  * Build for all Python versions again (using pyversions -r)
+    (Closes: #582561)
+
+ -- Bernd Zeimetz <bzed@debian.org>  Sat, 22 May 2010 00:17:53 +0200
+
+megahal (9.1.1a-6) unstable; urgency=low
+
+  * QA upload
+  * Changed menu section to "Applications/Science/Social" (closes:  #482933).
+  * debian/megahal-personal: fix bashism (closes: #471876). Thanks to
+    Raphael Geissert for the patch!
+  * Fixed some typos in megahal manpages (closes: #488923). Thanks to
+    A. Costa for the patch!
+  * Standards-Version 3.8.0 (no change)
+  * put DH compatibility level in debian/compat.
+  * Added homepage field to debian/control
+
+ -- Ralf Treinen <treinen@debian.org>  Tue, 17 Feb 2009 21:13:14 +0100
+
+megahal (9.1.1a-5) unstable; urgency=low
+
+  * QA upload.
+  * Fix stack corruption (closes: #454212).
+    Patch by Niko Tyni.
+  * Fix misc. capitalization issues with project names.
+  * Add missing perl dependencies (closes: #463146).
+
+ -- Laurent Fousse <laurent@komite.net>  Wed, 13 Feb 2008 22:39:17 +0100
+
+megahal (9.1.1a-4) unstable; urgency=low
+
+  * Orphaning package.
+
+ -- Laurent Fousse <laurent@komite.net>  Wed, 06 Feb 2008 22:56:49 +0100
+
+megahal (9.1.1a-3) unstable; urgency=low
+
+  * Added explicit build-depends on python2.5 (closes: #416040).
+  * Updated Standards-Versions to 3.7.2 (no change needed).
+
+ -- Laurent Fousse <laurent@komite.net>  Sat, 24 Mar 2007 11:44:58 +0100
+
+megahal (9.1.1a-2) unstable; urgency=low
+
+  * Update python bindings to >= 2.4 (closes: #403971).
+  * Switch to DH_COMPAT=4.
+  * Thanks Bernd Zeimetz who did the work for this
+    revision.
+
+ -- Laurent Fousse <laurent@komite.net>  Fri, 23 Mar 2007 22:13:50 +0100
+
+megahal (9.1.1a-1) unstable; urgency=low
+
+  * Fix misspelling in megahal.1 (closes: #338068).
+  * Repackaged tarball to get rid of editor backup.
+
+ -- Laurent Fousse <laurent@komite.net>  Fri, 23 Dec 2005 12:47:23 +0100
+
+megahal (9.1.1-4) unstable; urgency=low
+
+  * Fixed python "deallocating None" bug, thanks Joris Bontje.
+
+ -- Laurent Fousse <laurent@komite.net>  Fri, 11 Jun 2004 11:30:42 +0200
+
+megahal (9.1.1-3) unstable; urgency=low
+
+  * Change default directory to fix initialization error in python
+    (closes: #250082).
+
+ -- Laurent Fousse <laurent@komite.net>  Thu, 20 May 2004 22:22:46 +0200
+
+megahal (9.1.1-2) unstable; urgency=low
+
+  * Added warning in manpage about newline parsing (closes: #245581).
+
+ -- Laurent Fousse <laurent@komite.net>  Wed,  5 May 2004 21:09:06 +0200
+
+megahal (9.1.1-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Laurent Fousse <laurent@komite.net>  Wed, 25 Feb 2004 22:11:40 +0100
+
+megahal (9.0.3+20030103-2) unstable; urgency=low
+
+  * megahal-personal generates brain at first call, no longer displaying
+    a cryptic warning (closes: #178063).
+  * Updated Standards-Versions to 3.6.1 (no change needed).
+
+ -- Laurent Fousse <laurent@komite.net>  Sat, 31 Jan 2004 17:36:37 +0100
+
+megahal (9.0.3+20030103-1) unstable; urgency=low
+
+  * New maintainer (closes: #164903, #183197).
+  * Packaged latest CVS version (closes: #201904).
+  * Added perl script for quick importation of text files
+    (closes: #117336, #169805).
+  * Updated megahal_interfaces(3) manpage.
+  * Changed menu section to Apps/Science (closes: #201133).
+  * Updated copyright file.
+  * Changed Standards-Versions to 3.6.1.0
+  * Added patch by Matt Behrens for the example bot (closes: #226711).
+  * Cleaned code wrt signedness and tests.
+  * Follow best short description practices and drop the article.
+
+ -- Laurent Fousse <laurent@komite.net>  Thu,  8 Jan 2004 20:29:09 +0100
+
+megahal (9.0.3-7) unstable; urgency=low
+
+  * Orphaning package.  After numerous people have said they were going to
+    take it on, I give up.
+
+ -- David N. Welton <davidw@debian.org>  Sun,  2 Mar 2003 22:11:18 -0800
+
+megahal (9.0.3-6) unstable; urgency=low
+
+  * Added patch from Jose Carlos Garcia Sogo to depend on python1.5-dev
+    (closes: #133711)
+
+ -- David N. Welton <davidw@debian.org>  Sat, 16 Feb 2002 12:29:42 +0100
+
+megahal (9.0.3-5) unstable; urgency=low
+
+  * Added -fPIC to Makefile for HPPA folks.
+  * (closes: #103682)
+
+ -- David N. Welton <davidw@debian.org>  Fri,  6 Jul 2001 12:14:10 +0200
+
+megahal (9.0.3-4) unstable; urgency=low
+
+  * Updated build-depends to tcl8.3-dev.
+  * (closes: #100794)
+
+  * Cannot reproduce this bug anymore.  It was for an older version.
+  * (closes: #100794)
+
+  * Fix -h command line option.
+  * (closes: #100814)
+
+ -- David N. Welton <davidw@debian.org>  Fri, 15 Jun 2001 16:07:32 +0200
+
+megahal (9.0.3-3) unstable; urgency=low
+
+  * Added 'help' command line option.
+
+  * Closes #75775.
+
+ -- David N. Welton <davidw@debian.org>  Sun, 25 Feb 2001 20:43:36 +0100
+
+megahal (9.0.3-2) unstable; urgency=low
+
+  * Updated dependency.
+
+  * (Fixes #76916)
+
+ -- David N. Welton <davidw@debian.org>  Fri, 16 Feb 2001 14:47:33 +0100
+
+megahal (9.0.3-1) unstable; urgency=low
+
+  * Updated maintainer entry.
+
+  * New version with my own changes to megahal.  For now these are not
+    included in the .deb, but will move towards librarifying megahal, and
+    providing C, Tcl and Python API's to it.
+
+ -- David N. Welton <davidw@debian.org>  Tue,  7 Nov 2000 15:16:39 +0100
+
+megahal (8.6-11) unstable; urgency=low
+
+  * Updated Maintainer: entry.
+
+ -- David N. Welton <davidw@linuxcare.com>  Fri,  1 Sep 2000 19:23:25 +0200
+
+megahal (8.6-10) unstable; urgency=low
+
+  * Updated Maintainer: entry
+
+ -- Eric M. Schwartz <emschwar@debian.org>  Mon, 21 Aug 2000 19:41:00 -0600
+
+megahal (8.6-9) unstable; urgency=low
+
+  * Patch from Clint Adams <schizo@debian.org> to give the irc bot multiple
+    channel support. Closes: #64235
+
+ -- Joey Hess <joeyh@debian.org>  Tue, 16 May 2000 15:34:57 -0700
+
+megahal (8.6-8) unstable; urgency=low
+
+  * Build dep.
+
+ -- Joey Hess <joeyh@debian.org>  Sat,  4 Dec 1999 13:29:07 -0800
+
+megahal (8.6-7) unstable; urgency=low
+
+  * FHS
+
+ -- Joey Hess <joeyh@debian.org>  Mon,  6 Sep 1999 14:46:26 -0700
+
+megahal (8.6-6) unstable; urgency=low
+
+  * Added to examples directory, a complete megahal irc bot that connect to
+    a channel, listens and learns form everything said, and responds to users.
+  * Also added a perl module to make it easy for other perl programs
+    to interface to megahal.
+  * Changes by Adam Heath <doogie@debian.org>:
+    - This release has several patches used to make it more wrapper
+      friendly, ie when used inside of a perl wrapper for the purpose
+      of placing it on irc.
+   - Added command word '#QUIET' to toggle automatic responses.
+   - Added --no-prompt(-p) to remove prompt.
+   - Added --no-wrap(-w) to keep it from wrapping it replies.
+   - Added --no-banner(-b) to suppress the banner on startup.
+   - Added a clean target to the megahal makefile.
+  * Updated man page to document new command line switches.
+
+ -- Joey Hess <joeyh@debian.org>  Sat, 15 Dec 1998 00:21:28 -0600
+
+megahal (8.6-5) unstable; urgency=low
+
+  * Tried to fix bug #25729, a problem with non-static initializers.
+
+ -- Joey Hess <joeyh@debian.org>  Sat, 15 Aug 1998 12:32:58 -0700
+
+megahal (8.6-4) unstable; urgency=low
+
+  * Really reupload with full source.
+
+ -- Joey Hess <joeyh@debian.org>  Thu, 18 Jun 1998 11:51:08 -0700
+
+megahal (8.6-3) unstable; urgency=low
+
+  * Re-upload including full source, because the non-maintainer release
+    had a different source tarball somehow.
+
+ -- Joey Hess <joeyh@debian.org>  Fri, 12 Jun 1998 21:25:38 -0700
+
+megahal (8.6-2) unstable; urgency=low
+
+  * Grr. Someone did a non-maintainer release of version 8.6 becuase I was a
+    little slow getting it out. They called it version 8.6-1.1. This is
+    _wrong_. Whoever did that, please use -0.0 in the future, like it says
+    to in the packaging manual.
+  * No changes, just a new version number.
+
+ -- Joey Hess <joeyh@debian.org>  Thu, 28 May 1998 06:20:38 -0400
+
+megahal (8.6-1) unstable; urgency=low
+
+  * New upstream version.
+  * Seems to have integrated all the changes I sent to the author.
+
+ -- Joey Hess <joeyh@debian.org>  Tue, 26 May 1998 00:46:19 -0400
+
+megahal (8.1-3) unstable; urgency=low
+
+  * Fixed minor lintian errors.
+
+ -- Joey Hess <joeyh@debian.org>  Thu, 12 Mar 1998 22:29:32 -0800
+
+megahal (8.1-2) unstable; urgency=low
+
+  * Added a man page.
+  * Corrected fsf mailing address.
+
+ -- Joey Hess <joeyh@debian.org>  Mon,  9 Feb 1998 16:38:04 -0800
+
+megahal (8.1-1) unstable; urgency=low
+
+  * First release.
+  * Fixed 2 bugs that made it crash if it couldn't find a brain or training
+    file. (Forwarded to author.)
+
+ -- Joey Hess <joeyh@debian.org>  Sun, 11 Jan 1998 21:50:52 -0500
+
+
--- megahal-9.1.1a.orig/debian/hal.pl
+++ megahal-9.1.1a/debian/hal.pl
@@ -0,0 +1,162 @@
+#!/usr/bin/perl -w
+#
+# Megahal IRC bot.
+# By Joey Hess <joey@kitenet.net> GPL 1998
+#
+# This needs the debian patched version of megahal, with #quiet command
+# and prompt to stderr output. It also needs Hal.pm and the Net::IRC module.
+#
+# Version: 1.2
+
+# Local configuration.
+my $server=shift || 'irc.debian.org';
+# Multiple channels may be separated by ':' characters.
+my $channel=shift || "#megahal";
+my $nick=shift || "megahal";
+my $ircname=shift || "Megahal";
+my $port=shift || 6667;
+my $howvocal=shift || 10; # percentage of the time it talks publically
+my $save_interval=shift || 60 * 10; # how often to save megahal's brain
+
+my @ignore=('dpkg','apt','||');	# other bots and idiots to ignore
+my $verbose=1;
+
+#
+
+use strict;
+use Net::IRC;
+use Hal;
+
+# Catch ctrl-c properly and stop megahal.
+$SIG{TERM}=sub {
+	Hal::quit();
+	exit;
+};
+$SIG{INT}=$SIG{TERM};
+
+# Set everything up.
+push @ignore,$nick;
+
+my $last_save=time();
+
+my $greet=Hal::start();
+
+my $irc=Net::IRC->new;
+my $conn=$irc->newconn(
+	Nick => $nick,
+	Server => $server,
+	Port => $port,
+	Ircname => $ircname,
+);
+$conn->add_handler('376', \&on_connect); # 376 = end of MOTD: we're connected.
+$conn->add_handler('msg', \&on_msg);
+$conn->add_handler('public', \&on_public);
+$conn->add_handler('kick', \&on_kick);
+
+$irc->start;
+
+# What to do when the bot successfully connects.
+sub on_connect {
+	my $self=shift;
+	my $chan;
+	
+	foreach my $chan (split(/:/, $channel)) {
+		print "Joining $chan..\n" if $verbose;
+		$self->join($chan);
+	
+		# Say hi (or something vaguely along those lines).
+		print "<$nick> $greet\n";
+		$self->privmsg($chan,$greet);
+	}
+	
+}
+
+# Listen to dialog on the channel and send it to megahal for processing.
+sub on_public {
+	my $self=shift;
+	my $event=shift;
+	
+	my $from=$event->nick;
+	my $chan=$event->to;
+	
+	return if grep {$from eq $_} @ignore;
+	
+	$_=join("\n",$event->args)."\n";
+	s/[\x00-\x1f]//g; # no bold crap.
+	
+	return if /^\!.*/; # other bot talk (ie, eggdrop).
+	
+	print "<$from> $_\n" if $verbose;
+	if (/^$nick[:,;](.*)/i) {
+		if ($1) {
+			my $ret="$from: ".Hal::talk($1."\n");
+			# Respond in kind. Sometimes privatly, sometimes
+			# publically. Making it always respond publically
+			# is bad if you want to use the channel it's on
+			# for anything else, people tend to abuse the bot.
+			if (rand() * 100 <= $howvocal) {
+				print "<$nick> $ret";
+				$self->privmsg($chan,$ret);
+			}
+			else {
+				print "($nick) $ret";
+				$self->privmsg($from,$ret);
+			}
+		}
+	}
+	else {
+		# Let's try to strip out nicks and other garbage.
+		s/^.*?[:,;]//;
+		# Send it quietly - no reply is generated, to save CPU.
+		Hal::tell($_) if $_;
+	}
+
+	CheckSave();
+}
+
+
+# Listen to private messages, send to megahal for processing, and respond
+# privatly.
+sub on_msg {
+	my $self=shift;
+	my $event=shift;
+	
+	my $from=$event->nick;
+	
+	return if grep {$from eq $_} @ignore;
+	
+	$_=join("\n",$event->args)."\n";
+	s/[\x00-\x1f]*//g; # no bold crap.
+	
+	print "($from) $_\n";
+	my $ret=Hal::talk($_);
+	print "($nick) $ret";
+	$self->privmsg($from,$ret);
+
+	CheckSave();
+}
+
+# Check to see if it's been long enough since the last save for us to save
+# megahal's brain again.
+sub CheckSave {
+	if ($last_save + $save_interval < time()) {
+		$last_save=time();
+		Hal::save();		
+	}
+}
+
+# Auto-join on kick
+sub on_kick {
+	my $self=shift;
+	my $event=shift;
+	my $from=$event->nick;
+	my @stuff=$event->args;
+	my $chan=$stuff[0];
+	my $reason=$stuff[1];
+
+	$self->join($chan);
+	my $ret="$from: ".Hal::talk($reason."\n");
+	$self->privmsg($chan, $ret);
+}
+
+# All this and I still doen't understand IRC. Scaaaarey! -- Joey
--- megahal-9.1.1a.orig/debian/rules
+++ megahal-9.1.1a/debian/rules
@@ -0,0 +1,91 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper. GNU copyright 1997 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+-include /usr/share/python/python.mk
+
+CFLAGS= -g -W -Wall -Wmissing-prototypes
+
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+        CFLAGS += -O0
+else
+	CFLAGS += -O2
+endif
+		
+
+TCLVERSION=8.3
+PYVERS=$(shell pyversions -r)
+
+build: build-stamp
+build-stamp:
+	dh_testdir	
+	$(MAKE) "C_INCLUDE_PATH=/usr/include/tcl$(TCLVERSION)" "CFLAGS=$(CFLAGS)" 
+	$(MAKE) perlmodule
+	
+	for python in $(PYVERS); do \
+		$$python setup.py build; \
+	done	
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp
+	make clean
+	for python in $(PYVERS); do \
+		$$python setup.py clean; \
+	done
+	dh_clean
+
+# Build architecture-independent files here.
+binary-indep: build
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build
+	dh_testdir
+	dh_testroot
+	dh_prep
+	dh_installdirs 
+	install -s megahal debian/megahal/usr/bin
+	install debian/megahal-personal debian/megahal/usr/bin
+	cp -a megahal.aux megahal.ban megahal.grt megahal.swp megahal.trn \
+		debian/megahal/usr/lib/megahal
+	cp -a libmh_tcl.so debian/megahal/usr/lib/megahal
+# Python module
+	for python in $(PYVERS); do \
+		$$python setup.py install --root=debian/megahal \
+					$(py_setup_install_args) ;\
+	done
+	dh_pysupport
+# Perl module
+	make perlmodule-install DESTDIR=$(CURDIR)/debian/megahal
+	dh_installdocs 
+	dh_installexamples 
+	dh_installmenu
+#	dh_installinit
+	dh_installcron
+	dh_installman docs/megahal.1
+	dh_installman docs/megahal_interfaces.3
+	dh_link /usr/share/man/man1/megahal.1 /usr/share/man/man1/megahal-personal.1
+#	dh_undocumented
+	dh_installchangelogs
+	dh_strip
+	dh_compress
+	dh_fixperms
+	dh_installdeb
+	dh_perl Megahal
+	dh_shlibdeps
+	dh_gencontrol
+#	dh_makeshlibs
+	dh_md5sums
+	dh_builddeb
+
+source diff:                                                                  
+	@echo >&2 'source and diff are obsolete - use dpkg-source -b'; false
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary
--- megahal-9.1.1a.orig/debian/README.debian
+++ megahal-9.1.1a/debian/README.debian
@@ -0,0 +1,36 @@
+Note that MegaHal depends on some data files to give it something to say. It
+can be started up without reading those files, but then it will be rather
+dumb until you teach it a lot.
+
+I've written a wrapper script for MegaHal, that creates a ~/.megahal/
+directory, and populates it with MegaHal data files. If you use this
+wrapper, each you start up MegaHal, it will resume where you left off 
+before. The wrapper is installed as /usr/bin/megahal-personal, since it lets
+you use your own personal MegaHal, which learns from you each time you use
+it.
+
+If you prefer access to the real MegaHal program, which looks for the data
+files in the current directory, run /usr/bin/megahal instead.
+
+It's also possible to set up a MegaHal that is accessed by all users of the
+system. Just make a world-writable directory and copy the MegaHal data files
+into it.
+
+Also, there is an example of an IRC bot that uses megahal in the examples/
+directory.
+
+-- Joey Hess <joeyh@debian.org>
+
+Packaging issues:
+=================
+
+ o The lintian warning package-installs-nonbinary-perl-in-usr-lib-perl5
+   is afaik a lintian bug (#199898).
+
+ o The perl library is not complete but it should be enough for batch
+   learning and simplifying the bot provided by Joey. If you feel like
+   doing the transition, send a wishlist bug with the patch.
+
+ o The python library sometimes segfaults at exit. Help appreciated.
+   
+-- Laurent Fousse <laurent@komite.net>
--- megahal-9.1.1a.orig/debian/source/format
+++ megahal-9.1.1a/debian/source/format
@@ -0,0 +1 @@
+1.0
