From: Paul Gevers <elbrus@debian.org>
Date: Sun, 13 Mar 2022 12:29:11 +0100
X-Dgit-Generated: 2.3.0+ds-2.1 5528a716afc2e62e4fc6377770c623c238e02259
Subject: test: update for latest pygments

Closes: #1006047

---

--- ruby-pygments.rb-2.3.0+ds.orig/test/test_pygments.rb
+++ ruby-pygments.rb-2.3.0+ds/test/test_pygments.rb
@@ -71,7 +71,7 @@ class PygmentsHighlightTest < Test::Unit
 
   def test_highlight_formatter_bbcode
     code = P.highlight(RUBY_CODE, formatter: 'bbcode')
-    assert_match 'color=#408080][i]#!/usr/bin/ruby[/i]', code
+    assert_match 'color=#3D7B7B][i]#!/usr/bin/ruby[/i]', code
   end
 
   def test_highlight_formatter_terminal
@@ -181,7 +181,7 @@ class PygmentsLexerClassTest < Test::Uni
     assert_equal P::Lexer['PHP'], P::Lexer.find_by_extname('.php4')
     assert_equal P::Lexer['PHP'], P::Lexer.find_by_extname('.php5')
     assert_equal P::Lexer['Groff'], P::Lexer.find_by_extname('.1')
-    assert_equal P::Lexer['Groff'], P::Lexer.find_by_extname('.3')
+#    assert_equal P::Lexer['Groff'], P::Lexer.find_by_extname('.3')
     assert_equal P::Lexer['C'], P::Lexer.find_by_extname('.c')
     assert_equal P::Lexer['Python'], P::Lexer.find_by_extname('.py')
     assert_equal P::Lexer['Java'], P::Lexer.find_by_extname('.java')
@@ -213,7 +213,7 @@ class PygmentsCssTest < Test::Unit::Test
   end
 
   def test_css_default
-    assert_match '.c { color: #408080; font-style: italic }', P.css
+    assert_match '.c { color: #3D7B7B; font-style: italic }', P.css
   end
 
   def test_css_colorful
