--- Makefile.old	2013-09-16 01:04:56.000000000 -0400
+++ Makefile	2013-09-16 01:09:03.000000000 -0400
@@ -34,15 +34,15 @@
 # For UNIX-like systems
 
 SYSTEM =	UNIX
-SUBSYSTEM =	SOLARIS
+SUBSYSTEM =	LINUX
 
 # Locations
-DIR =		/home/dick
-BINDIR =	$(DIR)/bin.`$(DIR)/bin/arch`
-MAN1DIR =	$(DIR)/man/man1
+DIR =		$(DESTDIR)/usr
+BINDIR =	$(DIR)/bin
+MAN1DIR =	$(DIR)/share/man/man1
 
 # Commands
-COPY =		cp -p
+COPY =		install -D
 EXE =		#
 LEX =		flex
 LN =		ln
@@ -51,24 +51,24 @@
 ################################################################
 # For MSDOS + MinGW
 
-SYSTEM =	MSDOS
-SUBSYSTEM =	MinGW
+#SYSTEM =	MSDOS
+#SUBSYSTEM =	MinGW
 
 # Locations
-DIR =		C:/BIN
-BINDIR =	C:/BIN
-MAN1DIR =	C:/BIN
+#DIR =		C:/BIN
+#BINDIR =	C:/BIN
+#MAN1DIR =	C:/BIN
 
 # Commands (cp required, since xcopy cannot handle forward slashes)
-COPY =		cp -p
-EXE =		.exe
-LEX =		flex
-LN =		ln
-ZIP =		zip -o
+#COPY =		cp -p
+#EXE =		.exe
+#LEX =		flex
+#LN =		ln
+#ZIP =		zip -o
 
 ################################################################
 # General, compiling:
-CC =		gcc -D$(SYSTEM) -D$(SUBSYSTEM)
+CC =		$(USER_CC) -D$(SYSTEM) -D$(SUBSYSTEM)
 LINT =		lint -ansi -D$(SYSTEM) -D$(SUBSYSTEM)
 LINTFLAGS =	-xh
 
@@ -89,15 +89,15 @@
 
 # Compiling
 MEMORY =	-DMEMLEAK -DMEMCLOBBER
-CFLAGS =	-DVERSION=$(VERSION) $(MEMORY) -O4
+CFLAGS =	-DVERSION=$(VERSION) $(MEMORY) $(USER_CFLAGS)
 LIBFLAGS =	#
 LINTFLAGS =	$(MEMORY) -h# -X
-LOADFLAGS =	-s#			# strip symbol table
+LOADFLAGS =	$(LDFLAGS) #			# strip symbol table
 LOADER =	$(CC) $(LOADFLAGS)
 
 # Debugging
-CFLAGS +=	-DDEBUG
-DEBUG_C =	debug.c
-DEBUG_O =	debug.o
-DEBUG_H =	debug.h
+#CFLAGS +=	-DDEBUG
+#DEBUG_C =	debug.c
+#DEBUG_O =	debug.o
+#DEBUG_H =	debug.h
 
 #	T E S T   P A R A M E T E R S
 
