--- a/Makefile	2013-09-16 01:04:56.000000000 -0400
+++ b/Makefile	2013-09-16 01:09:03.000000000 -0400
@@ -34,15 +34,15 @@
 # For UNIX|Linux
 
 SYSTEM =	UNIX
-SUBSYSTEM =	SOLARIS
+SUBSYSTEM =	LINUX
 
 # Locations
-DIR =		/home/dick
-BINDIR =	$(DIR)/bin.`$(DIR)/bin/arch`
-MAN1DIR =	$(DIR)/man/man1
+DIR =		$(DESTDIR)/usr
+BINDIR =	$(DIR)/bin
+MAN1DIR =	$(DIR)/share/man/man1
 
 # Commands
-COPY =		cp -p
+COPY =		install -D
 EXE =		#
 LEX =		flex
 LN =		ln
@@ -51,27 +51,27 @@
 ################################################################
 # For MSDOS + MinGW
 
-SYSTEM =	MSDOS
-SUBSYSTEM =	MinGW
+#SYSTEM =	MSDOS
+#SUBSYSTEM =	MinGW
 
 # Locations
-DIR =		C:/BIN
-BINDIR =	C:/BIN
-MAN1DIR =	C:/BIN
+#DIR =		C:/BIN
+#BINDIR =	C:/BIN
+#MAN1DIR =	C:/BIN
 
 # File names
-NULLFILE =	nul
+#NULLFILE =	nul
 
 # Commands (cp required, since xcopy cannot handle forward slashes)
-COPY =		cp -p
-EXE =		.exe
-LEX =		flex
-LN =		ln
-ZIP =		zip -o
+#COPY =		cp -p
+#EXE =		.exe
+#LEX =		flex
+#LN =		ln
+#ZIP =		zip -o
 
 ################################################################
 # General, C compilation:
-CC =		gcc -D$(SYSTEM) -D$(SUBSYSTEM)
+CC =		$(USER_CC) -D$(SYSTEM) -D$(SUBSYSTEM)
 LINT =		lint -ansi -D$(SYSTEM) -D$(SUBSYSTEM)
 LINTFLAGS =	-xh
 
@@ -84,14 +84,14 @@
 
 # Compiling
 MEMORY =	-DMEMCHECK -DMEMCLOBBER
-CFLAGS =	$(VERSION) $(MEMORY) -O4 # -Dlint -DLIB # for all db active
+CFLAGS =	$(VERSION) $(MEMORY) $(USER_CFLAGS) # -Dlint -DLIB # for all db active
 LIBFLAGS =	#
 LINTFLAGS =	-Dlint_test $(MEMORY) -h# -X
-LOADFLAGS =	-s#			# strip symbol table
+LOADFLAGS =	$(LDFLAGS)
 LOADER =	$(CC) $(LOADFLAGS)
 
 # Debugging
-CFLAGS +=	-DDEBUG
+#CFLAGS +=	-DDEBUG
 
 #	T E S T   P A R A M E T E R S
 
