diff -up kaya-0.5.2/compiler/CodegenCPP.hs.ghc7
kaya-0.5.2/compiler/CodegenCPP.hs
--- kaya-0.5.2/compiler/CodegenCPP.hs.ghc7      2010-12-01 17:13:57.989909984 +0100
+++ kaya-0.5.2/compiler/CodegenCPP.hs   2010-12-01 17:15:26.590910022 +0100
@@ -1,3 +1,4 @@
+
 {-
     Kaya - My favourite toy language.
     Copyright (C) 2004, 2005 Edwin Brady
@@ -386,10 +387,10 @@ mkfcall n lib ty args res
       (conv ty) ++ "("++n++"("++stackconv 0 args++")))"
  where
     popvals n 0 = ""
-    popvals n (a+1) = 
+    popvals n a =
 		    -- "HEAPVAL("++show (tmpval n)++"); " ++ 
                     "POPARG(" ++ show (tmpval n) ++ "); " ++
-                    popvals (n+1) a
+                    popvals (n+1) (a-1)
 
 {- CIM 12/7/05 changed to KVOID to avoid clashes with MinGW -}
     conv (Prim Void) = "KVOID("

