diff -ru stdlib.old//Makefile.in stdlib/Makefile.in
--- stdlib.old//Makefile.in	2011-09-30 14:38:13.000000000 -0400
+++ stdlib/Makefile.in	2011-09-30 14:38:58.000000000 -0400
@@ -87,7 +87,7 @@
 Compress.o: Binary.ki zlib_glue.h Strings.ki Regex.ki
 Reflect.o: Builtins.ki Prelude.ki
 Pickle.o: Prelude.ki Binary.ki Reflect.ki Strings.ki Parse.ki
-Plugins.o: dl.h
+Plugins.o: dl.h Array.ki Regex.ki
 KayaDoc.o: ElementTree.ki Strings.ki HTMLDocument.ki Time.ki Regex.ki
 ElementTreeData.o: Dict.ki
 ElementTree.o: XMLentities.ki ElementTreeData.ki Regex.ki
