diff -ru neoleo-7.0.0.old/tests/inbuilt-tests.sh neoleo-7.0.0/tests/inbuilt-tests.sh
--- neoleo-7.0.0.old/tests/inbuilt-tests.sh	2018-10-10 08:40:07.931308775 -0400
+++ neoleo-7.0.0/tests/inbuilt-tests.sh	2018-10-10 08:41:35.588212913 -0400
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
 
-neoleo -T
+NEO=$ABUILDDIR/../src/neoleo
+$NEO -T
 exit $?
diff -ru neoleo-7.0.0.old/tests/ref.sh neoleo-7.0.0/tests/ref.sh
--- neoleo-7.0.0.old/tests/ref.sh	2018-10-10 08:40:07.934308737 -0400
+++ neoleo-7.0.0/tests/ref.sh	2018-10-10 08:41:37.486189185 -0400
@@ -5,7 +5,9 @@
 TMPFILE=`mktemp`
 SS=$ASRCDIR/ref.oleo
 echo "SS=$SS"
-neoleo $SS -H <<< "save-spreadsheet $TMPFILE"
+NEO=$ABUILDDIR/../src/neoleo
+echo "NEO=$NEO"
+$NEO $SS -H <<< "save-spreadsheet $TMPFILE"
 set_productions $SS
 mv $TMPFILE $OUTFILE
 diff $OUTFILE $VERFILE
