diff -ru a/Puma/tools/lemon/lemon.c b/Puma/tools/lemon/lemon.c
--- a/Puma/tools/lemon/lemon.c	2010-12-09 03:18:27.000000000 -0500
+++ b/Puma/tools/lemon/lemon.c	2010-12-09 03:19:49.000000000 -0500
@@ -32,13 +32,8 @@
 #include <stdarg.h>
 #include <string.h>
 #include <ctype.h>
-
-extern void qsort();
-extern double strtod();
-extern long strtol();
-extern void free();
-extern int access();
-extern int atoi();
+#include <stdlib.h>
+#include <unistd.h>
 
 /* #define PRIVATE static */
 #define PRIVATE
