diff -ru a/src/thaw/plugins/mDns/MDNSDiscovery.java b/src/thaw/plugins/mDns/MDNSDiscovery.java
--- a/src/thaw/plugins/mDns/MDNSDiscovery.java	2017-03-20 12:44:07.207645427 -0400
+++ b/src/thaw/plugins/mDns/MDNSDiscovery.java	2017-03-20 12:45:31.191648115 -0400
@@ -34,7 +34,7 @@
 		try {
 			// Spawn the mdns listener
 			Logger.info(this, "Starting JMDNS ...");
-			this.jmdns = new JmDNS();
+			this.jmdns = JmDNS.create();
 
 			// Start listening for new nodes
 			jmdns.addServiceListener(MDNSDiscoveryPanel.FCP_SERVICE_TYPE, new FCPMDNSListener());
