--- starplot-0.95.5.orig/debian/watch
+++ starplot-0.95.5/debian/watch
@@ -0,0 +1,2 @@
+version=3
+http://www.starplot.org/download.html downloads/starplot-(.*)\.tar\.gz
--- starplot-0.95.5.orig/debian/postinst
+++ starplot-0.95.5/debian/postinst
@@ -0,0 +1,24 @@
+#!/bin/sh
+set -e
+
+
+case "$1" in
+    configure)
+	if [ -x /usr/sbin/register-stardata ] ; then
+            register-stardata install all starplot
+	fi
+    ;;
+    
+    abort-upgrade|abort-remove|abort-deconfigure)
+
+    ;;
+
+    *)
+        echo "postinst called with unknown argument \`$1'" >&2
+        exit 1
+    ;;
+esac
+
+#DEBHELPER#
+exit 0
+
--- starplot-0.95.5.orig/debian/README.source
+++ starplot-0.95.5/debian/README.source
@@ -0,0 +1,26 @@
+This package uses dpatch to manage all modifications to the upstream
+source. Changes are stored in the source package as diffs in
+debian/patches and applied during the build.
+
+To get the fully patched source after unpacking the source package, cd
+to the root level of the source package and run:
+
+    debian/rules patch
+
+Removing a patch is as simple as removing its entry from the
+debian/patches/00list file, and please also remove the patch file
+itself.
+
+Creating a new patch is done with "dpatch-edit-patch patch XX_patchname"
+where you should replace XX with a new number and patchname with a
+descriptive shortname of the patch. You can then simply edit all the
+files your patch wants to edit, and then simply "exit 0" from the shell
+to actually create the patch file.
+
+To tweak an already existing patch, call "dpatch-edit-patch XX_patchname"
+and replace XX_patchname with the actual filename from debian/patches
+you want to use.
+
+To clean up afterwards again, "debian/rules unpatch" will do the
+work for you - or you can of course choose to call
+"fakeroot debian/rules clean" all together.
--- starplot-0.95.5.orig/debian/changelog
+++ starplot-0.95.5/debian/changelog
@@ -0,0 +1,349 @@
+starplot (0.95.5-4) unstable; urgency=low
+
+  * Updated to Debian policy 3.8.3
+  * Fixed the problem with FTBFS thanks for the patch to Kamal Mostafa
+    <kamal@whence.com> (Closes: #560544).
+  * Added debian/README.source file explaining how dpatch works, according to 
+    Debian Policy.
+  * Added ${misc:Depends} depends to bynary package, required by debhelper.
+
+ -- Francisco Manuel Garcia Claramonte <francisco@debian.org>  Thu, 07 Jan 2010 22:24:30 +0100
+
+starplot (0.95.5-3) unstable; urgency=low
+
+  * Updated Debian policy to 8.3.2
+  * Updated Maintainer control field to @debian.org.
+  * Updated debian/rules file:
+      - Changed dh_clean -k to dh_prep.
+      - Removed deprecated dh_desktop call.
+      - Updated debhelper dependency version from 5 to 7.
+      - Updated debhelper compatibility to 7.
+
+ -- Francisco Manuel Garcia Claramonte <francisco@debian.org>  Sun, 26 Jul 2009 19:41:36 +0200
+
+starplot (0.95.5-2) unstable; urgency=low
+
+  * Updated Standard-version control field to 3.8.0.1
+  * Added a dpatch to update the starplot.desktop file provided by Kevin B.
+    McCarty <kmccarty@gmail.com> with the non-localized comment field. 
+    Thanks to "Nathan Handler" <nhandler@ubuntu.com>. (Closes: #491448).
+  * The new upstream manual is enough complete and documented.
+    (Closes: #344622).
+  * Removed the debian/starplot32x32.xpm file, It was added to the
+    upstream version. Clean up debian/rules to remove installing icon
+    command.
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Mon, 01 Dec 2008 13:33:22 +0100
+
+starplot (0.95.5-1) unstable; urgency=low
+
+  * New upstream release.
+  * Added debian/watch file.
+  * Modified debian/rules to configure and install properly html docs and 
+    examples directory.
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Sat, 17 May 2008 11:52:11 +0200
+
+starplot (0.95.4-5) unstable; urgency=low
+
+  * Edited starplot data file generation script "starplot.sh" to do nothing
+    if a pre-existing data file in starplot format is already newer than the
+    times of installation of the starplot package and relevant data package.
+  * Otherwise cleaned up starplot.sh a bit.
+  * Adjusted CFLAGS definition in debian/rules to prevent FTBFS and ensured
+    that g++ used them (by setting CXXFLAGS to the same string).
+  * Removed extraneous ';' following namespace and extern "C" blocks in
+    lib/compat.h, src/classes/mathdefs.h, src/classes/strings.h,
+    src/gui/starplot.h, src/gui/hrdiagram.cc, src/gui/infodialogs.cc
+    and src/gui/starplot.cc.  Needed to prevent FTBFS with -pedantic flag.
+  * Updated doc-base section to new policy: Apps/Science -> Science/Astronomy.
+  * Changed substitution tables in gliese and yale spec files to be
+    case-sensitive in order to enhance starconvert speed.
+
+ -- Kevin B. McCarty <kmccarty@debian.org>  Fri, 11 Apr 2008 14:08:17 -0700
+
+starplot (0.95.4-4) unstable; urgency=low
+
+  * Added "#include <cstdlib>" in src/classes/strings.cc, and omitted the
+    std:: prefix from vsprintf(), vsnprintf(), in hopes of fixing the FTBFS
+    on arm, mips, mipsel.
+
+ -- Kevin B. McCarty <kmccarty@debian.org>  Thu, 28 Feb 2008 15:14:22 -0800
+
+starplot (0.95.4-3) unstable; urgency=low
+
+  * Updated to debian policy 3.7.3.
+  * Added debian/menu icon field with starplot32x32.xpm file.
+  * Added starplot.desktop file, thanks to Marco Rodrigues <gothicx@sapo.pt>
+    (Closes: #457836)
+  * Cleaned debian/rules file.
+  * Added Homepage control field.
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Wed, 23 Jan 2008 00:07:56 +0100
+
+starplot (0.95.4-2) unstable; urgency=low
+
+  * Patched source code to fix compile error with GCC 4.3.
+    (Closes: #454869)
+  * Modified debian/rules clean process to caught error messages from 
+    the clean or distclean rules.
+  * Changed debian/menu Section field to Applications/Science/Astronomy
+    according to the Debian menu sub-policy.
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Sat, 15 Dec 2007 18:33:58 +0100
+
+starplot (0.95.4-1) unstable; urgency=low
+
+  * New upstream release
+  * Updated to debian policy 3.7.2.2
+  * Created debian/compat file and removed DH_COMPAT field from
+    debian/rules.
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Thu, 03 May 2007 00:07:53 +0200
+
+starplot (0.95.3-2) unstable; urgency=low
+
+  * Fixed an error with POSIX sh string equality "=" in 
+    /usr/lib/stardata-common/starplot file. (Closes: #342331)
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Wed,  7 Dec 2005 14:10:24 +0100
+
+starplot (0.95.3-1) unstable; urgency=low
+
+  [ Francisco Garcia ]
+  * New upstream release
+  * Fixed the compilation failure on 64-bit architectures. 
+    Thanks to Andreas Jochens <aj@andaco.de> (Closes: #298418)
+  * Removed the hack to generate es.gmo in debian/rules file.
+
+  [ Kevin McCarty ]
+  * Update to debhelper version 5.
+  * Add postrm removing /var/lib/starplot directory.
+  * Update gliese.spec and yale.spec files to most recent versions.
+  * Add NEWS file about the new stardata-common package.
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Thu, 22 Sep 2005 22:01:12 +0200
+
+starplot (0.95.2-3) unstable; urgency=low
+
+  * Added a script in usr/lib/stardata-common called starplot to install
+    and remove starplot own data format catalogues, conform to the 
+    stardata-common policy.
+  * Modified postinst and prerm scripts, now both of these just call to
+    register-stardata program to register (install or remove) stardata
+    catalogues in starplot data format.
+  * Added Recommends relationship with stardata-common package.
+  * Added provides control field with gliese-viewer and yale-viewer
+    virtual packages, suggested by Kevin B. McCarty <kmccarty@Princeton.EDU>
+  * Added gliese.spec and yale.spec files to /usr/share/starplot/specfiles to
+    convert those catalogues to starplot data format. Suggested by 
+    Kevin B. McCarty <kmccarty@Princeton.EDU>.
+  * Updated to Debian Policy 3.6.2.1
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Mon, 30 May 2005 18:21:55 +0200
+
+starplot (0.95.2-2) unstable; urgency=low
+
+  * Couldn't resist and reviewed the spanish translation...
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Sun, 20 Mar 2005 01:21:24 +0100
+
+starplot (0.95.2-1) unstable; urgency=low
+  
+  * New upstream release
+  * debian/rules now configure properly '--with-webbrowser' parameter.
+  * Removed the article "A" in short description, in Description field
+    of debian/control, recommended by Developers Reference.
+
+ -- Francisco Garcia <franciscomanuel.garcia@hispalinux.es>  Mon, 24 Jan 2005 21:44:20 +0100
+
+starplot (0.95-3) unstable; urgency=low
+
+  * Small postinst change to properly remove symlinks (now for real)
+  * Also remove es.gmo on 'clean' to avoid FTFBS the second time around
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Wed,  8 Sep 2004 09:07:50 +0200
+
+starplot (0.95-2) unstable; urgency=low
+
+  * Postinst will now remove the file that is used as symlink regardless
+     of it's a regular file or not (to avoid breaking on upgrades)
+  * Introduce a small hack in debian/rules to generate es.gmo so that 
+    the Spanish translation is installed properly. It should be removed
+    when 0.96 is uploaded (this was a consequence of modifying the 
+    es_ES definitions in the previous upload). That should teach me _not_
+    to play with po files without asking upstream first.
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Mon,  6 Sep 2004 23:46:45 +0200
+
+starplot (0.95-1) unstable; urgency=low
+
+  * New upstream release
+  * New maintainer, I'm setting myself as co-maintainer (Closes: #216876)
+    Kevin B. McCarty <kmccarty_AT_Princeton.EDU> will probably co-maintain
+    this package in the future too.
+  * Fixed Francisco's README.Debian for clarity
+  * Generated stardata files are removed in a safer way when uninstalled this
+    includes removing files under /usr/share/starplot (for the case when
+    using old gliese or yale package)
+  * Create symlinks to generated data files in /usr/share/starplot so the
+    user does not have to navigate through the whole root tree to see
+    the generated stardata files. Remove those symlinks on removal.
+  * Remove config.status after distcleaning in debian/rules or the clean
+    process will not work properly.
+  * Quick revision of the spanish translation (needs to be revised further
+    and merged upstream). Also, use this translation as the generic 
+    spanish one instead of limiting it to es_ES and es_ES@euro users.
+  [ changes by Francisco Garcia <franciscomanuel.garcia@hispalinux.es> ]
+  * Switched dh_installmanpages by dh_installman in rules file
+  * Changed the starplot data sets location, from /usr/share/starplot
+    to /var/lib/starplot
+  * Updated to Debian Policy 3.6.1
+  * Fixed the lintian warnings in menu file
+  * Upgraded the version of debhelper to 3.0.0 in Build-Depends 
+    source control field
+  * Added rm config.status to clean rule in rules file
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Fri,  3 Sep 2004 22:53:15 +0200
+
+starplot (0.94.1-2) unstable; urgency=low
+
+  * New upstream version included into Debian (Closes: #199420)
+  * Incorporated upstream's changes to the Debian package including
+    the section change from Apps/Graphics to Apps/Science
+  * Modified doc/Makefile.in to properly install to DESTDIR in 
+    the install-data-local target
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Sun, 24 Aug 2003 01:24:13 +0200
+
+starplot (0.94.1-1) unstable; urgency=low
+ 
+  * New upstream release: only change is fix for Mac OS X.
+ 
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Mon, 30 Jun 2003 16:32:58 -0400
+ 
+starplot (0.94-1) unstable; urgency=low
+ 
+  * New upstream release
+  * Update Policy to 3.5.10
+ 
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Wed, 25 Jun 2003 19:48:38 -0400
+
+starplot (0.93-3) unstable; urgency=low
+
+  * Modified starkpg to not look for orig-data directories since the
+    information of star catalogues is placed in /usr/share/stardata/
+    with a subdirectory per catalogue and _no_ orig-data subdirectory.
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Thu, 29 May 2003 13:18:51 +0200
+
+starplot (0.93-2) unstable; urgency=low
+
+  * Adopted by Debian maintainer, submitted to Debian including
+    some of the changes required by policy (for example, do not
+    mess with /usr/local on postinst) (Closes: #169603)
+  * Added /usr/share/starplot as the standard directory for starpkg
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Wed, 28 May 2003 21:26:52 +0200
+
+starplot (0.93-1) unstable; urgency=low
+
+  * New upstream release
+  * No longer is required by .deb packages of data files, but they will
+    be regenerated by postinst scripts any time starplot or the data file
+    package is upgraded.
+  * Add support for DEB_BUILD_OPTIONS
+
+ -- Kevin McCarty <kmccarty@arcturus.lngs.infn.it>  Sat,  9 Nov 2002 13:04:49 +0100
+
+
+starplot (0.92.3-3) unstable; urgency=low
+
+  * Fixed typo (Closes: #125383)
+  * Fixed control fields (now suggests gliese and yale)
+  * Removed /usr/local/ twinkling in postinst/postrm
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Wed, 19 Dec 2001 18:33:28 +0100
+
+starplot (0.92.3-2) unstable; urgency=low
+
+  * Sent by Debian maintainer.
+  * Moved to science section.
+  * Added Debian changelog to upstream maintainer's
+  * Maintainer new version fixes/rewords control file
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Fri,  7 Dec 2001 10:05:40 +0100
+
+starplot (0.92.3-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Thu, 6 Dec 2001 10:08:25 -0500
+
+starplot (0.92.2-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Sun, 23 Sep 2001 01:29:59 -0400
+ 
+starplot (0.92pre-3) unstable; urgency=low
+
+  * Fixed typo in description.
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Mon, 26 Nov 2001 17:42:38 +0100
+
+starplot (0.92pre-2) unstable; urgency=low
+
+  * Adopted and uploaded to Debian.
+
+ -- Javier Fernandez-Sanguino Pen~a <jfs@computer.org>  Tue, 13 Mar 2001 01:25:21 +0100
+
+
+starplot (0.92.2-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Sun, 23 Sep 2001 01:24:20 -0400
+
+starplot (0.92-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Wed, 23 May 2001 23:00:00 -0400
+
+starplot (0.91.2-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Wed, 31 Jan 2001 21:16:40 -0500
+
+starplot (0.91.1-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Tue, 30 Jan 2001 20:45:36 -0500
+
+starplot (0.91-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Sun, 28 Jan 2001 12:43:00 -0500
+
+starplot (0.90-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Sun, 14 Jan 2001 20:03:20 -0500
+
+starplot (0.86-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Thu, 11 Jan 2001 00:51:23 -0500
+
+starplot (0.85.2-1) unstable; urgency=low
+
+  * Initial Release.
+  * My first Debian package - wheee!
+
+ -- Kevin B. McCarty <kmccarty@princeton.edu>  Tue,  9 Jan 2001 04:09:40 -0500
--- starplot-0.95.5.orig/debian/doc-base
+++ starplot-0.95.5/debian/doc-base
@@ -0,0 +1,10 @@
+Document: starplot
+Title: Debian StarPlot Manual
+Author: Kevin B. McCarty
+Abstract: Documentation for the StarPlot 3-dimensional star map viewer.
+Section: Science/Astronomy
+
+Format: HTML
+Index: /usr/share/doc/starplot/html/index.html
+Files: /usr/share/doc/starplot/html/*.html
+
--- starplot-0.95.5.orig/debian/copyright
+++ starplot-0.95.5/debian/copyright
@@ -0,0 +1,13 @@
+This package was debianized by Kevin B. McCarty <kmccarty@princeton.edu> on
+Tue,  9 Jan 2001 04:09:40 -0500.
+
+It was downloaded from http://www.starplot.org.
+(alternative site: http://www.princeton.edu/~kmccarty/starplot.html)
+
+Upstream Author: Kevin B. McCarty <kmccarty@princeton.edu>
+This software is copyright (c) 2000-2002 by Kevin B. McCarty.
+
+You are free to distribute this software under the terms of
+the GNU General Public License.
+On Debian systems, the complete text of the GNU General Public
+License can be found in /usr/share/common-licenses/GPL file.
--- starplot-0.95.5.orig/debian/rules
+++ starplot-0.95.5/debian/rules
@@ -0,0 +1,93 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+include /usr/share/dpatch/dpatch.make
+
+
+# DEB_BUILD_OPTIONS support
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+	CFLAGS=-Wall -pedantic -O0 -g
+else
+	CFLAGS=-Wall -pedantic -O2 -g
+endif
+# "nostrip" is seen automatically by dh_strip
+
+# configure options as per autotools-dev usr/share/doc/README.Debian
+export DEB_HOST_GNU_TYPE  ?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
+export DEB_BUILD_GNU_TYPE ?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+ 
+# FOR AUTOCONF 2.52 AND NEWER ONLY
+
+ifeq ($(DEB_BUILD_GNU_TYPE), $(DEB_HOST_GNU_TYPE))
+  confflags += --build $(DEB_HOST_GNU_TYPE)
+else
+  confflags += --build $(DEB_BUILD_GNU_TYPE) --host $(DEB_HOST_GNU_TYPE)
+endif
+
+
+# begin Make targets ----------------------------------------------------------
+                                                                                
+configure: configure-stamp
+configure-stamp:
+	dh_testdir
+	# Add here commands to configure the package.
+	./configure $(confflags) --prefix=/usr --with-webbrowser=sensible-browser \
+		--htmldir=/usr/share/doc/starplot
+	touch configure-stamp
+
+build: configure-stamp build-stamp
+build-stamp: patch-stamp
+	dh_testdir
+
+	# Add here commands to compile the package.
+	$(MAKE) all CFLAGS="$(CFLAGS)" CXXFLAGS="$(CFLAGS)" \
+		INSTALLPGM=$(INSTALLPGM) \
+		docdir=/usr/share/doc/starplot mandir=/usr/share/man sysconfdir=/etc
+	touch build-stamp
+
+clean: unpatch
+	dh_testdir
+	dh_testroot
+	# Add here commands to clean up after the build process.
+	[ ! -f Makefile ] || $(MAKE) distclean
+	rm -f build-stamp configure-stamp config.status config.log
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_prep
+	dh_installdirs
+
+	# Add here commands to install the package into debian/starplot.
+	$(MAKE) install DESTDIR=$(CURDIR)/debian/starplot/ \
+		docdir=/usr/share/doc mandir=/usr/share/man sysconfdir=/etc
+	install -D $(CURDIR)/debian/starplot.sh $(CURDIR)/debian/starplot/usr/lib/stardata-common/starplot
+	mkdir -p $(CURDIR)/debian/starplot/usr/share/starplot/specfiles
+	install -m 0664 $(CURDIR)/debian/*.spec $(CURDIR)/debian/starplot/usr/share/starplot/specfiles
+	mv $(CURDIR)/debian/starplot/usr/share/doc/examples $(CURDIR)/debian/starplot/usr/share/doc/starplot/examples 
+	install $(CURDIR)/debian/README.source $(CURDIR)/debian/starplot/usr/share/doc/starplot/
+
+
+binary-indep: build install
+
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_installdocs
+	dh_installmenu
+	dh_installman
+	dh_installchangelogs ChangeLog
+	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install configure
--- starplot-0.95.5.orig/debian/gliese.spec
+++ starplot-0.95.5/debian/gliese.spec
@@ -0,0 +1,302 @@
+# starconvert specification file for Gliese data catalog
+# version 3-0.95
+
+comments; 189; 257
+
+[coordinates]
+type; celestial
+
+ra-hours; 13; 14
+ra-min;   16; 17
+ra-sec;   19; 20
+
+dec-sign; 22
+dec-deg;  23; 24
+dec-min;  26; 29
+
+[characteristics]
+specclass; 55; 66
+
+distance;  milliarcsec; 109; 114
+magnitude; absolute;    122; 126
+
+[systems]
+component;  9; 10
+separation; comments; a = ; Sep; sep
+
+[names]
+bayer;          comments # e.g. "Alpha Cen", "Tau Cet"
+flamsteed;      comments # e.g. "40 Eri"
+const-specific; comments # includes, e.g., "q Car", "VV Ori"
+other;          comments; Ross; Wolf
+
+other;          147; 152; HD # Henry Draper catalog number
+dm;             154; 165     # Durchmusterung number BD / CD / CP
+other;          comments; L = Luyten # i.e. substitute "Luyten" for "L"
+
+other;          167; 175 # for Gliese catalog, this field is the Giclas number
+other;          177; 181; LHS # for Gliese catalog, this field is LHS number
+other;          183; 187 # for Gliese catalog, includes some other designations
+other;          1; 8     # for Gliese catalog, includes Gliese and Woolley des.
+
+# A whole bunch of other designations appear in the Gliese catalog "Remarks"
+#  field, out of which I have no idea which ones are "preferred" by
+#  astronomers; hence, they're alphabetical.  Rearrange as you like.
+
+other;          comments; ADS; BPM; BS; CFS; DON; FK; GD; GJ; GR; GSC
+other;          comments; Hei; Hy; IE; LDS; LE; LFT; LP; LTT; MW; NSV
+other;          comments; PG; PS; RGO; Rob; RST; SA; San; Sm; Steph
+other;          comments; USN; USNO; VA; VB; VVO; Wo; Wor; WOR
+
+[substitutions]
+case-sensitive
+
+# Well-known common names of stars in the Gliese catalog:
+Alpha CMa;         Sirius
+Alpha Boo;         Arcturus
+Alpha Lyr;         Vega
+Alpha Aur;         Capella
+Alpha CMi;         Procyon
+Alpha Aql;         Altair
+Alpha Tau;         Aldebaren
+Beta Gem;          Pollux
+Alpha PsA;         Fomalhaut
+Alpha Gem;         Castor
+
+# Gliese et al missed a lot of Bayer designations, so I've filled them in here.
+#  There are probably others which I have missed.
+
+HD 739;            Theta Scl
+HD 3003;           Beta(3) Tuc
+HD 3302;           Lambda(2) Phe
+HD 11171;          Chi Cet
+HD 17094;          Mu Cet   
+HD 18907;          Epsilon For
+HD 33111;          Beta Eri
+HD 39091;          Pi Men
+HD 40183;          Beta Aur
+HD 56537;          Lambda Gem
+HD 71878;          Beta Vol
+HD 79096;          Pi(1) Cnc
+HD 81997;          Tau(1) Hya
+HD 95418;          Beta UMa
+HD 96202;          Chi(1) Hya
+HD 101198;         Iota Crt
+HD 116656;         Zeta UMa # Mizar; see below
+HD 118098;         Zeta Vir
+HD 124850;         Iota Vir
+HD 125161;         Iota Boo
+HD 129502;         Mu Vir
+HD 129972;         Omicron Boo
+HD 130819;         Alpha(1) Lib
+HD 130841;         Alpha(2) Lib
+HD 134505;         Zeta Lup
+HD 137108;         Eta CrB
+HD 137391;         Mu(1) Boo
+HD 137392;         Mu(2) Boo
+HD 138905;         Gamma Lib
+HD 141795;         Epsilon Ser
+HD 142908;         Lambda CrB
+HD 144070;         Xi Sco
+HD 144197;         Delta Nor
+HD 146362;         Sigma CrB
+HD 146686;         Gamma(2) Nor
+HD 146791;         Epsilon Oph
+HD 147449;         Sigma Ser
+HD 152786;         Zeta Ara
+HD 153580;         Epsilon(2) Ara
+HD 154906;         Mu Dra
+HD 156164;         Delta Her
+HD 160922;         Omega Dra
+HD 162004;         Psi(1) Dra
+HD 164764;         Tau Oph
+HD 175190;         Nu(2) Sgr
+HD 177241;         Omicron Sgr
+HD 177716;         Tau Sgr
+HD 177724;         Zeta Aql
+HD 181577;         Rho(1) Sgr
+HD 184406;         Mu Aql
+HD 191862;         Xi(2) Cap
+HD 197157;         Eta Ind
+HD 201184;         Chi Cap
+HD 202730;         Theta Ind
+HD 210418;         Theta Peg
+HD 211336;         Epsilon Cep
+HD 212061;         Gamma Aqr
+HD 215665;         Lambda Peg
+HD 215789;         Epsilon Gru
+HD 216131;         Mu Peg
+HD 216336;         Gamma PsA
+HD 216385;         Sigma Peg
+HD 216435;         Tau(1) Gru
+HD 219571;         Gamma Tuc
+HD 219693;         Phi Gru
+HD 219784;         Gamma Scl
+HD 222107;         Lambda And
+HD 222661;         Omega(2) Aqr
+
+# Here are a large number of HD -> Flamsteed conversions.
+#  (Gliese et al included very few Flamsteed numbers in their catalog.)
+HD 693;            6 Cet
+HD 1835;           9 Cet
+HD 3196;           13 Cet
+HD 3651;           54 Psc
+HD 4307;           18 Cet
+HD 4676;           64 Psc
+HD 7439;           37 Cet
+HD 10476;          107 Psc
+HD 13612;          66 Cet
+HD 15814;          29 Ari
+HD 16141;          79 Cet
+HD 16739;          12 Per
+HD 18803;          51 Ari
+HD 19994;          94 Cet
+HD 22484;          10 Tau
+HD 25680;          39 Tau
+HD 25998;          50 Per
+HD 29503;          53 Eri
+HD 30495;          58 Eri
+HD 32537;          9 Aur
+HD 32923;          104 Tau
+HD 33021;          13 Ori
+HD 33256;          68 Eri
+HD 35296;          11 Tau
+HD 37507;          49 Ori
+HD 43042;          71 Ori
+HD 43386;          74 Ori
+HD 50635;          38 Gem
+HD 50692;          37 Gem
+HD 58855;          22 Lyn
+HD 64096;          9 Pup
+HD 68146;          18 Pup
+HD 70958;          1 Hya
+HD 78209;          15 UMa
+HD 78418;          75 Cnc
+HD 79028;          16 UMa
+HD 79439;          18 UMa
+HD 81937;          23 UMa
+HD 82210;          24 UMa
+HD 86146;          19 LMi
+HD 86728;          20 LMi
+HD 87696;          21 LMi
+HD 89125;          39 Leo
+HD 89449;          40 Leo
+HD 90839;          36 UMa
+HD 95128;          47 UMa
+HD 99491;          83 Leo
+HD 100563;         89 Leo
+HD 101501;         61 UMa
+HD 106251;         12 Vir
+HD 110897;         10 CVn
+HD 111998;         38 Vir
+HD 115202;         57 Vir
+HD 115383;         59 Vir
+HD 115617;         61 Vir
+HD 116842;         80 UMa # Alcor; see below
+HD 116976;         69 Vir
+HD 117176;         70 Vir
+HD 119756;         1 Cen
+HD 123999;         12 Boo
+HD 124206;         50 Hya
+HD 125451;         18 Boo
+HD 132052;         16 Lib
+HD 134083;         45 Boo
+HD 142267;         39 Ser
+HD 143333;         49 Lib
+HD 146233;         18 Sco
+HD 149661;         12 Oph
+HD 153597;         19 Dra
+HD 155886;         36 Oph
+HD 157214;         72 Her
+HD 157792;         44 Oph
+HD 160269;         26 Dra
+HD 160915;         58 Oph
+HD 165341;         70 Oph
+HD 165777;         72 Oph
+HD 165908;         99 Her
+HD 168151;         36 Dra
+HD 173667;         110 Her
+HD 174116;         29 Sgr
+HD 180777;         59 Dra
+HD 182572;         31 Aql
+HD 186408;         16 Cyg
+HD 187013;         17 Cyg
+HD 190406;         15 Sge
+HD 198809;         31 Vul
+HD 207978;         15 Peg
+HD 212698;         53 Aqr
+HD 212754;         34 Peg
+HD 217014;         51 Peg
+HD 219080;         7 And
+HD 219834;         94 Aqr
+HD 224930;         85 Peg
+
+# You can substitute more than once for the same star, but the substitutions
+#  have to be in the right order.
+Zeta UMa;          Mizar # follows "HD 116656" -> "Zeta UMa" substitution above
+80 UMa;            Alcor # follows "HD 116842" -> "80 UMa" substitution above
+
+# A few less well-known common names of stars.
+# These should be inserted after their Bayer designations in priority.
+# This only works with starconvert >= 0.95:
+Alpha Cen;         Rigil Kent;	2
+Beta Leo;          Denebola;	2
+Alpha Oph;         Ras Alhague;	2
+Beta UMa;          Merak;	2
+Delta UMa;         Megrez;	2
+Epsilon UMa;       Alioth;	2
+
+# Well-known nearby dwarf stars
+G001-027;          van Maanen's Star
+HD 33793;          Kapteyn's Star
+HD 95735;          Lalande 21185
+BD +04∞3561a;      Barnard's Star
+
+# Companion stars
+HD 10360;          P Eri B
+HD 60178;          Castor B
+HD 98230;          Xi UMa B
+HD 110380;         Gamma Vir B
+HD 114379;         Alpha Com B
+HD 116657;         Mizar B
+HD 128621;         Alpha Cen B
+HD 144069;         Xi Sco B
+HD 154905;         Mu Dra B
+HD 155885;         36 Oph B
+HD 177475;         Gamma CrA B
+HD 201092;         61 Cyg B
+
+Gl 23;             13 Cet B
+Gl 105.4;          Epsilon Cet B
+Gl 106.1;          Gamma Cet B
+Gl 171.1;          Aldebaren B
+Gl 188;            104 Tau B
+Gl 194;            Capella B
+Gl 271;            Delta Gem B
+Gl 291;            9 Pup B
+Gl 337;            Pi(1) Cnc B
+Gl 351;            Psi Vel B
+Gl 354;            Theta UMa B
+Gl 366.1;          Theta Ant B
+Gl 387;            39 Leo B
+Gl 527;            Tau Boo B
+Gl 550.2;          Phi Vir B
+Gl 560;            Alpha Cir B
+Gl 566;            Xi Boo B
+Gl 575;            I Boo B
+Gl 635;            Zeta Her B
+Gl 656.1;          Eta Oph B
+Gl 695;            Mu Her B
+Gl 816.2;          Eta Cap B
+Gl 822;            Delta Equ B
+
+Wo 9220;           38 Gem B
+W049;              Sirius B
+W053;              Procyon B
+LHS 122;           Eta Cas B
+LHS 459;           70 Oph B
+LHS 3325;          Mu Her B
+
+BD -02∞2902;       Tau(1) Hya B
+BD -03∞335;        66 Cet B
+
--- starplot-0.95.5.orig/debian/menu
+++ starplot-0.95.5/debian/menu
@@ -0,0 +1,2 @@
+?package(starplot):needs="X11" section="Applications/Science/Astronomy"\
+  icon="starplot32x32.xpm" title="StarPlot" command="/usr/bin/starplot"
--- starplot-0.95.5.orig/debian/control
+++ starplot-0.95.5/debian/control
@@ -0,0 +1,26 @@
+Source: starplot
+Section: science
+Priority: optional
+Maintainer: Francisco Manuel Garcia Claramonte <francisco@debian.org>
+Uploaders: Javier Fernandez-Sanguino Pen~a <jfs@computer.org>, Kevin B. McCarty <kmccarty@debian.org>
+Build-Depends: debhelper (>> 7.0.0), libgtk2.0-dev, dpatch
+Standards-Version: 3.8.3
+Homepage: http://starplot.org
+
+Package: starplot
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Suggests: gliese, yale
+Recommends: stardata-common
+Provides: gliese-viewer, yale-viewer
+Description: 3-dimensional perspective star map viewer
+ StarPlot is a GTK+ based program that can be used interactively to view
+ three-dimensional perspective charts of stars.  Charts can be re-centered,
+ rotated, or zoomed in or out with a mouse click (this can also, of course,
+ be done via dialog boxes for more precision).  Stars may be viewed (or
+ ignored) by spectral class and absolute magnitude.
+ .
+ StarPlot is packaged with starconvert, a utility that converts
+ line-oriented stellar data records to StarPlot format.  Most star data
+ files available on the Internet can be converted this way if a short file
+ describing the original file format is provided to starconvert.
--- starplot-0.95.5.orig/debian/prerm
+++ starplot-0.95.5/debian/prerm
@@ -0,0 +1,22 @@
+#!/bin/sh
+set -e
+
+#DEBHELPER#
+
+
+case "$1" in
+    remove|upgrade|deconfigure)
+    if [ -x /usr/sbin/register-stardata ] ; then
+       register-stardata remove all starplot
+    fi
+    ;;
+
+    failed-upgrade)
+        ;;
+    *)
+        echo "prerm called with unknown argument \`$1'" >&2
+        exit 1
+    ;;
+esac
+
+exit 0
\ No newline at end of file
--- starplot-0.95.5.orig/debian/yale.spec
+++ starplot-0.95.5/debian/yale.spec
@@ -0,0 +1,89 @@
+# starconvert specification file for Yale bright stars catalog
+# version 5-0.95
+
+# no comments field for Yale catalog; if I'm feeling ambitious, I might
+#  eventually add the ability to read comments from the accompanying notes.dat
+#  file.
+
+[coordinates]
+type; celestial
+
+ra-hours; 76; 77
+ra-min;   78; 79
+ra-sec;   80; 83
+
+dec-sign; 84
+dec-deg;  85; 86
+dec-min;  87; 88
+dec-sec;  89; 90
+
+[characteristics]
+specclass;           128; 147
+
+distance; arcsec;    162; 166; 0.04 # don't trust parallaxes < 0.04"
+magnitude; visual;   103; 107
+
+distance; specclass
+magnitude; visual;   103; 107
+
+[systems]
+# haven't yet figured out how to extract multiple star data from Yale catalog
+
+[names]
+bayer;          8; 14 
+flamsteed;      5; 14
+const-specific; 52; 60 # "const-specific" includes, e.g., "q Car", "VV Ori"
+other;          26; 31; HD  # Henry Draper catalog number
+other;          32; 37; SAO # Smithsonian Astrophysical Observatory catalog no.
+dm;             15; 25      # Durchmusterung number BD / CD / CP
+other;          45; 51; ADS # American Double Star catalog number
+# other;        5; 14
+
+[substitutions]
+case-sensitive
+
+Alpha CMa;         Sirius
+Alpha Car;         Canopus
+Alpha Boo;         Arcturus
+Alpha Lyr;         Vega
+Alpha Aur;         Capella
+Beta Ori;          Rigel
+Alpha CMi;         Procyon
+Alpha Ori;         Betelgeuse
+Alpha Eri;         Achernar
+Alpha Aql;         Altair
+Alpha Tau;         Aldebaren
+Alpha Vir;         Spica
+Alpha Sco;         Antares
+Beta Gem;          Pollux
+Alpha PsA;         Fomalhaut
+Alpha Cyg;         Deneb
+Alpha Leo;         Regulus
+Alpha Gem;         Castor
+Omicron Cet;       Mira
+Alpha UMi;         Polaris
+Beta Per;          Algol
+Zeta UMa;          Mizar
+80 UMa;            Alcor
+
+# A few less well-known common names of stars.
+# These should be inserted after their Bayer designations in priority.
+# This only works with starconvert >= 0.95:
+Alpha And;         Alpheratz;	2
+Alpha(1) Cen;      Rigil Kent;	2
+Beta Cen;          Hadar;	2
+Alpha(1) Cru;      Acrux;	2
+Beta Cru;          Mimosa;	2
+Beta(1) Cyg;       Albireo;	2
+Alpha Dra;         Thuban;	2
+Alpha(1) Her;      Ras Algethi;	2
+Beta Leo;          Denebola;	2
+Alpha Oph;         Ras Alhague;	2
+Gamma Ori;         Bellatrix;	2
+Eta Tau;           Alcyone;	2
+Alpha UMa;         Dubhe;	2
+Beta UMa;          Merak;	2
+Gamma UMa;         Phecda;	2
+Delta UMa;         Megrez;	2
+Epsilon UMa;       Alioth;	2
+Eta UMa;           Alkaid;	2
--- starplot-0.95.5.orig/debian/README.Debian
+++ starplot-0.95.5/debian/README.Debian
@@ -0,0 +1,18 @@
+StarPlot for Debian
+-------------------
+
+This package, from version 0.95.2-3, is according to the stardata-common 
+policy. Thus, starplot package provides a script to install and remove
+stardata catalogues, converting these catalogues to its own data format.
+The register-stardata program calls this script for creation and
+deletion of those star data sets.
+
+Read more about stardata-common in 
+http://alioth.debian.org/projects/stardata-common
+
+The converted stardata catalogues are stored in /var/lib/starplot, with a 
+symlink in /usr/share/starplot. 
+All the contents of that data directory will be removed when
+the package is uninstalled. 
+
+-- Francisco Garcia  <franciscomanuel.garcia@hispalinux.es>, Sun, 7 November 2004 23:41:20 +0200
--- starplot-0.95.5.orig/debian/starplot.sh
+++ starplot-0.95.5/debian/starplot.sh
@@ -0,0 +1,82 @@
+#!/bin/sh
+set -e
+
+test -x /usr/bin/starplot || exit 0
+
+if [ -z "$2" ] ; then
+    exit 0
+fi
+ 
+
+###################################
+## starplot install|remove stardata
+##
+###################################
+
+case $1 in
+    install)
+     # update any starplot data files
+     SP_DIR=/var/lib/starplot
+     STD_DIR=/usr/share/stardata
+     SPEC_DIR=/usr/share/starplot/specfiles
+          
+     if [ "$2" = "gliese" ] || [ "$2" = "yale" ] ; then
+	 IN=$STD_DIR/$2/catalog.dat
+	 SPEC=$SPEC_DIR/$2.spec
+	 OUT=$SP_DIR/$2.stars
+	 CONV=/usr/bin/starconvert
+
+         [ ! -d "$SP_DIR" ] && mkdir $SP_DIR || true
+	 if [ ! -x $CONV ] ; then
+	     echo "Error: Unexpectedly missing starconvert binary!" 1>&2
+	     exit 1
+         elif [ ! -e $IN ] || [ ! -e $SPEC ] ; then
+	     echo "Warning: Unexpectedly missing $2 catalog or spec file!" 1>&2
+	     exit 0
+	 fi
+
+	 # Test that output file is newer than package installation time
+	 # of starplot and star catalogue packages.  If not, needs to be
+	 # updated.  Have to use the files under /var/lib/dpkg/info since
+	 # the mtimes for normal files in installed packages are the time of
+	 # creation of the package rather than the time of installation.
+	 UPDATE=true
+	 if [ -f $OUT ] && [ -f /var/lib/dpkg/info/starplot.list ] \
+		        && [ -f /var/lib/dpkg/info/$2.list       ] ; then
+	     UPDATE=false
+	     [ $OUT -nt /var/lib/dpkg/info/starplot.list ] || UPDATE=true
+	     [ $OUT -nt /var/lib/dpkg/info/$2.list       ] || UPDATE=true
+	 fi
+
+	 if [ "$UPDATE" = true ] ; then
+             echo "Updating $2 starplot data sets (in $SP_DIR)..."
+	     $CONV $SPEC $IN $OUT > /dev/null 2>&1
+	 else
+	     echo "Starplot data sets for $2 are already up to date."
+	 fi
+	 
+	 if [ -f "$OUT" ] ; then
+	     linkfile=/usr/share/starplot/$2.stars
+	     if [ -f "$linkfile" ] || [ -L "$linkfile" ] ; then
+	        rm -f $linkfile
+	     fi
+	     ln -s $OUT $linkfile
+	 fi    
+     fi
+    ;;
+
+    remove)
+    if [ "$2" = "gliese" ] || [ "$2" = "yale" ] ; then
+	echo "Removing StarPlot $2 catalogue."
+	find /var/lib/starplot/ -name "$2.stars" -type f -print0 | xargs -0 rm -f
+	find /usr/share/starplot/ -name "$2.stars" -type l -print0 | xargs -0 rm -f
+	find /usr/share/starplot/ -name "$2.stars" -type f -print0 | xargs -0 rm -f
+    fi
+    ;;
+
+    *)
+       exit 0
+    ;;
+esac
+
+exit 0
--- starplot-0.95.5.orig/debian/compat
+++ starplot-0.95.5/debian/compat
@@ -0,0 +1 @@
+7
--- starplot-0.95.5.orig/debian/NEWS
+++ starplot-0.95.5/debian/NEWS
@@ -0,0 +1,9 @@
+starplot (0.95.3-1) unstable; urgency=low
+
+  As of this version of StarPlot, data files are controlled via the
+  register-stardata script in the stardata-common package.  If you wish
+  to have the Gliese and/or Yale star data files converted to StarPlot
+  format automatically, you should install stardata-common as well as
+  the gliese and/or yale package.
+
+ -- Kevin B. McCarty <kmccarty@debian.org>  Mon,  5 Dec 2005 09:11:01 -0500
--- starplot-0.95.5.orig/debian/postrm
+++ starplot-0.95.5/debian/postrm
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+set -e
+
+case "$1" in
+	remove|disappear)
+		rmdir /var/lib/starplot > /dev/null 2>&1 || true
+	;;
+	
+	purge|upgrade|failed-upgrade|abort-install|abort-upgrade)
+
+        ;;
+
+	*)
+        	echo "postrm called with unknown argument \`$1'" >&2
+        	exit 1
+	;;
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+exit 0
--- starplot-0.95.5.orig/debian/patches/00list
+++ starplot-0.95.5/debian/patches/00list
@@ -0,0 +1,2 @@
+01-starplot_desktop_file.dpatch
+02-fix-ftbfs-and-hrdiagram-opts.dpatch
--- starplot-0.95.5.orig/debian/patches/02-fix-ftbfs-and-hrdiagram-opts.dpatch
+++ starplot-0.95.5/debian/patches/02-fix-ftbfs-and-hrdiagram-opts.dpatch
@@ -0,0 +1,101 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 02-fix-ftbfs-and-hrdiagram-opts.dpatch by Kamal Mostafa <kamal@whence.com>
+##
+## DP: Fixes:
+## DP:  - FTBFS - must call set_*_item() accessors, not set_item() template
+## DP:  - Make HR Diagram window Options menu correctly reflect the state of
+## DP:    the main window opts for Labels and Diameters on HR window startup.
+##
+## License: This patch is licensed under terms of GPL2, or (at your option)
+## any later version.
+
+@DPATCH@
+=== modified file 'src/gui/hrdiagram.cc'
+--- a/src/gui/hrdiagram.cc	2008-05-17 11:52:11 +0000
++++ b/src/gui/hrdiagram.cc	2010-01-07 18:17:30 +0000
+@@ -29,7 +29,7 @@
+ /* Functions to make sure the HR Diagram window stays updated */
+ 
+ namespace globals {
+-  static GtkWidget  *program_hr_window = NULL;
++  GtkWidget         *program_hr_window = NULL;
+   static GdkPixmap  *program_hr_pixmap = NULL;
+ }
+ 
+@@ -88,6 +88,7 @@
+   delete globals::program_hr_viewer;
+   globals::program_hr_viewer = NULL;
+   gtk_widget_destroy(globals::program_hr_window);
++  globals::program_hr_window = NULL;
+   return;
+ }
+ 
+@@ -254,8 +255,7 @@
+     hr_options_label_widgets[i] = item;
+     
+     if ((GTK_CHECK_MENU_ITEM (options_label_widgets[i]))->active)
+-      set_item(&globals::chart_rules.StarLabels,
+-	       globals::chart_rules.StarLabels, item, false);
++      set_label_item(globals::chart_rules.StarLabels, false);
+   }
+ 
+   for (unsigned int i = 0; i < OPTIONS_DIAMETER_MENU_SIZE; i++) {
+@@ -265,8 +265,7 @@
+     hr_options_diameter_widgets[i] = item;
+ 
+     if ((GTK_CHECK_MENU_ITEM (options_diameter_widgets[i]))->active)
+-      set_item(&globals::chart_rules.StarDiameters,
+-	       globals::chart_rules.StarDiameters, item, false);
++      set_diameter_item(globals::chart_rules.StarDiameters, false);
+   }
+ 
+   if (menubar) {
+
+=== modified file 'src/gui/menuops.cc'
+--- a/src/gui/menuops.cc	2008-05-17 11:52:11 +0000
++++ b/src/gui/menuops.cc	2010-01-07 18:08:32 +0000
+@@ -303,12 +303,12 @@
+ 
+ void set_label_item(star_label_t l, bool redraw)
+ {
+-  if (globals::chart_rules.StarLabels == l) return;
+-
+-  if (globals::program_hr_viewer) {
++  if (globals::program_hr_window) {
+     set_item(&globals::chart_rules.StarLabels, l,
+ 	     hr_options_label_widgets[l], false);
+   }
++
++  if (globals::chart_rules.StarLabels == l) return;
+   set_item(&globals::chart_rules.StarLabels, l,
+ 	   options_label_widgets[l], redraw);
+ }
+@@ -320,12 +320,12 @@
+ 
+ void set_diameter_item(star_diameter_t d, bool redraw)
+ {
+-  if (globals::chart_rules.StarDiameters == d) return;
+-
+-  if (globals::program_hr_viewer) {
++  if (globals::program_hr_window) {
+     set_item(&globals::chart_rules.StarDiameters, d,
+ 	     hr_options_diameter_widgets[d], false);
+   }
++
++  if (globals::chart_rules.StarDiameters == d) return;
+   set_item(&globals::chart_rules.StarDiameters, d,
+ 	   options_diameter_widgets[d], redraw);
+ }
+
+=== modified file 'src/gui/starplot.h'
+--- a/src/gui/starplot.h	2008-05-17 11:52:11 +0000
++++ b/src/gui/starplot.h	2010-01-07 18:01:36 +0000
+@@ -134,6 +134,7 @@
+   extern GTKViewer    *program_viewer;      // The GTKViewer used by StarPlot
+   extern GtkWidget    *program_canvas;      // The GTK_DRAWING_AREA used
+   extern GdkPixmap    *program_pixmap;	    // The backing pixmap
++  extern GtkWidget    *program_hr_window;   // HR Diagram GTK_WINDOW
+   extern GTKViewer    *program_hr_viewer;   // HR Diagram GTKViewer
+   extern GtkWidget    *program_hr_canvas;   // HR Diagram GTK_DRAWING_AREA  
+   extern GdkPixbuf    *program_icon;	    // used by the window manager
+
--- starplot-0.95.5.orig/debian/patches/01-starplot_desktop_file.dpatch
+++ starplot-0.95.5/debian/patches/01-starplot_desktop_file.dpatch
@@ -0,0 +1,22 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 01-starplot_desktop_file.dpatch by  Francisco Manuel Garc√≠a Claramonte
+## <franciscomanuel.garcia@hispalinux.es>
+##
+## DP: This patch updates the starplot.desktop file, with the with the 
+## non-localized comment field.
+##
+## License: This patch is licensed under terms of GPL2, or (at your option)
+## any later version.
+
+@DPATCH@
+diff -urNad starplot-0.95.5~/starplot.desktop starplot-0.95.5/starplot.desktop
+--- starplot-0.95.5~/starplot.desktop	2008-03-05 00:05:42.000000000 +0100
++++ starplot-0.95.5/starplot.desktop	2008-12-03 13:22:02.000000000 +0100
+@@ -2,6 +2,7 @@
+ Version=1.0
+ Name=StarPlot Star Chart Viewer
+ GenericName=Star Chart Viewer
++Comment=View detailed star charts in 3D
+ Comment[en]=View detailed star charts in 3D
+ Comment[es]=Vista detallada de gr√°ficos de estrellas en 3D
+ Type=Application
