--- acinclude.m4.orig	2008-08-31
+++ acinclude.m4	2008-08-31
@@ -229,18 +218,7 @@
 
   dnl ---- Optimization flags
   
-  if test "x$gambas_optimization" = "xyes"; then
     AM_CFLAGS_OPT="$AM_CFLAGS -O3"
-    AM_CFLAGS="$AM_CFLAGS -Os"
-    AM_CXXFLAGS="$AM_CXXFLAGS -Os -fno-omit-frame-pointer"
-  else
-    AM_CFLAGS_OPT="$AM_CFLAGS -O0"
-    AM_CFLAGS="$AM_CFLAGS -O0"
-    AM_CXXFLAGS="$AM_CXXFLAGS -O0"
-  fi
-
-  CFLAGS=""
-  CXXFLAGS=""
   
   AC_SUBST(AM_CFLAGS)
   AC_SUBST(AM_CFLAGS_OPT)
