--- Makefile.in.old	2015-07-02 21:48:38.654124911 -0400
+++ Makefile.in	2015-07-02 21:48:51.608125325 -0400
@@ -242,7 +242,7 @@
 
 malshow: $(MALSHOW_OBJS) libmalaga.la
 	@echo "Linking malshow"
-	@$(LIBTOOL) --mode=link $(LINK) $(MALSHOW_OBJS) $(GTK_LIBS) \
+	@$(LIBTOOL) --mode=link $(LINK) $(MALSHOW_OBJS) $(GTK_LIBS) -lm \
 	  libmalaga.la -o malshow 
 
 # malsym ----------------------------------------------------------------------
