--- qorganizer-3.1.orig/debian/README.Debian
+++ qorganizer-3.1/debian/README.Debian
@@ -0,0 +1,8 @@
+qorganizer for Debian
+---------------------
+
+Original qOrganizer compiles translation files (.qm ones) directly into main
+binary. However, Debian build doesn't and translation files are placed into
+qorganizer-translations package because of size reasons.
+
+ -- Eugene V. Lyubimkin <jackyf.devel@gmail.com>  Thu, 04 Sep 2008 22:14:28 +0300
--- qorganizer-3.1.orig/debian/changelog
+++ qorganizer-3.1/debian/changelog
@@ -0,0 +1,30 @@
+qorganizer (3.1-2) unstable; urgency=low
+
+  * debian/patches:
+    - New 010-fix-compilation-with-qt4.5.patch, fixes FTBFS with Qt 4.5.
+      Patch by Gennadi Motsyo. (Closes: #527668)
+  * debian/control:
+    - Added '${misc:Depends}' to Depends to satisfy lintian.
+    - Bumped 'Standards-Version' to 3.8.1, no changes needed.
+
+ -- Eugene V. Lyubimkin <jackyf.devel@gmail.com>  Fri, 08 May 2009 23:42:43 +0300
+
+qorganizer (3.1-1) unstable; urgency=low
+
+  * New upstream release:
+    - Added Polish translation.
+  * debian/control:
+    - Small fixes in short and long descriptions.
+    - Priority changed to 'optional' from 'extra'.
+  * debian/patches:
+    - 100-use-system-path-for-translations.patch: refreshed.
+    - 200-fix-64bit.patch: went upstream, deleted.
+
+ -- Eugene V. Lyubimkin <jackyf.devel@gmail.com>  Sat, 08 Nov 2008 09:39:59 +0200
+
+qorganizer (3.1.3-1) unstable; urgency=low
+
+  * Initial release. (Closes: #498046)
+
+ -- Eugene V. Lyubimkin <jackyf.devel@gmail.com>  Thu, 04 Sep 2008 22:14:28 +0300
+
--- qorganizer-3.1.orig/debian/compat
+++ qorganizer-3.1/debian/compat
@@ -0,0 +1 @@
+7
--- qorganizer-3.1.orig/debian/control
+++ qorganizer-3.1/debian/control
@@ -0,0 +1,45 @@
+Source: qorganizer
+Section: x11
+Priority: optional
+Maintainer: Eugene V. Lyubimkin <jackyf.devel@gmail.com>
+Build-Depends: debhelper (>= 7), quilt (>= 0.40), libqt4-dev (>= 4.2.0), imagemagick
+Standards-Version: 3.8.1
+Homepage: http://qorganizer.sourceforge.net
+
+Package: qorganizer
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Recommends: qorganizer-translations (>= ${binary:Version})
+Description: featured lightweight graphical organizer
+ qOrganizer is a general organizer that includes a calendar with
+ schedule, reminders, journal/notes for every day, to-do list. 
+ .
+ It also has couple of additional features:
+  - interface designed to be clear and simple
+  - automatic saving of all data
+  - loud reminders
+  - a rich text journal for everyday in which you can even drag and drop images
+  - searching through every entry in the journal and in the schedules
+  - a clear and simple to-do list (general one, not for every day)
+  - a flexible timetable for students which even supports different content for
+    odd and even weeks (this is used by universities)
+  - a student booklet in which the user can keep track of the marks he gets in
+    school, it even calculates averages and stores the date when the mark was
+    given; it also can contain absences
+  - an ability to print any page (the calendar, the to-do list, the timetable
+    and the booklet are separate pages)
+  - the ability to choose the storing mode: text files, SQLite database or
+    MySQL database over the network
+  - the ability to upload the stored data to an FTP server and download it
+  - ability to save between storing modes (so the risk of data loss is even
+    lower)
+
+Package: qorganizer-translations
+Architecture: all
+Depends: ${misc:Depends}
+Recommends: qorganizer
+Description: featured lightweight graphical organizer (translations)
+ qOrganizer is a general organizer that includes a calendar with
+ schedule, reminders, journal/notes for every day, to-do list. 
+ .
+ This package contains translations for qorganizer.
--- qorganizer-3.1.orig/debian/copyright
+++ qorganizer-3.1/debian/copyright
@@ -0,0 +1,32 @@
+This package was debianized by Eugene V. Lyubimkin <jackyf.devel@gmail.com> on
+Thu, 04 Sep 2008 22:14:28 +0300.
+
+It was downloaded from
+http://sourceforge.net/project/showfiles.php?group_id=199558 .
+
+Upstream Author:
+
+    Balázs Béla <balazsbela@gmail.com>
+
+Copyright:
+
+    Copyright (C) 2007, Balázs Béla
+
+License:
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License version 2 for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the
+    Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,  
+    MA 02110-1301, USA                                                    
+
+The Debian packaging is (C) 2008, Eugene V. Lyubimkin <jackyf.devel@gmail.com> and
+is licensed under the GPLv3, see `/usr/share/common-licenses/GPL-3'.
--- qorganizer-3.1.orig/debian/menu
+++ qorganizer-3.1/debian/menu
@@ -0,0 +1,2 @@
+?package(qorganizer):needs="X11" section="Applications/Office"\
+  title="qorganizer" command="/usr/bin/qorganizer" icon="/usr/share/qorganizer/menuicon.xpm"
--- qorganizer-3.1.orig/debian/qorganizer-translation.dirs
+++ qorganizer-3.1/debian/qorganizer-translation.dirs
@@ -0,0 +1 @@
+/usr/share/qorganizer/translations
--- qorganizer-3.1.orig/debian/qorganizer-translations.install
+++ qorganizer-3.1/debian/qorganizer-translations.install
@@ -0,0 +1 @@
+src/lang/*.qm usr/share/qorganizer/translations
--- qorganizer-3.1.orig/debian/qorganizer.1
+++ qorganizer-3.1/debian/qorganizer.1
@@ -0,0 +1,35 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH QORGANIZER 1 "September  4, 2008"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+qorganizer \- program to organize your life
+.SH SYNOPSIS
+.B qorganizer
+.SH DESCRIPTION
+This manual page documents briefly the
+.B qorganizer
+command.
+.PP
+.\" TeX users may be more comfortable with the \fB<whatever>\fP and
+.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
+.\" respectively.
+\fBqorganizer\fP is a program that contains calendar, to-do list, booklet.
+.SH AUTHOR
+qorganizer was written by Balázs Béla.
+.PP
+This manual page was written by Eugene V. Lyubimkin <jackyf.devel@gmail.com>,
+for the Debian project (but may be used by others).
--- qorganizer-3.1.orig/debian/qorganizer.dirs
+++ qorganizer-3.1/debian/qorganizer.dirs
@@ -0,0 +1 @@
+usr/bin
--- qorganizer-3.1.orig/debian/qorganizer.install
+++ qorganizer-3.1/debian/qorganizer.install
@@ -0,0 +1,2 @@
+src/qorganizer usr/bin
+menuicon.xpm /usr/share/qorganizer
--- qorganizer-3.1.orig/debian/qorganizer.manpages
+++ qorganizer-3.1/debian/qorganizer.manpages
@@ -0,0 +1 @@
+debian/qorganizer.1
--- qorganizer-3.1.orig/debian/rules
+++ qorganizer-3.1/debian/rules
@@ -0,0 +1,48 @@
+#!/usr/bin/make -f
+# -*- makefile -*-
+
+#export DH_VERBOSE=1
+include /usr/share/quilt/quilt.make
+
+%:
+	dh $@
+
+DEBDIR=$(CURDIR)/debian
+
+clean: unpatch
+	rm -f src/*.o src/qorganizer \
+		src/qrc_application.cpp \
+		src/moc* src/Makefile \
+		menuicon.xpm
+	dh clean
+
+
+menuicon.xpm: qOrganizer.png
+	convert -resize 32x32 $< $@
+
+build: build-stamp
+build-stamp: patch menuicon.xpm
+	dh build --before dh_auto_build
+	cd src && qmake-qt4 && \
+		sed -i 's/$$(CXXFLAGS)/$$(CXXFLAGS) $$(DEBFLAGS)/g' Makefile \
+		&& $(MAKE) DEBFLAGS="$(CFLAGS)"
+	dh build --after dh_auto_build
+	touch build-stamp
+
+install:
+	# do nothing, binary-arch do all stuff
+
+binary-indep:
+	dh binary-indep --before dh_auto_install
+	dh binary-indep --after dh_auto_install
+
+binary-arch: build
+	dh binary-arch --before dh_auto_install
+	dh binary-arch --after dh_auto_install --before dh_installchangelogs
+	dh_installchangelogs CHANGELOG.txt
+	dh binary-arch --after dh_installchangelogs
+
+binary-arch: install
+
+binary: binary-indep binary-arch
+.PHONY: build binary-indep binary-arch binary
--- qorganizer-3.1.orig/debian/watch
+++ qorganizer-3.1/debian/watch
@@ -0,0 +1,2 @@
+version=3
+http://sf.net/qorganizer/qOrganizer-v(\d\.\d)-(\d)\.tar\.gz
--- qorganizer-3.1.orig/debian/patches/100-use-system-path-for-translations.patch
+++ qorganizer-3.1/debian/patches/100-use-system-path-for-translations.patch
@@ -0,0 +1,50 @@
+Index: qorganizer-3.1/src/main.cpp
+===================================================================
+--- qorganizer-3.1.orig/src/main.cpp	2008-11-10 21:06:50.862870490 +0200
++++ qorganizer-3.1/src/main.cpp	2008-12-07 10:20:59.580343661 +0200
+@@ -58,7 +58,7 @@
+       QTranslator *translator = new QTranslator;
+       app.installTranslator(translator);  
+       readSettings(); //Read the language
+-      translator->load(C_LANGUAGE,":/lang"); //set it
++      translator->load(C_LANGUAGE,"/usr/share/qorganizer/translations"); //set it
+       QSplashScreen *splash = new QSplashScreen;
+       splash -> setPixmap(QPixmap(":/images/splash.png"));
+       splash -> show();
+Index: qorganizer-3.1/src/application.qrc
+===================================================================
+--- qorganizer-3.1.orig/src/application.qrc	2008-11-10 21:06:50.842870532 +0200
++++ qorganizer-3.1/src/application.qrc	2008-12-07 10:20:59.580343661 +0200
+@@ -14,19 +14,6 @@
+       <file>images/exit.png</file>
+       <file>images/upload.png</file>
+       <file>images/download.png</file>
+-      <file>lang/Hungarian.qm</file>
+-      <file>lang/Romanian.qm</file>
+-      <file>lang/Slovenian.qm</file>
+-      <file>lang/Portuguese.qm</file>
+-      <file>lang/Russian.qm</file>
+-      <file>lang/Estonian.qm</file>
+-      <file>lang/Spanish.qm</file>
+-      <file>lang/Albanian.qm</file>
+-      <file>lang/Macedonian.qm</file>
+-      <file>lang/Dutch.qm</file>
+-      <file>lang/German.qm</file>
+-      <file>lang/French.qm</file>
+-      <file>lang/Polish.qm</file>
+       <file>images/sound.wav</file>
+       <file>images/clear.png</file>
+       <file>images/upload.png</file>
+Index: qorganizer-3.1/src/settings.cpp
+===================================================================
+--- qorganizer-3.1.orig/src/settings.cpp	2008-11-10 21:06:50.886869559 +0200
++++ qorganizer-3.1/src/settings.cpp	2008-12-07 10:20:59.612341932 +0200
+@@ -428,7 +428,7 @@
+ 
+ void SettingsDialog::readLangDir()
+ {
+-        QDir dir = QDir(":/lang");
++        QDir dir = QDir("/usr/share/qorganizer/translations");
+         QString filter = "*.qm";
+ 	QDir::Filters filters = QDir::Files | QDir::Readable;
+ 	QDir::SortFlags sort = QDir::Name;
--- qorganizer-3.1.orig/debian/patches/300-produce-lowercase-binary.patch
+++ qorganizer-3.1/debian/patches/300-produce-lowercase-binary.patch
@@ -0,0 +1,13 @@
+Index: qorganizer-3.1.3/src/qOrganizer.pro
+===================================================================
+--- qorganizer-3.1.3.orig/src/qOrganizer.pro	2008-09-06 22:21:23.892153961 +0300
++++ qorganizer-3.1.3/src/qOrganizer.pro	2008-09-06 22:21:53.267785290 +0300
+@@ -3,7 +3,7 @@
+ ######################################################################
+ 
+ TEMPLATE = app
+-TARGET = 
++TARGET = qorganizer 
+ DEPENDPATH += . \
+     lang
+ INCLUDEPATH += .
--- qorganizer-3.1.orig/debian/patches/series
+++ qorganizer-3.1/debian/patches/series
@@ -0,0 +1,3 @@
+010-fix-compilation-with-qt4.5.patch 
+100-use-system-path-for-translations.patch
+300-produce-lowercase-binary.patch
