--- a/Makefile	2021-05-30 21:12:07.873794986 -0400
+++ b/Makefile	2021-05-30 21:12:53.273796202 -0400
@@ -17,12 +17,12 @@
 SUBSYSTEM =	SOLARIS
 
 # Locations
-DIR =		/home/dick
-BINDIR =	$(DIR)/bin.`$(DIR)/bin/arch`
-MAN1DIR =	$(DIR)/man/man1
+DIR =		/usr
+BINDIR =	$(DIR)/bin
+MAN1DIR =	$(DIR)/share/man/man1
 
 # Commands
-COPY =		cp -p
+COPY =		install -D
 EXE =		#
 LEX =		flex
 LN =		ln
@@ -32,29 +32,8 @@
 NULLFILE =	/dev/null
 
 ################################################################
-# For MSDOS + MinGW
-
-SYSTEM =	MSDOS
-SUBSYSTEM =	MinGW
-
-# Locations
-DIR =		C:/BIN
-BINDIR =	C:/BIN
-MAN1DIR =	C:/BIN
-
-# File names
-NULLFILE =	nul
-
-# Commands (cp required, since xcopy cannot handle forward slashes)
-COPY =		cp -p
-EXE =		.exe
-LEX =		flex
-LN =		ln
-ZIP =		zip -o
-
-################################################################
 # General, C compilation:
-CC =		gcc -D$(SYSTEM) -D$(SUBSYSTEM)
+CC =		gcc
 LINT =		lint -ansi -D$(SYSTEM) -D$(SUBSYSTEM)
 LINTFLAGS =	-xh
 
@@ -103,7 +82,7 @@
 LOG2ERRMSG_HDR =	             #module.h #debug.h
 
 log2errmsg$(EXE):$(LOG2ERRMSG_OBJ)
-		$(CC) $(LOG2ERRMSG_OBJ) -o $@ $(LIBFLAGS)
+		$(CC) -D$(SYSTEM) -D$(SUBSYSTEM) $(CFLAGS) $(LOG2ERRMSG_OBJ) -o $@ $(LIBFLAGS)
 GARBAGE +=	log2errmsg log2errmsg.exe
 
 # Test parameters for 'log2errmsg'
@@ -145,7 +124,7 @@
 install:	$(BINDIR)/log2errmsg$(EXE)
 
 $(BINDIR)/log2errmsg$(EXE):	log2errmsg$(EXE)
-		cp -p log2errmsg$(EXE) $@
+		$(COPY) log2errmsg$(EXE) $@
 
 # Miscellaneous
 
