--- configure.in.old	2010-11-22 17:11:17.000000000 -0500
+++ configure.in	2010-11-22 17:23:27.000000000 -0500
@@ -23,18 +23,34 @@
 dnl CFLAGS="$CFLAGS -Wall"
 dnl AC_SUBST(CFLAGS)
 
-dnl
-dnl READLINE_HOOK (script-if-readline-found, failflag)
-dnl
-dnl If failflag is "failure", script aborts due to lack of GNU readline
-dnl 
-dnl Check for availability of the GNU readline library
-
 AC_DEFUN([READLINE_HOOK],[
 	TERMCAP_LIB="-ltermcap"
-	AC_CHECK_LIB(readline,readline, [READLINE_LIB="-lreadline"],
-		AC_MSG_ERROR(Readline library not found),
-		$TERMCAP_LIB )
+   AC_CHECK_LIB(readline, readline,
+           READLINE_LIBS="-lreadline",
+        AC_MSG_CHECKING([for readline in -lreadline with -ltermcap])
+        ac_save_LIBS="$LIBS"
+        LIBS="$ac_save_LIBS -lreadline -ltermcap"
+        AC_TRY_LINK([#include <stdio.h>
+#include <readline/readline.h>],[readline("");],
+            AC_MSG_RESULT([yes])
+            READLINE_LIBS="-lreadline -ltermcap"
+            LIBS="$ac_save_LIBS"
+            have_readline=yes,
+          AC_MSG_RESULT([no])
+          AC_MSG_CHECKING([for readline in -lreadline with -lncurses])
+          LIBS="$ac_save_LIBS -lreadline -lncurses"
+          AC_TRY_LINK([#include <stdio.h>
+#include <readline/readline.h>],[readline("");],
+              AC_MSG_RESULT([yes])
+              READLINE_LIBS="-lreadline -lncurses"
+              LIBS="$ac_save_LIBS"
+              have_readline=yes,
+            AC_MSG_RESULT([no])
+            READLINE_LIBS=""
+            LIBS="$ac_save_LIBS"
+          )
+        )
+    )
 	READLINE_LIBS="$TERMCAP_LIB $READLINE_LIB"
 	AC_SUBST(READLINE_LIBS)
 ])
