--- configure.old	2010-11-22 17:37:01.000000000 -0500
+++ configure	2010-11-22 18:11:16.000000000 -0500
@@ -4902,11 +4902,12 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char readline ();
+#include <stdio.h>
+#include <readline/readline.h>
 int
 main ()
 {
-readline ();
+readline ("");
   ;
   return 0;
 }
