--- wotsap-0.7.orig/dl-latest.wot
+++ wotsap-0.7/dl-latest.wot
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# dl-latest.wot - Downloads latest description of the OpenPGP Web of Trust from
+# the site of wotsap main writer
+# Copyright (C) 2007 Giovanni Mascellani <g.mascellani@gmail.com>
+# Written for wotsap (Web of Trust statistics and pathfinder) Debian package
+# 
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+# 02111-1307, USA.
+
+WGET=/usr/bin/wget
+
+if [ -x $WGET ] ; then
+	wget http://www.lysator.liu.se/~jc/wotsap/wots2/latest.wot -O ~/.wotsapdb
+fi
+
--- wotsap-0.7.orig/debian/control
+++ wotsap-0.7/debian/control
@@ -0,0 +1,31 @@
+Source: wotsap
+Section: misc
+Priority: extra
+Maintainer: Giovanni Mascellani <g.mascellani@gmail.com>
+Build-Depends: debhelper (>= 5)
+Build-Depends-Indep: xsltproc, docbook-xml, docbook-xsl
+Standards-Version: 3.7.3
+Homepage: http://www.lysator.liu.se/~jc/wotsap/index.html
+
+Package: wotsap
+Architecture: all
+Depends: python, python-imaging, ttf-freefont | ttf-dejavu
+Suggests: gnupg, wget
+Description: OpenPGP Web of Trust analyzer and pathfinder
+ Wotsap is a tool that analyzes a OpenPGP Web of Trust description and reports
+ to the user stastistics about the single keys and the whole network, or
+ searches for paths from one key to another.
+ .
+ OpenPGP is the most widely used email encryption standard, used by encryption
+ software such as the GNU Privacy Guard (gpg).  To encrypt to someone or verify
+ someone's signature, you need that persons OpenPGP key. Say you want to verify
+ a digital signature made by Bob. To get Bob's key is easy with some help from
+ keyservers, being sure you got the right key is the tricky part. This is
+ accomplished either by meeting Bob in person and exchanging signatures, or by
+ trusting someone else, who you have met in person, who claims to have met Bob.
+ Or by trusting someone who has met someone who has met Bob. This gives rise to
+ a completely decentralized network of trusts between people.
+ .
+ Wotsap lets you explore the Web of Trust. It works on a compressed copy of the
+ Web of Trust in the .wot file format, generated daily and distributed on the
+ site of wotsap main writer.
--- wotsap-0.7.orig/debian/compat
+++ wotsap-0.7/debian/compat
@@ -0,0 +1 @@
+5
--- wotsap-0.7.orig/debian/wotsap.manpages
+++ wotsap-0.7/debian/wotsap.manpages
@@ -0,0 +1 @@
+man/wotsap.1
--- wotsap-0.7.orig/debian/rules
+++ wotsap-0.7/debian/rules
@@ -0,0 +1,44 @@
+#!/usr/bin/make -f
+# -*- makefile -*-
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+build: build-stamp
+
+build-stamp:
+	dh_testdir
+	$(MAKE)
+	touch $@
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp
+	$(MAKE) clean
+	dh_clean 
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_installdirs
+
+binary-indep: build install
+	dh_testdir
+	dh_testroot
+	dh_installchangelogs ChangeLog
+	dh_installdocs
+	dh_installexamples
+	dh_install
+	dh_installman
+	dh_compress
+	dh_fixperms
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary-arch: build install
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install
+
--- wotsap-0.7.orig/debian/wotsap.docs
+++ wotsap-0.7/debian/wotsap.docs
@@ -0,0 +1,2 @@
+wotfileformat.txt
+README
--- wotsap-0.7.orig/debian/wotsap.install
+++ wotsap-0.7/debian/wotsap.install
@@ -0,0 +1 @@
+wotsap usr/bin
--- wotsap-0.7.orig/debian/copyright
+++ wotsap-0.7/debian/copyright
@@ -0,0 +1,99 @@
+
+Package original author:
+   Jörgen Cederlöf <jc@lysator.liu.se>
+
+With contributions from:
+   Marco Bodrato <gpg@bodrato.it> (wanted signatures)
+
+Original tarball was downloaded from:
+   http://www.lysator.liu.se/~jc/wotsap/index.html
+
+Copyright holders:
+   2003,2004  Jörgen Cederlöf <jc@lysator.liu.se>
+   2004  Marco Bodrato <gpg@bodrato.it>
+
+The dl-latest.wot bash script was written by Giovanni Mascellani
+   <g.mascellani@gmail.com> and is by him copyrighted.
+
+The Debian packaging is (C) 2007 Giovanni Mascellani <g.mascellani@gmail.com>
+and is licensed under the GPL, see below.
+
+
+License:
+
+    This package is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ 
+    This package is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+ 
+    You should have received a copy of the GNU General Public License
+    along with this package; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+
+On Debian systems, the complete text of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL'.
+
+
+The font included in the package main binary is extracted from the Debian
+package xfonts-75dpi and is:
+Copyright 1984-1989, 1994 Adobe Systems Incorporated.
+Copyright 1988, 1994 Digital Equipment Corporation.
+
+Adobe is a trademark of Adobe Systems Incorporated which may be
+registered in certain jurisdictions.
+Permission to use these trademarks is hereby granted only in
+association with the images described in this file.
+ 
+Permission to use, copy, modify, distribute and sell this software
+and its documentation for any purpose and without fee is hereby
+granted, provided that the above copyright notices appear in all
+copies and that both those copyright notices and this permission
+notice appear in supporting documentation, and that the names of
+Adobe Systems and Digital Equipment Corporation not be used in
+advertising or publicity pertaining to distribution of the software
+without specific, written prior permission.  Adobe Systems and
+Digital Equipment Corporation make no representations about the
+suitability of this software for any purpose.  It is provided "as
+is" without express or implied warranty.
+
+--------
+
+File: wotsap
+Copyright: Copyright 2003, 2004 Jörgen Cederlöf <jc@lysator.liu.se>
+           Copyright 2004 Marco Bodrato <gpg@bodrato>
+	       Copyright 1984-1989, 1994 Adobe Systems Incorporated
+	       Copyright 1988, 1994 Digital Equipment Corporation
+License: GPL-2+, other
+ The file contains a font extracted from Debian package xfonts-75dpi.
+ This font is licensed under the following terms:
+ .
+ Adobe is a trademark of Adobe Systems Incorporated which may be
+ registered in certain jurisdictions.
+ Permission to use these trademarks is hereby granted only in
+ association with the images described in this file.
+ .
+ Permission to use, copy, modify, distribute and sell this software
+ and its documentation for any purpose and without fee is hereby
+ granted, provided that the above copyright notices appear in all
+ copies and that both those copyright notices and this permission
+ notice appear in supporting documentation, and that the names of
+ Adobe Systems and Digital Equipment Corporation not be used in
+ advertising or publicity pertaining to distribution of the software
+ without specific, written prior permission.  Adobe Systems and
+ Digital Equipment Corporation make no representations about the
+ suitability of this software for any purpose.  It is provided "as
+ is" without express or implied warranty.
+
+File: Webware-Main.py
+Copyright: Copyright 2003 Jörgen Cederlöf <jc@lysator.liu.se>
+License: GPL-2+
+
+File: dl-latest.wot, debian/*, man/*
+Copyright: Copyright 2007 Giovanni Mascellani <g.mascellani@gmail.com>
+License: GPL-2+
+
--- wotsap-0.7.orig/debian/wotsap.examples
+++ wotsap-0.7/debian/wotsap.examples
@@ -0,0 +1,3 @@
+dl-latest.wot
+pks2wot
+Webware-Main.py
--- wotsap-0.7.orig/debian/watch
+++ wotsap-0.7/debian/watch
@@ -0,0 +1,2 @@
+version=3
+http://www.lysator.liu.se/~jc/wotsap/download/wotsap-(.*)\.tgz debian
--- wotsap-0.7.orig/debian/changelog
+++ wotsap-0.7/debian/changelog
@@ -0,0 +1,16 @@
+wotsap (0.7-2) unstable; urgency=low
+
+  * Updated to policy 3.7.3.
+  * Homepage tag moved from the description to the control field.
+  * Removed references to ttf-bitstream-vera, which will be removed
+    soon. (Closes: #461298)
+  * Added debian/watch. (Closes: #466599)
+  * Moved to machine-interpretable debian/copyright.
+
+ -- Giovanni Mascellani <g.mascellani@gmail.com>  Sun, 24 Feb 2008 17:43:12 +0100
+
+wotsap (0.7-1) unstable; urgency=low
+
+  * Initial release. (Closes: #434395)
+
+ -- Giovanni Mascellani <g.mascellani@gmail.com>  Wed, 08 Aug 2007 10:50:52 +0200
--- wotsap-0.7.orig/man/wotsap.dbk
+++ wotsap-0.7/man/wotsap.dbk
@@ -0,0 +1,308 @@
+<?xml version='1.0' encoding='ISO-8859-1'?>
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+"http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd" [
+
+  <!ENTITY dhfirstname "<firstname>Giovanni</firstname>">
+  <!ENTITY dhsurname   "<surname>Mascellani</surname>">
+  <!ENTITY dhdate      "<date>July 2007</date>">
+  <!ENTITY dhsection   "<manvolnum>1</manvolnum>">
+  <!ENTITY dhemail     "<email>g.mascellani@gmail.com</email>">
+  <!ENTITY dhusername  "Giovanni Mascellani">
+  <!ENTITY dhucpackage "<refentrytitle>WOTSAP</refentrytitle>">
+  <!ENTITY dhpackage   "wotsap">
+
+  <!ENTITY debian      "<productname>Debian</productname>">
+  <!ENTITY gnu         "<acronym>GNU</acronym>">
+  <!ENTITY gpl         "&gnu; <acronym>GPL</acronym>">
+]>
+
+<refentry>
+  <refentryinfo>
+    <address>
+      &dhemail;
+    </address>
+    <copyright>
+      <year>2007</year>
+      <holder>&dhusername;</holder>
+    </copyright>
+    &dhdate;
+  </refentryinfo>
+  <refmeta>
+    &dhucpackage;
+
+    &dhsection;
+  </refmeta>
+  <refnamediv>
+    <refname>&dhpackage;</refname>
+
+    <refpurpose>Web of Trust statistics and pathfinder</refpurpose>
+  </refnamediv>
+  <refsynopsisdiv>
+    <cmdsynopsis>
+      <command>&dhpackage;</command>
+
+      <arg><option>OPTIONS</option></arg>
+      <arg>bottomkey <arg>topkey</arg></arg>
+    </cmdsynopsis>
+  </refsynopsisdiv>
+  <refsect1>
+    <title>DESCRIPTION</title>
+
+    <para><command>&dhpackage;</command> is a pathfinder for the OpenPGP Web
+      of Trust, which also produces some statistics and similar things. It
+      obtains the description of the Web of Trust from a .wot file.</para>
+
+    <para>OpenPGP is the most widely used email encryption standard, used by
+      encryption software such as the GNU Privacy Guard (see
+      <command>gnupg</command> (7)). To encrypt to someone or verify someone's
+      signature, you need that persons OpenPGP key. Say you want to verify a
+      digital signature made by Bob. To get Bob's key is easy
+      with some help from keyservers, being sure you got the right key is the
+      tricky part. This is accomplished either by meeting Bob in person and
+      exchanging signatures, or by trusting someone else, who you have met in
+      person, who claims to have met Bob. Or by trusting someone who has met
+      someone who has met Bob. This gives rise to a completely decentralized
+      network of trusts between people.</para>
+
+    <para><command>&dhpackage;</command> lets you explore the Web of
+      Trust. It works on a compressed copy of the Web of Trust in the
+      .wot file format, generated daily and distributed on the site of
+      <command>&dhpackage;</command> main writer.</para>
+
+    <para>Running <command>&dhpackage;</command> you can specify on the command
+      line just bottomkey, bottomkey and topkey or neither of them. They are the
+      hexadecimal ID of the keys you want to investigate:</para>
+
+    <itemizedlist>
+
+      <listitem><para>If you don't specify any of bottomkey and topkey,
+        <command>&dhpackage;</command> will just produce some piece of
+        information on the Wot of Trust, such as the total number of key and
+        signatures and the average signatures per key.</para></listitem>
+
+      <listitem><para>If you specify only bottomkey, <command>&dhpackage;</command>
+        will output detailed statistics about that key: how far are the other
+        key in the Web from this, the Mean Shortest Distance for this key, which
+        key have signed this key or are signed by this key. The Mean Shortest
+        Distance is the average of the distances from all the keys in the Web of
+        Trust to the key you under considaration. The distance from a key to
+        another is the length of a minimal path from the first to the
+        second.</para></listitem>
+
+      <listitem><para>If you specify either bottomkey and topkey,
+        <command>&dhpackage;</command> will search for all the minimal paths
+        from bottomkey to topkey and output them.</para></listitem>
+
+    </itemizedlist>
+
+  </refsect1>
+  <refsect1>
+    <title>OPTIONS</title>
+
+    <variablelist>
+
+      <varlistentry>
+        <term><option>-h</option>
+          <option>--help</option>
+        </term>
+        <listitem>
+          <para>Show help.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>--version</option>
+        </term>
+        <listitem>
+          <para>Show version.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-w</option>
+          <option>--wot=FILE</option>
+        </term>
+        <listitem>
+          <para>Read Web of Trust information from FILE. Defaults to
+            ~/.wotsapdb. You can find the latest version of this file (generated
+            daily) at http://www.lysator.liu.se/~jc/wotsap/wots2/latest.wot.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-m</option>
+          <option>--modify=STR</option>
+        </term>
+        <listitem>
+          <para>Use STR of WoT modification string.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-g</option>
+          <option>--group</option>
+        </term>
+        <listitem>
+          <para>Print signature matrix of comma separated keys. If there aren't
+            any commas in the key specification, <command>&dhpackage;</command>
+            will interpret it as a search string, and print the signature matrix
+            of all the key, whose name or email contains the string.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-G</option>
+          <option>--nounknowns</option>
+        </term>
+        <listitem>
+          <para>Don't print unknown keys in the signature matrix.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-o</option>
+          <option>--png=FILE</option>
+        </term>
+        <listitem>
+          <para>Write an image of the graph of all the minimal paths between
+            the two specified keys in FILE, in .png file format.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-O</option>
+          <option>--show-png=PRG</option>
+        </term>
+        <listitem>
+          <para>Like <option>-o</option>, but shows the image with PRG instead
+            of writing it in a file. More precisely, it saves the minimal paths
+            image in a temporary file, the executes 'PRG image_file' and, when
+            PRG is over, deletes the temporary file.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-s</option>
+          <option>--size=NNNxMMM</option>
+        </term>
+        <listitem>
+          <para>With <option>-o</option> or <option>-O</option> options, set the
+            size of the generated image to NNNxMMM.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-F</option>
+          <option>--font=FILE</option>
+        </term>
+        <listitem>
+          <para>Use FILE as the font file, in .pil/.pbm format. Point
+            it to the .pil file, with the .pbm file in the same directory.
+            If no font specifications are given, <command>&dhpackage;</command>
+            will try to use some fonts installed in the system, or fall back to
+            a built-in one.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-T</option>
+          <option>--ttffont=FILE</option>
+        </term>
+        <listitem>
+          <para>Like <option>-F</option>, but with a TrueType font file.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-S</option>
+          <option>--ttfsize=NUM</option>
+        </term>
+        <listitem>
+          <para>With <option>-T</option>, set the TrueType font size. Defaults
+            to 16.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-p</option>
+          <option>--print</option>
+        </term>
+        <listitem>
+          <para>Print the whole Web of Trust in human readable format (very long
+            output).</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-D</option>
+          <option>--print-debug</option>
+        </term>
+        <listitem>
+          <para>Print the debug information in the .wot file.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-d</option>
+          <option>--diff=FILE</option>
+        </term>
+        <listitem>
+          <para>Print all differences between two .wot files (namely, between
+            the file specified with <option>-w</option> and that specified with
+            this option).</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-M</option>
+          <option>--msd</option>
+        </term>
+        <listitem>
+          <para>Just show MSD for the specified key instead of full statistics.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-W</option>
+          <option>--wanted[=NUM]</option>
+        </term>
+        <listitem>
+          <para>Show the NUM (defaults to 10) 'most wanted signatures' for key
+            (very long calculation time, try using <option>-r</option>).</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>-r</option>
+          <option>--restrict=STR</option>
+        </term>
+        <listitem>
+          <para>Restrict wanted signatures with STR, implies -W.</para>
+        </listitem>
+      </varlistentry>
+
+    </variablelist>
+  </refsect1>
+  <refsect1>
+    <title>SEE ALSO</title>
+
+    <para><command>gnupg</command> (7), <command>gpg</command> (1).</para>
+  </refsect1>
+  <refsect1>
+    <title>AUTHOR</title>
+
+    <para>This manual page was written by &dhusername; &dhemail; for
+      the &debian; system (but may be used by others), beacause the
+      original program doesn't have a manual page.  Permission is
+      granted to copy, distribute and/or modify this document under
+      the terms of the &gnu; General Public License, Version 2 any 
+      later version published by the Free Software Foundation.
+    </para>
+	<para>
+	  On Debian systems, the complete text of the GNU General Public
+	  License can be found in /usr/share/common-licenses/GPL.
+	</para>
+
+  </refsect1>
+</refentry>
+
--- wotsap-0.7.orig/man/Makefile
+++ wotsap-0.7/man/Makefile
@@ -0,0 +1,14 @@
+
+DB2MAN=/usr/share/xml/docbook/stylesheet/nwalsh/manpages/docbook.xsl
+XP=xsltproc -''-nonet
+
+all: wotsap.1
+
+wotsap.1: wotsap.dbk
+	$(XP) $(DB2MAN) $<
+
+clean:
+	rm -fr wotsap.1 *~
+
+.PHONY: all clean
+
--- wotsap-0.7.orig/Makefile
+++ wotsap-0.7/Makefile
@@ -0,0 +1,10 @@
+
+all:
+	make -C man
+
+clean:
+	rm -fr *~ core *.pyc
+	make -C man clean
+
+.PHONY: all clean
+
--- wotsap-0.7.orig/wotsap
+++ wotsap-0.7/wotsap
@@ -1494,6 +1494,20 @@
             self.font = ImageFont.load(fontfile)
             return
 
+        # Modified by Giovanni Mascellani <g.mascellani@gmail.com> for Debian packaging
+        # If no fonts are specified on command line tries first for existing TTF
+        # before falling back to the built-in one.
+	try:
+            self.font = ImageFont.truetype("/usr/share/fonts/truetype/ttf-dejavu/DejaVuSans.ttf", ttffilesize)
+            return
+	except IOError:
+            pass
+	try:
+            self.font = ImageFont.truetype("/usr/share/fonts/truetype/freefont/FreeSans.ttf", ttffilesize)
+            return
+	except IOError:
+            pass
+
         import StringIO, base64, zlib
         self.font = ImageFont.ImageFont()
 
