--- a/rarcrack.c	2020-06-10 22:38:21.639813083 -0400
+++ b/rarcrack.c	2020-06-10 22:39:10.984814404 -0400
@@ -280,7 +280,7 @@
 				}
 			} else if (strcmp(argv[i],"--type") == 0) {
 				if ((i + 1) < argc) {
-					sscanf(argv[++i], "%s", &test);
+					sscanf(argv[++i], "%s", test);
 					for (j = 0; strcmp(TYPE[j], "") != 0; j++) {
 						if (strcmp(TYPE[j], test) == 0) {
 							strcpy(finalcmd, CMD[j]);
