--- examples/sample/Makefile.in.old	2010-11-24 00:55:36.000000000 -0500
+++ examples/sample/Makefile.in	2010-11-24 00:55:52.000000000 -0500
@@ -186,7 +186,7 @@
 	latex sample
 	dvips sample -o sample.ps
 
-sample_gantt_1.eps: sample.sched
+sample_gantt_1.eps sample_network_1.eps: sample.sched
 	../../src/opensched sample.sched
 
 png :: sample_gantt_1.png
--- examples/toffee/Makefile.in.old	2010-11-24 00:57:50.000000000 -0500
+++ examples/toffee/Makefile.in	2010-11-24 00:58:25.000000000 -0500
@@ -184,7 +184,7 @@
 	latex toffee
 	dvips toffee -o toffee.ps
 
-toffee_gantt_all.eps : toffee.sched
+toffee_gantt_all.eps toffee_network_all.eps: toffee.sched
 	../../src/opensched toffee.sched
 
 png :: toffee_gantt_all.png
