--- a/Makefile	2021-05-30 22:17:31.188900049 -0400
+++ b/Makefile	2021-05-30 22:19:26.092903126 -0400
@@ -17,12 +17,12 @@
 SUBSYSTEM =	SOLARIS
 
 # Locations
-DIR =		/home/dick
-BINDIR =	$(DIR)/bin.`$(DIR)/bin/arch`
-MAN1DIR =	$(DIR)/man/man1
+DIR =		/usr
+BINDIR =	$(DIR)/bin
+MAN1DIR =	$(DIR)/share/man/man1
 
 # Commands
-COPY =		cp -p
+COPY =		install -D
 EXE =		#
 LEX =		flex
 LN =		ln
@@ -32,29 +32,8 @@
 NULLFILE =	/dev/null
 
 ################################################################
-# For MSDOS + MinGW
-
-SYSTEM =	MSDOS
-SUBSYSTEM =	MinGW
-
-# Locations
-DIR =		C:/BIN
-BINDIR =	C:/BIN
-MAN1DIR =	C:/BIN
-
-# File names
-NULLFILE =	nul
-
-# Commands (cp required, since xcopy cannot handle forward slashes)
-COPY =		cp -p
-EXE =		.exe
-LEX =		flex
-LN =		ln
-ZIP =		zip -o
-
-################################################################
 # General, C compilation:
-CC =		gcc -D$(SYSTEM) -D$(SUBSYSTEM)
+CC =		gcc
 LINT =		lint -ansi -D$(SYSTEM) -D$(SUBSYSTEM)
 LINTFLAGS =	-xh
 
@@ -115,7 +94,7 @@
 		rm foo bar
 
 utime$(EXE):	$(UTIME_OBJ)
-		$(CC) $(UTIME_OBJ) -o $@ $(LIBFLAGS)
+		$(CC) -D$(SYSTEM) -D$(SUBSYSTEM) $(CFLAGS) $(UTIME_OBJ) -o $@ $(LIBFLAGS)
 
 
 
@@ -125,10 +104,10 @@
 install:	$(BINDIR)/utime$(EXE) $(MAN1DIR)/utime.1
 
 $(BINDIR)/utime$(EXE):	utime$(EXE)
-		cp -p utime$(EXE) $@
+		$(COPY) utime$(EXE) $@
 
 $(MAN1DIR)/utime.1:	utime.1
-		cp -p utime.1 $@
+		$(COPY) utime.1 $@
 
 # Miscellaneous
 lint:		$(UTIME_SRC) $(UTIME_HDR)
