diff -Nur mailliststat-1.3/examples/example.php mailliststat-1.3.new/examples/example.php
--- mailliststat-1.3/examples/example.php	2005-12-01 21:44:46.000000000 +0100
+++ mailliststat-1.3.new/examples/example.php	2005-12-01 22:16:50.000000000 +0100
@@ -6,7 +6,6 @@
 require("mls_class.php");
 $mls=new MailListStat;
 $mls->title="Statistics for MOBiL (year 2002)";
-$mls->default_lang="sk";
 $mls->default_topX="25";
 $mls->path="/usr/local/bin/mailliststat"; // mailliststat executable
 $mls->input="/home/marki/2002.cache"; // input file
diff -Nur mailliststat-1.3/html/mls_class.php mailliststat-1.3.new/html/mls_class.php
--- mailliststat-1.3/html/mls_class.php	2005-12-01 21:44:46.000000000 +0100
+++ mailliststat-1.3.new/html/mls_class.php	2005-12-01 22:16:13.000000000 +0100
@@ -27,7 +27,6 @@
 ===cut===example===list.php===
 require("mls.php");
 $mls=new MailListStat;
-$mls->default_lang="SK";
 $mls->input="/home/marki/list.cache";
 $mls->is_cache=true;
 $mls->Run();
@@ -49,8 +48,7 @@
  // graphs to show (-g in mailliststat; don't specify to use default in
  // mailliststat)
  var $graph="";
- // default values for LANG & topXX
- var $default_lang="EN";
+ // default values for topXX
  var $default_topX="10";
  // alter email addresses in output to prevent spam?
  // Specify regexp search & replace patterns; empty to do nothing
@@ -59,7 +57,6 @@
  var $email_srch="@";
  var $email_repl=" (at) ";
  //$email_repl="@NOSPAM.";
- var $lang="";
  var $topX="";
  var $timer_start=0;
 
@@ -70,12 +67,10 @@
  function Run() { // main function called by user
    global $HTTP_GET_VARS;
    // get user submitted values (or use default)
-   $this->lang=$HTTP_GET_VARS['lang']; // language (-l option in mailliststat)
    $this->topX=$HTTP_GET_VARS['topX']; // print topXX (-n)
-   if (!$this->lang) $this->lang=$this->default_lang;
    if (!$this->topX) $this->topX=$this->default_topX;
    // check user (web) submitted data for validity
-   if (!eregi("^[a-z]{2}$",$this->lang) || !eregi("^[0-9]{1,3}$",$this->topX))
+   if (!eregi("^[0-9]{1,3}$",$this->topX))
       $this->error("You have specified invalid parameters!");
    echo $this->RunMLS(); // run mailliststat & print output
    $this->timeEnd();
@@ -96,7 +91,7 @@
    if ($this->name)  $exec.=" -t ".escapeShellArg($this->name);
    if ($this->title) $exec.=" -T ".escapeShellArg($this->title);
    if ($this->graph) $exec.=" -g ".escapeShellArg($this->graph);
-   $exec.=" -n $this->topX -l $this->lang $inp ".escapeShellArg($this->input);
+   $exec.=" -n $this->topX $inp ".escapeShellArg($this->input);
    $exec.=" -q"; // quiet
    exec($exec, $a_out, $ret);
    // check return value
@@ -119,26 +114,15 @@
    } else return $buff;
  }// RunMLS()
 
- function PrepareLinks() { // prepare links for changing LANG & topXX
+ function PrepareLinks() { // prepare links for changing topXX
    global $HTTP_SERVER_VARS;
    $self=$HTTP_SERVER_VARS['PHP_SELF'];
    $added="<center>\n";
-   $added.="<b>Language:</b>&nbsp;[&nbsp;";
-   $a_lang1=array("de","en","fr","it","sk","es","sr","br");
-   $a_lang2=array("Deutsch","English","Francais","Italiano","Slovak","Spanish","Serbian","Portugues Brasil");
-   for ($i=0; $i < count($a_lang1); $i++) {
-     $o1=$a_lang1[$i];
-     $o2=$a_lang2[$i];
-     if (StrToUpper($this->lang)!=StrToUpper($o1)) $o2="<a href=\"$self?topX=$this->topX&lang=$o1\">$o2</a>";
-     $o2.="&nbsp;";
-     if ($i) $o2="|&nbsp;".$o2;
-     $added.=$o2;
-   }
-   $added.="]<br>\n<b>Print TOP:</b>&nbsp;[&nbsp;";
+   $added.="<b>Print TOP:</b>&nbsp;[&nbsp;";
    $a_topX=array("5","10","20","25","50","100","200","250","500","999");
    for ($i=0; $i < count($a_topX); $i++) {
      $o1=$a_topX[$i];
-     if ($this->topX != $o1) $o1="<a class=\"wr\" href=\"$self?topX=$o1&lang=$this->lang\">$o1</a>";
+     if ($this->topX != $o1) $o1="<a class=\"wr\" href=\"$self?topX=$o1\">$o1</a>";
      $o1.="&nbsp;";
      if ($i) $o1="|&nbsp;".$o1;
      $added.=$o1;
diff -Nur mailliststat-1.3/INSTALL mailliststat-1.3.new/INSTALL
--- mailliststat-1.3/INSTALL	2005-12-01 21:44:46.000000000 +0100
+++ mailliststat-1.3.new/INSTALL	2005-12-01 22:10:32.000000000 +0100
@@ -33,16 +33,4 @@
 * PHP wrapper - see file README
 * Examples    - see dir  examples/
 
- How to add another language
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-* You can just translate everything in mls_lang.h and send it to me and I will
-  include it in next release.
-
-* Or can add another language as specified in mls_lang.h, but you will need to
-  change also mls.c
-  * add line for your language to parameters-parsing switch(c) - it's at the
-    beginning of main()
-
-* Sorry for such a bad implementation, but I have no time to learn gettext()
-  and it has disadvantage that it needs specials files to be located somewhere.
 /****************************************************************************/
diff -Nur mailliststat-1.3/Makefile mailliststat-1.3.new/Makefile
--- mailliststat-1.3/Makefile	2005-12-01 22:01:57.000000000 +0100
+++ mailliststat-1.3.new/Makefile	2005-12-01 22:03:18.000000000 +0100
@@ -8,18 +8,20 @@
 OPTIMIZE=-O3
 
 PACKAGE=mailliststat
+DESTDIR=/usr/local
+LOCALEDIR=$(DESTDIR)/share/locale
 # debug symbols will be stripped anyway during 'make install'
 DEBUG=-g
 
 #### DON'T CHANGE ANYTHING BELOW ####
-DESTDIR=/usr/local
 CFLAGS=-Wall $(OPTIMIZE) $(DEBUG)
 LIBS=-lm
-CC=gcc
+CC=gcc -DLOCALEDIR="\"$(LOCALEDIR)\"" -DPACKAGE="\"$(PACKAGE)\""
 
 OBJS=mls.o mls_mime.o mls_text.o mls_list.o mls_stat.o
 
 all: mailliststat
+	make -C po/ all
 	make -C man/ all
 
 $(PACKAGE): $(OBJS) Makefile
@@ -37,7 +39,7 @@
 mls_stat.o: mls_stat.c mls_stat.h Makefile
 	$(CC) $(CFLAGS) -c mls_stat.c -o mls_stat.o
 
-mls_text.o: mls_text.c mls_text.h mls_lang.h Makefile
+mls_text.o: mls_text.c mls_text.h Makefile
 	$(CC) $(CFLAGS) -c mls_text.c -o mls_text.o
 
 install: $(PACKAGE)
@@ -51,12 +53,16 @@
 	@echo "*** See also 'examples' subdir for some hints...           ***"
 	@echo "******************** MLS install complete ********************"
 	make -C man/ install DESTDIR=$(DESTDIR)
+	make -C po/ install LOCALEDIR=$(LOCALEDIR)
 
 clean:
 	rm -f $(wildcard *.o) $(PACKAGE)
+	make -C po/ clean
 	make -C man/ clean
 
+
 uninstall:
 	rm -fv $(DESTDIR)/bin/$(PACKAGE)
 	rm -fv $(DESTDIR)/man/man1/$(PACKAGE).1.gz
 	make -C man/ uninstall DESTDIR=$(DESTDIR)
+	make -C po/ uninstall LOCALEDIR=$(LOCALEDIR)
diff -Nur mailliststat-1.3/mls.c mailliststat-1.3.new/mls.c
--- mailliststat-1.3/mls.c	2005-12-01 21:44:46.000000000 +0100
+++ mailliststat-1.3.new/mls.c	2005-12-01 22:08:45.000000000 +0100
@@ -23,12 +23,13 @@
 #include "mls_mime.h"
 #include "mls_list.h"
 #include "mls_stat.h"
+#include <libintl.h>
+
 
 /* ************************************************** GLOBAL VARIABLES *** */
 FILE *fInp,*fOut;  // In/Out file handler
 long nSprav=0;     // total number of messages
 long nDlzka=0;     // total size of msgs (only body parts are counted)
-int lang=99;       // language to use on output (99=not def.)
 int verbose=0;     // verbose mode
 int quiet=0;       // quiet mode
 int onlytitle=0;   // show only user-defined text as title? (w/o oldest/newest date)
@@ -64,7 +65,6 @@
   fprintf(stderr,"  -h      ... print this help text and exit\n");
   fprintf(stderr,"  -v      ... verbose mode - print some \"useful\" info\n");
   fprintf(stderr,"  -q      ... be quiet (print only errors to stderr)\n");
-  fprintf(stderr,"  -l lang ... select output language (EN,SK,IT,FR,DE,ES,SR,BR)*\n");
   fprintf(stderr,"  -i file ... name of input file (if not specified, stdin)\n");
   fprintf(stderr,"  -o file ... name of output file (if not specified, stdout)\n");
   fprintf(stderr,"  -r file ... read input from cache file instead of mailbox\n");
@@ -255,24 +255,16 @@
   struct rusage r_usage;      // for resource usage stats
 #endif
 
+  setlocale (LC_ALL, "");
+  bindtextdomain (PACKAGE, LOCALEDIR);
+  textdomain (PACKAGE);
+
   if (gettimeofday(&t_beg,NULL)) myWarn("Error: gettimeofday()",strerror(errno),0);
   t_oldest=t_newest=0;
   // parse parameters
   opterr=0;
   while ((c=getopt(argc, argv, "hl:i:o:t:T:n:r:w:u:m:g:vq"))>=0) {
     switch (c) {
-      case 'l': // output language
-                lang=99;
-                if (!strncasecmp(optarg,"SK\000",3)) lang=0;
-                if (!strncasecmp(optarg,"EN\000",3)) lang=1;
-                if (!strncasecmp(optarg,"IT\000",3)) lang=2;
-                if (!strncasecmp(optarg,"FR\000",3)) lang=3;
-                if (!strncasecmp(optarg,"DE\000",3)) lang=4;
-                if (!strncasecmp(optarg,"ES\000",3)) lang=5;
-                if (!strncasecmp(optarg,"SR\000",3)) lang=6;
-                if (!strncasecmp(optarg,"BR\000",3)) lang=7;
-		if (lang==99) myWarn("Invalid language!","\nSK=Slovak  EN=English IT=Italiano ES=Spanish\nSR=Serbian DE=Deutsch FR=Francais BR=Portugues Brasil",2);
-                break;
       case 'm': // mode of output stats
                 mode=99;
                 if (!strncasecmp(optarg,"text\000" ,5)) mode=0;
@@ -374,11 +366,6 @@
      fOut=stdout;
   }
 
-  if ((lang==99) && !(files & (F_WRITE | F_UPDATE))) { // no language && doing normal output (= not doing cache write/update)
-     myWarn("No output language specified, using default English.","",0);
-     lang=1;
-  }
-
   if ((msgQ=(nQptr)malloc(sizeof(struct bestQUOTE)))==NULL) myWarn("Not enough memory!","",4);
   msgQ->auth[0]='\0';
   msgQ->subj[0]='\0';
diff -Nur mailliststat-1.3/mls_mime.c mailliststat-1.3.new/mls_mime.c
--- mailliststat-1.3/mls_mime.c	2005-12-01 21:44:46.000000000 +0100
+++ mailliststat-1.3.new/mls_mime.c	2005-12-01 22:42:37.000000000 +0100
@@ -94,7 +94,7 @@
 
 void compile_mime_regex() {
  char sB[]="(=\\?([0-9a-z-]*)\\?B\\?([0-9a-z+/=]*)\\?=)";
- char sQ[]="(=\\?([0-9a-z-]*)\\?Q\\?(([!-<]|[>-~]|=[0-9a-f]{2}|\\0x9| |_)*)\\?=)";
+ char sQ[]="(=\\?([0-9a-z-]*)\\?Q\\?(([:print:]*[^? ])*)\\?=)";
  char sH[]="(=[0-9a-z]{2})";
 
  if (regcomp(&r_base64, sB, REG_EXTENDED | REG_ICASE) > 0) myWarn("Error compiling regex!","(mime_base64)",5);
diff -Nur mailliststat-1.3/README.txt mailliststat-1.3.new/README.txt
--- mailliststat-1.3/README.txt	2005-12-01 21:44:46.000000000 +0100
+++ mailliststat-1.3.new/README.txt	2005-12-01 22:28:18.000000000 +0100
@@ -38,10 +38,6 @@
   -r file ... read input from cache file instead of mailbox
   -w file ... write cache file (no stats produced)
   -u file ... update cache file (read cache, read input, write cache)
-  -l lang ... output language - SK (Slovak), EN (English), IT (Italiano),
-              FR (Francais), DE (Deutsch), ES (Spanish), SR (Serbian),
-	      BR (Portugues Brasil) {EN}
-              diagnostic messages are always printed in English
   -m mode ... mode of output (text, html, html2) {text}
   -n XX   ... print TOP XX authors, subjects and quoting {10}
   -t text ... name of mailing list, see below for details
@@ -66,9 +62,9 @@
   reporting oldest message found.
 
 * Output
-  Statistics is put into output file (or stdout if unspecified) in specified
-  language. All diagnostic messages are written to stderr and are in English.
-  Output consists of several statistical data - tables, graphs and summaries.
+  Statistics is put into output file (or stdout if unspecified). All
+  diagnostic messages are written to stderr. Output consists of several
+  statistical data - tables, graphs and summaries.
   The title has two formats depending on "-t" parameter. If it's not specified,
   it looks like "Statistics from 16.8.2001 to 7.9.2001", where first date is
   date of the oldest message found in input and second is date of the newest
@@ -109,8 +105,7 @@
   file is very fast, so it is useful for integration with web pages. Now you
   can update the cache file just after new mail was received. Users can view
   actual stats using MLS as CGI script. It has an advantage over static stats
-  that user can choose language and others options and it will be generated
-  in a moment!
+  that user can choose options and it will be generated in a moment!
 
   To update cache file, use the -u option. It works like this: first, the stats
   are loaded from cache file (doesn't have to exist) and then new message(s)
@@ -132,12 +127,12 @@
 
 * PHP wrapper
   I have written also PHP wrapper for MLS to make it more "interactive". It has
-  two major advantages over plain HTML output from MLS: User can choose output
-  language and number of TOP items to show. It works by running MLS with
-  appopriate command-line options. It's safe, because only two items from user
-  are language and topXX which are checked using regexp, so running arbitrary
-  code is not possible. You can also alter MLS output - for example change @
-  in email addresses to (at) to prevent spamming.
+  one major advantage over plain HTML output from MLS: User can choose number
+  of TOP items to show. It works by running MLS with appopriate command-line
+  options. It's safe, because only one item from user is topXX which is
+  checked using regexp, so running arbitrary code is not possible. You can
+  also alter MLS output - for example change @ in email addresses to (at) to
+  prevent spamming.
   
   You can have normal MBOX file as input, but I recommend using cache file.
   When using cache file, the stats are produced in a moment. You can see how
