=== modified file 'libdbustest/Makefile.am'
--- libdbustest/Makefile.am	2012-10-02 03:11:56 +0000
+++ libdbustest/Makefile.am	2012-12-14 17:43:38 +0000
@@ -28,6 +28,7 @@
 	$(COVERAGE_CFLAGS) \
 	-I$(top_srcdir) \
 	-DDEFAULT_SESSION_CONF="\"$(datadir)/dbus-test-runner/session.conf\"" \
+	-DBUSTLE_DUAL_MONITOR="\"$(pkgdatadir)/dbus-test-bustle-handler\"" \
 	-DG_LOG_DOMAIN=\"libdbustest\" \
 	-Wall -Werror -Wextra
 
@@ -41,7 +42,11 @@
 	-no-undefined \
 	-export-symbols-regex "^[^_].*"
 
+pkgdata_SCRIPTS = \
+	dbus-test-bustle-handler
+
 EXTRA_DIST = \
+	dbus-test-bustle-handler \
 	dbustest.pc.in
 
 pkgconfig_DATA = dbustest-$(API_VERSION).pc

=== modified file 'libdbustest/bustle.c'
--- libdbustest/bustle.c	2012-10-02 03:11:56 +0000
+++ libdbustest/bustle.c	2012-12-14 21:29:07 +0000
@@ -29,7 +29,6 @@
 	gchar * executable;
 
 	guint watch;
-	GIOChannel * stdout;
 	GIOChannel * stderr;
 	GIOChannel * file;
 	GPid pid;
@@ -47,9 +46,6 @@
 static void process_run                 (DbusTestTask * task);
 static DbusTestTaskState get_state      (DbusTestTask * task);
 static gboolean get_passed              (DbusTestTask * task);
-static gboolean bustle_writes           (GIOChannel *          channel,
-                                         GIOCondition          condition,
-                                         gpointer              data);
 static gboolean bustle_write_error      (GIOChannel *          channel,
                                          GIOCondition          condition,
                                          gpointer              data);
@@ -81,10 +77,9 @@
 	self->priv = DBUS_TEST_BUSTLE_GET_PRIVATE(self);
 
 	self->priv->filename = g_strconcat(g_get_current_dir(), G_DIR_SEPARATOR_S, "bustle.log", NULL);
-	self->priv->executable = g_strdup("bustle-dbus-monitor");
+	self->priv->executable = g_strdup(BUSTLE_DUAL_MONITOR);
 
 	self->priv->watch = 0;
-	self->priv->stdout = NULL;
 	self->priv->stderr = NULL;
 	self->priv->file = NULL;
 	self->priv->pid = 0;
@@ -106,19 +101,13 @@
 	}
 
 	if (bustler->priv->pid != 0) {
-		/* TODO: Send a single dbus message */
+		gchar * command = g_strdup_printf("kill -INT %d", bustler->priv->pid);
+		g_spawn_command_line_sync(command, NULL, NULL, NULL, NULL);
+		g_free(command);
 
 		g_spawn_close_pid(bustler->priv->pid);
 	}
 
-	if (bustler->priv->stdout != NULL) {
-		while (G_IO_IN & g_io_channel_get_buffer_condition(bustler->priv->stdout)) {
-			bustle_writes(bustler->priv->stdout, 0 /* unused */, bustler->priv->file);
-		}
-
-		g_clear_pointer(&bustler->priv->stdout, g_io_channel_unref);
-	}
-
 	if (bustler->priv->stderr != NULL) {
 		while (G_IO_IN & g_io_channel_get_buffer_condition(bustler->priv->stderr)) {
 			bustle_write_error(bustler->priv->stderr, 0 /* unused */, bustler);
@@ -220,34 +209,6 @@
 	return TRUE;
 }
 
-static gboolean
-bustle_writes (GIOChannel * channel, G_GNUC_UNUSED GIOCondition condition, gpointer data)
-{
-	gchar * line;
-	gsize termloc;
-
-	GIOStatus status = g_io_channel_read_line (channel, &line, NULL, &termloc, NULL);
-
-	if (status != G_IO_STATUS_NORMAL) {
-		return FALSE;
-	}
-
-	g_io_channel_write_chars((GIOChannel *)data,
-							 line,
-							 termloc,
-							 NULL,
-							 NULL);
-	g_io_channel_write_chars((GIOChannel *)data,
-							 "\n",
-							 1,
-							 NULL,
-							 NULL);
-
-	g_free(line);
-
-	return TRUE;
-}
-
 static void
 process_run (DbusTestTask * task)
 {
@@ -271,12 +232,11 @@
 		return;
 	}
 
-	gint bustle_stdout_num;
 	gint bustle_stderr_num;
 	
 	gchar ** bustle_monitor = g_new0(gchar *, 3);
 	bustle_monitor[0] = (gchar *)bustler->priv->executable;
-	bustle_monitor[1] = "--session";
+	bustle_monitor[1] = (gchar *)bustler->priv->filename;
 
 	g_spawn_async_with_pipes(g_get_current_dir(),
 	                         bustle_monitor, /* argv */
@@ -287,7 +247,7 @@
 	                         NULL, /* child setup data */
 	                         &bustler->priv->pid, /* PID */
 	                         NULL, /* stdin */
-	                         &bustle_stdout_num, /* stdout */
+	                         NULL, /* stdout */
 	                         &bustle_stderr_num, /* stderr */
 	                         &error); /* error */
 
@@ -308,12 +268,6 @@
 	}
 	bustler->priv->watch = g_child_watch_add(bustler->priv->pid, bustle_watcher, bustler);
 
-	bustler->priv->stdout = g_io_channel_unix_new(bustle_stdout_num);
-	g_io_add_watch(bustler->priv->stdout,
-	               G_IO_IN | G_IO_PRI, /* conditions */
-	               bustle_writes, /* func */
-	               bustler->priv->file); /* func data */
-
 	bustler->priv->stderr = g_io_channel_unix_new(bustle_stderr_num);
 	g_io_add_watch(bustler->priv->stderr,
 	               G_IO_IN, /* conditions */

=== added file 'libdbustest/dbus-test-bustle-handler'
--- libdbustest/dbus-test-bustle-handler	1970-01-01 00:00:00 +0000
+++ libdbustest/dbus-test-bustle-handler	2012-12-14 20:22:42 +0000
@@ -0,0 +1,13 @@
+#!/bin/bash -e
+
+if [ -x /usr/bin/bustle-pcap ]; then
+	exec /usr/bin/bustle-pcap $1
+fi
+
+if [ -x /usr/bin/bustle-dbus-monitor ]; then
+	exec /usr/bin/bustle-dbus-monitor --session > $1
+fi
+
+echo "Unable to find a suitable Bustle capture tool"
+exit 1
+

=== modified file 'libdbustest/service.c'
--- libdbustest/service.c	2012-10-03 03:15:59 +0000
+++ libdbustest/service.c	2012-12-14 21:30:05 +0000
@@ -125,6 +125,21 @@
 	g_return_if_fail(DBUS_TEST_IS_SERVICE(object));
 	DbusTestService * self = DBUS_TEST_SERVICE(object);
 
+	if (!g_queue_is_empty(&self->priv->tasks_last)) {
+		g_queue_foreach(&self->priv->tasks_last, task_unref, NULL);
+		g_queue_clear(&self->priv->tasks_last);
+	}
+
+	if (!g_queue_is_empty(&self->priv->tasks_normal)) {
+		g_queue_foreach(&self->priv->tasks_normal, task_unref, NULL);
+		g_queue_clear(&self->priv->tasks_normal);
+	}
+
+	if (!g_queue_is_empty(&self->priv->tasks_first)) {
+		g_queue_foreach(&self->priv->tasks_first, task_unref, NULL);
+		g_queue_clear(&self->priv->tasks_first);
+	}
+
 	if (self->priv->dbus_watch != 0) {
 		g_source_remove(self->priv->dbus_watch);
 		self->priv->dbus_watch = 0;
@@ -141,6 +156,7 @@
 		self->priv->dbus_io = NULL;
 	}
 
+	g_print("DBus daemon: Shutdown\n");
 	if (self->priv->dbus != 0) {
 		gchar * cmd = g_strdup_printf("kill -9 %d", self->priv->dbus);
 		g_spawn_command_line_async(cmd, NULL);
@@ -150,21 +166,6 @@
 		self->priv->dbus = 0;
 	}
 
-	if (!g_queue_is_empty(&self->priv->tasks_first)) {
-		g_queue_foreach(&self->priv->tasks_first, task_unref, NULL);
-		g_queue_clear(&self->priv->tasks_first);
-	}
-
-	if (!g_queue_is_empty(&self->priv->tasks_normal)) {
-		g_queue_foreach(&self->priv->tasks_normal, task_unref, NULL);
-		g_queue_clear(&self->priv->tasks_normal);
-	}
-
-	if (!g_queue_is_empty(&self->priv->tasks_last)) {
-		g_queue_foreach(&self->priv->tasks_last, task_unref, NULL);
-		g_queue_clear(&self->priv->tasks_last);
-	}
-
 	if (self->priv->mainloop != NULL) {
 		g_main_loop_unref(self->priv->mainloop);
 		self->priv->mainloop = NULL;

=== modified file 'libdbustest/task.c'
--- libdbustest/task.c	2012-10-03 03:15:59 +0000
+++ libdbustest/task.c	2012-12-14 21:28:07 +0000
@@ -124,6 +124,8 @@
 	g_return_if_fail(DBUS_TEST_IS_TASK(object));
 	DbusTestTask * self = DBUS_TEST_TASK(object);
 
+	g_print("%s: Shutting down\n", self->priv->name);
+
 	g_free(self->priv->name);
 	g_free(self->priv->name_padded);
 	g_free(self->priv->wait_for);

=== modified file 'tests/Makefile.am'
--- tests/Makefile.am	2012-10-03 17:53:05 +0000
+++ tests/Makefile.am	2012-12-14 21:47:47 +0000
@@ -152,9 +152,9 @@
 	@echo "#!/bin/sh -e" > $@
 	@echo "$(DBUS_RUNNER) --task cat --parameter \"$(top_srcdir)/src/dbus-test-runner.c\" --task-name \"cat1\" --task cat --parameter \"$(top_srcdir)/src/dbus-test-runner.c\" --task-name \"cat2\" > testcat.output.txt" >> $@
 	@echo "echo Finding cat1 data" >> $@
-	@echo "grep ^cat1: testcat.output.txt | tail -n +2 > testcat.output.cat1.txt" >> $@
+	@echo "grep ^cat1: testcat.output.txt | tail -n +2 | head -n -1 > testcat.output.cat1.txt" >> $@
 	@echo "echo Finding cat2 data" >> $@
-	@echo "grep ^cat2: testcat.output.txt | tail -n +2 > testcat.output.cat2.txt" >> $@
+	@echo "grep ^cat2: testcat.output.txt | tail -n +2 | head -n -1 > testcat.output.cat2.txt" >> $@
 	@echo "echo Filtering cat1 data" >> $@
 	@echo "sed -e s/cat1:\\ //g testcat.output.cat1.txt > testcat.output.cat1.filtered.txt" >> $@
 	@echo "echo Filtering cat2 data" >> $@
@@ -167,18 +167,23 @@
 DISTCLEANFILES += testcat.output.txt testcat.output.cat1.txt testcat.output.cat2.txt testcat.output.cat1.filtered.txt testcat.output.cat2.filtered.txt
 
 TESTS += test-bustle
-test-bustle: Makefile.am test-bustle.reference
+test-bustle: Makefile.am test-bustle.reference test-bustle.0.4.reference
 	@echo "#!/bin/sh -e" > $@
-	@echo $(DBUS_RUNNER) --bustle-data \"$(builddir)/test-bustle.bustle\" --task $(srcdir)/test-bustle-list.sh >> $@
+	@echo $(DBUS_RUNNER) --bustle-monitor $(top_srcdir)/libdbustest/dbus-test-bustle-handler --bustle-data \"$(builddir)/test-bustle.bustle\" --task $(srcdir)/test-bustle-list.sh >> $@
+	@echo "if [ -x /usr/bin/bustle-count ] ; then" >> $@
 	@echo "grep ^sig \"$(builddir)/test-bustle.bustle\" | grep ":1.1" | grep "com.launchpad.dbustestrunner" | cut -f 5-9 > test-bustle.filtered" >> $@
 	@echo "diff \"$(srcdir)/test-bustle.reference\" \"$(builddir)/test-bustle.filtered\"" >> $@
+	@echo "else" >> $@
+	@echo "bustle --count \"$(builddir)/test-bustle.bustle\" > \"$(builddir)/test-bustle.filtered\"" >> $@
+	@echo "diff \"$(srcdir)/test-bustle.0.4.reference\" \"$(builddir)/test-bustle.filtered\"" >> $@
+	@echo "fi" >> $@
 	@chmod +x $@
 DISTCLEANFILES += test-bustle.bustle test-bustle.filtered
 
 TESTS += test-bustle-bad-file
 test-bustle-bad-file: Makefile.am
 	@echo "#!/bin/sh -e" > $@
-	@echo $(DBUS_RUNNER) --bustle-data \"$(builddir)\" --task true >> $@
+	@echo $(DBUS_RUNNER) --bustle-monitor $(top_srcdir)/libdbustest/dbus-test-bustle-handler --bustle-data \"$(builddir)\" --task true >> $@
 	@chmod +x $@
 XFAIL_TESTS += test-bustle-bad-file
 
@@ -201,7 +206,7 @@
 TESTS += test-bustle-data
 test-bustle-data: Makefile.am
 	@echo "#!/bin/sh -e" > $@
-	@echo "$(DBUS_RUNNER) --bustle-data \"$(builddir)/test-bustle-data.bustle\" \\" >> $@
+	@echo "$(DBUS_RUNNER) --bustle-monitor $(top_srcdir)/libdbustest/dbus-test-bustle-handler --bustle-data \"$(builddir)/test-bustle-data.bustle\" \\" >> $@
 	@echo "--task $(srcdir)/test-bustle-list.sh \\" >> $@
 	@echo "--task $(srcdir)/test-bustle-list.sh \\" >> $@
 	@echo "--task $(srcdir)/test-bustle-list.sh \\" >> $@
@@ -214,7 +219,11 @@
 	@echo "--task $(srcdir)/test-bustle-list.sh \\" >> $@
 	@echo "--task $(srcdir)/test-bustle-list.sh \\" >> $@
 	@echo "--task $(srcdir)/test-bustle-list.sh" >> $@
+	@echo "if [ -x /usr/bin/bustle-count ] ; then" >> $@
 	@echo "$(srcdir)/test-bustle-data-check.sh \"$(builddir)/test-bustle-data.bustle\" 12" >> $@
+	@echo "else" >> $@
+	@echo "$(srcdir)/test-bustle-data-check.0.4.sh \"$(builddir)/test-bustle-data.bustle\" 12" >> $@
+	@echo "fi" >> $@
 	@chmod +x $@
 DISTCLEANFILES += test-bustle-data.bustle
 
@@ -323,5 +332,7 @@
 	test-wait-output.reference \
 	delayrm.sh \
 	test-bustle.reference \
+	test-bustle.0.4.reference \
 	test-bustle-data-check.sh \
+	test-bustle-data-check.0.4.sh \
 	test-bustle-list.sh

=== added file 'tests/test-bustle-data-check.0.4.sh'
--- tests/test-bustle-data-check.0.4.sh	1970-01-01 00:00:00 +0000
+++ tests/test-bustle-data-check.0.4.sh	2012-12-14 21:47:47 +0000
@@ -0,0 +1,3 @@
+#!/bin/bash -e
+
+[ `bustle --count $1 | cut -d " " -f 4` -eq $2 ]

=== added file 'tests/test-bustle.0.4.reference'
--- tests/test-bustle.0.4.reference	1970-01-01 00:00:00 +0000
+++ tests/test-bustle.0.4.reference	2012-12-14 21:37:29 +0000
@@ -0,0 +1,1 @@
+    1 signal com.launchpad.dbustestrunner.signal

=== modified file 'tests/test-wait-output.reference'
--- tests/test-wait-output.reference	2012-10-02 21:45:47 +0000
+++ tests/test-wait-output.reference	2012-12-14 21:42:44 +0000
@@ -3,3 +3,4 @@
 Three
 Four
 Five
+Shutting down

