--- get-flash-videos-1.24.orig/lib/FlashVideo/Site/Googlevideosearch.pm                                         
+++ get-flash-videos-1.24/lib/FlashVideo/Site/Googlevideosearch.pm                                              
@@ -10,6 +10,8 @@ sub search {                                                                                  

   my $browser = FlashVideo::Mechanize->new;
   
+  $browser->allow_redirects;                                                                                   
+                                                                                                               
   $browser->get('http://video.google.com/');
 
   $browser->submit_form(
