diff -up /usr/share/screenruler/ruler_window.rb.orig /usr/share/screenruler/ruler_window.rb
--- /usr/share/screenruler/ruler_window.rb.orig	2011-03-30 00:26:46.000000000 -0400
+++ /usr/share/screenruler/ruler_window.rb	2013-03-20 14:26:17.878840196 -0400
@@ -16,7 +16,8 @@
  #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  ###############################################################################
 
-require 'glade_window', 'ruler_popup_menu', 'canvas', 'unique_timeout'
+require 'glade_window', 'canvas', 'unique_timeout'
+require_relative 'ruler_popup_menu'
 
 Unit = Struct.new('Unit', :name, :tick_pattern, :units_per_pattern_repetition, :per_inch)
 
diff -up /usr/share/screenruler/screenruler.rb.orig /usr/share/screenruler/screenruler.rb
--- /usr/share/screenruler/screenruler.rb.orig	2011-03-30 00:26:46.000000000 -0400
+++ /usr/share/screenruler/screenruler.rb	2013-03-20 14:29:12.584115347 -0400
@@ -1,5 +1,5 @@
 #!/usr/bin/env ruby
-#coding: utf-8
+# -*- encoding: utf-8; -*-
 
  ###############################################################################
  #  Copyright 2008 Ian McIntosh <ian@openanswers.org>
@@ -45,7 +45,10 @@ GCONF_ROOT		= '/apps/screenruler'
 puts _('Loading libraries...')
 
 require 'addons_ruby'									# for multi-file 'require'
-require 'gtk2', 'settings', 'addons_gtk', 'ruler_window', 'preferences_window', 'help_window'
+require 'gtk2', 'settings', 'libglade2', 'gconf2', 'addons_gtk'
+require_relative 'ruler_window'
+require_relative 'preferences_window'
+require_relative 'help_window'
 
 ###################################################################
 # Main
