diff -ru src.old/nb_global.c src/nb_global.c
--- src.old/nb_global.c	2010-12-03 16:25:27.000000000 -0500
+++ src/nb_global.c	2010-12-03 16:26:33.000000000 -0500
@@ -30,6 +30,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <libintl.h>
 
 #include "ui/nb_ui_interface.h"
 #include "ui/nb_ui_note_view.h"
diff -ru src.old/nb_util.c src/nb_util.c
--- src.old/nb_util.c	2010-12-03 16:25:27.000000000 -0500
+++ src/nb_util.c	2010-12-03 16:26:26.000000000 -0500
@@ -19,6 +19,7 @@
 
 #include <glib.h>
 #include <time.h>
+#include <libintl.h>
 
 #include "nb_util.h"
 
diff -ru src.old/ui/nb_ui_attachment_view.c src/ui/nb_ui_attachment_view.c
--- src.old/ui/nb_ui_attachment_view.c	2010-12-03 16:25:27.000000000 -0500
+++ src/ui/nb_ui_attachment_view.c	2010-12-03 16:26:41.000000000 -0500
@@ -21,6 +21,7 @@
 #include <gtk/gtk.h>
 #include <glade/glade.h>
 #include <libgnomeui/libgnomeui.h>
+#include <libintl.h>
 
 #include "../nb_global.h"
 #include "../nb_attachment.h"
diff -ru src.old/ui/nb_ui_interface.c src/ui/nb_ui_interface.c
--- src.old/ui/nb_ui_interface.c	2010-12-03 16:25:27.000000000 -0500
+++ src/ui/nb_ui_interface.c	2010-12-03 16:26:37.000000000 -0500
@@ -26,6 +26,7 @@
 #include <glade/glade.h>
 #include <gconf/gconf-client.h>
 #include <libgnomeui/gnome-about.h>
+#include <libintl.h>
 
 #include "../nb_global.h"
 #include "nb_ui_interface_callbacks.h"
