diff -ru src.old/menus.c src/menus.c
--- src.old/menus.c	2015-12-22 23:23:44.631524429 -0500
+++ src/menus.c	2015-12-22 23:24:40.652526222 -0500
@@ -79,11 +79,6 @@
 		_("Import data for this list"), G_CALLBACK (file_import)},
 	{"ExportFileAction", GTK_STOCK_CONVERT, _("_Export ..."), "",
 		_("Export this list"), G_CALLBACK (file_export)},
-	{"PrintFileAction", GTK_STOCK_PRINT, _("_Print"), "<Control>P",
-		_("Print this report"), G_CALLBACK (print_cb)},
-	{"PreviewFileAction", GTK_STOCK_PRINT_PREVIEW, _("Print Previe_w"),
-		"<Control><Shift>P", _("Preview the printed report"),
-		G_CALLBACK (print_preview_cb)},
 	{"QuitAction", GTK_STOCK_QUIT, _("Quit QuickList"),
 			"<control>Q", _("Quit the QuickList application"),
 		G_CALLBACK (exit_action)},
diff -ru src.old/report.c src/report.c
--- src.old/report.c	2015-12-22 23:23:44.631524429 -0500
+++ src/report.c	2015-12-22 23:24:22.865525653 -0500
@@ -336,57 +336,6 @@
 	page_num++;
 }
 
-void
-print_cb (GtkWidget G_GNUC_UNUSED * widget, gpointer G_GNUC_UNUSED data)
-{
-	GnomePrintJob *print_master;
-	GnomePrintContext *print_context;
-
-	print_master = gnome_print_job_new (NULL);
-	gtk_html_print_set_master (html, print_master);
-	
-	print_context = gnome_print_job_get_context (print_master);
-	font = gnome_font_find_closest ((const guchar *) "Helvetica", 12);
-	page_num = 1;
-	pages = gtk_html_print_get_pages_num (html, print_context,
-		.0, gnome_font_get_ascender (font) - gnome_font_get_descender (font));
-	gtk_html_print_with_header_footer (html, print_context,
-		.0, gnome_font_get_ascender (font) - gnome_font_get_descender (font),
-		NULL, print_footer, NULL);
-	if (font)
-		g_object_unref (font);
-	gnome_print_job_render (print_master, print_context);
-}
-
-void
-print_preview_cb (GtkWidget G_GNUC_UNUSED * widget, gpointer G_GNUC_UNUSED data)
-{
-	GnomePrintJob *print_master;
-	GnomePrintContext *print_context;
-	GtkWidget *preview;
-
-	print_master = gnome_print_job_new (NULL);
-	gtk_html_print_set_master (html, print_master);
-
-	print_context = gnome_print_job_get_context (print_master);
-	font = gnome_font_find_closest ((const guchar *) "Helvetica", 12);
-
-	page_num = 1;
-	pages = gtk_html_print_get_pages_num (html, print_context,
-		.0, gnome_font_get_ascender (font) - gnome_font_get_descender (font));
-	gtk_html_print_with_header_footer (html, print_context,
-		.0, gnome_font_get_ascender (font) - gnome_font_get_descender (font),
-		NULL, print_footer, NULL);
-	if (font)
-		g_object_unref (font);
-	
-	gnome_print_job_close (print_master);
-	preview = gnome_print_job_preview_new (print_master, (const guchar *) "Print Preview");
-	gtk_widget_show (preview);
-
-	g_object_unref (print_master);
-}
-
 /** \todo remove old report_set_col_field code */
 
 /* Sets current column and current field. */
diff -ru src.old/report.h src/report.h
--- src.old/report.h	2015-12-22 23:23:44.632524429 -0500
+++ src/report.h	2015-12-22 23:24:27.065525787 -0500
@@ -43,7 +43,5 @@
 void report_col_right (GtkAction * w, gpointer data);
 void report_totals_group (GtkAction * w, gpointer data);
 void report_totals_total (GtkAction * w, gpointer data);
-void print_preview_cb (GtkWidget *widget, gpointer data);
-void print_cb (GtkWidget * widget, gpointer data);
 
 #endif /* _REPORT_H */
