diff -ru dact-0.8.42.old/Makefile.in dact-0.8.42/Makefile.in
--- dact-0.8.42.old/Makefile.in	2018-02-07 05:22:15.387000000 -0500
+++ dact-0.8.42/Makefile.in	2018-02-07 05:26:06.165000000 -0500
@@ -93,21 +93,21 @@
 	rm -rf Makefile.dep configure autom4te.cache/ config.h.in
 
 install-bin: dact$(EXEEXT)
-	-$(INSTALL) -d $(bindir)
-	$(INSTALL) -m 755 dact$(EXEEXT) $(bindir)/dact$(EXEEXT)
+	-$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 dact$(EXEEXT) $(DESTDIR)$(bindir)/dact$(EXEEXT)
 
 install: all install-bin
-	-$(INSTALL) -d $(mandir)/man1
-	-$(INSTALL) -d $(libdir)
-	-$(INSTALL) -d $(datadir)/dact
-	-$(INSTALL) -d $(sysconfdir)
-	$(INSTALL) -m 644 dact.conf $(CONF)
-	$(INSTALL) -m 644 Docs/dact.1 $(mandir)/man1/dact.1
-	-$(INSTALL) -m 755 libdact.$(SHOBJEXT) $(libdir)/libdact.$(SHOBJEXT)
-	-$(INSTALL) -m 644 libdact.$(SHOBJEXT).def $(libdir)/libdact.$(SHOBJEXT).def || :
-	-$(INSTALL) -m 644 libdact.$(SHOBJEXT).a $(libdir)/libdact.$(SHOBJEXT).a || :
-	-$(INSTALL) -m 644 libdact.a $(libdir)/libdact.a
-@MODS@	for so in $(MODS); do $(INSTALL) -m 755 $$so $(datadir)/dact/$$so; done
+	-$(INSTALL) -d $(DESTDIR)$(mandir)/man1
+	-$(INSTALL) -d $(DESTDIR)$(libdir)
+	-$(INSTALL) -d $(DESTDIR)$(datadir)/dact
+	-$(INSTALL) -d $(DESTDIR)$(sysconfdir)
+	$(INSTALL) -m 644 dact.conf $(DESTDIR)$(CONF)
+	$(INSTALL) -m 644 Docs/dact.1 $(DESTDIR)$(mandir)/man1/dact.1
+	-$(INSTALL) -m 755 libdact.$(SHOBJEXT) $(DESTDIR)$(libdir)/libdact.$(SHOBJEXT)
+	-$(INSTALL) -m 644 libdact.$(SHOBJEXT).def $(DESTDIR)$(libdir)/libdact.$(SHOBJEXT).def || :
+	-$(INSTALL) -m 644 libdact.$(SHOBJEXT).a $(DESTDIR)$(libdir)/libdact.$(SHOBJEXT).a || :
+	-$(INSTALL) -m 644 libdact.a $(DESTDIR)$(libdir)/libdact.a
+@MODS@	for so in $(MODS); do $(INSTALL) -m 755 $$so $(DESTDIR)$(datadir)/dact/$$so; done
 #	./patch-magic-file $(datadir)/dact/magic
 
 install-bin-local: dact$(EXEEXT)
