diff -ru bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/compilerversions/compile.sh bnd-2.1.0.REL/biz.aQute.bndlib.tests/compilerversions/compile.sh
--- bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/compilerversions/compile.sh	2018-02-05 17:33:29.173000000 -0500
+++ bnd-2.1.0.REL/biz.aQute.bndlib.tests/compilerversions/compile.sh	2018-02-05 17:36:09.091000000 -0500
@@ -4,11 +4,9 @@
 javac -target 1.4 -source 1.4 -cp src src/sun_1_4/*.java
 javac -target 1.5 -source 1.5 -cp src src/sun_1_5/*.java
 #javac -target 1.6 -source 1.6 -cp src src/sun_1_6/*.java
-javac -target jsr14 -source 1.5 -cp src src/sun_jsr14/*.java
 java -jar ../jar/ecj_3.2.2.jar -target 1.1 -source 1.3 -cp src src/eclipse_1_1/*.java
 java -jar ../jar/ecj_3.2.2.jar -target 1.2 -source 1.3 -cp src src/eclipse_1_2/*.java
 java -jar ../jar/ecj_3.2.2.jar -target 1.3 -source 1.3 -cp src src/eclipse_1_3/*.java
 java -jar ../jar/ecj_3.2.2.jar -target 1.4 -source 1.4 -cp src src/eclipse_1_4/*.java
 java -jar ../jar/ecj_3.2.2.jar -target 1.5 -source 1.5 -cp src src/eclipse_1_5/*.java
 java -jar ../jar/ecj_3.2.2.jar -target 1.6 -source 1.6 -cp src src/eclipse_1_6/*.java
-java -jar ../jar/ecj_3.2.2.jar -target jsr14 -source 1.5 -cp src src/eclipse_jsr14/*.java
Only in bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/compilerversions/src: eclipse_jsr14
Only in bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/compilerversions/src: sun_jsr14
diff -ru bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/src/test/BuilderTest.java bnd-2.1.0.REL/biz.aQute.bndlib.tests/src/test/BuilderTest.java
--- bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/src/test/BuilderTest.java	2018-02-05 17:33:29.356000000 -0500
+++ bnd-2.1.0.REL/biz.aQute.bndlib.tests/src/test/BuilderTest.java	2018-02-05 17:35:14.447000000 -0500
@@ -208,8 +208,7 @@
 
 	public static void testEEMacro2() throws Exception {
 		String[] packages = {
-				"eclipse_1_1", "eclipse_1_2", "eclipse_1_3", "eclipse_1_4", "eclipse_1_5", "eclipse_1_6",
-				"eclipse_jsr14"
+				"eclipse_1_1", "eclipse_1_2", "eclipse_1_3", "eclipse_1_4", "eclipse_1_5", "eclipse_1_6"
 		};
 
 		String[] ees = {
diff -ru bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/src/test/ClassReferenceTest.java bnd-2.1.0.REL/biz.aQute.bndlib.tests/src/test/ClassReferenceTest.java
--- bnd-2.1.0.REL.old/biz.aQute.bndlib.tests/src/test/ClassReferenceTest.java	2018-02-05 17:33:29.356000000 -0500
+++ bnd-2.1.0.REL/biz.aQute.bndlib.tests/src/test/ClassReferenceTest.java	2018-02-05 17:35:43.844000000 -0500
@@ -27,8 +27,8 @@
 		Properties properties = new Properties();
 		properties.put("-classpath", "compilerversions/compilerversions.jar");
 		String[] packages = {
-				"sun_1_1", "sun_1_6", "eclipse_1_1", "sun_1_2", "sun_1_3", "sun_1_4", "sun_1_5", "sun_jsr14",
-				"eclipse_1_5", "eclipse_1_6", "eclipse_1_2", "eclipse_1_3", "eclipse_1_4", "eclipse_jsr14"
+				"sun_1_1", "sun_1_6", "eclipse_1_1", "sun_1_2", "sun_1_3", "sun_1_4", "sun_1_5",
+				"eclipse_1_5", "eclipse_1_6", "eclipse_1_2", "eclipse_1_3", "eclipse_1_4"
 		};
 		for (int i = 0; i < packages.length; i++) {
 			Builder builder = new Builder();
diff -ru bnd-2.1.0.REL.old/biz.aQute.junit/bnd.bnd bnd-2.1.0.REL/biz.aQute.junit/bnd.bnd
--- bnd-2.1.0.REL.old/biz.aQute.junit/bnd.bnd	2018-02-05 17:33:29.710000000 -0500
+++ bnd-2.1.0.REL/biz.aQute.junit/bnd.bnd	2018-02-05 17:35:02.735000000 -0500
@@ -8,7 +8,7 @@
 Tester-Plugin: aQute.junit.plugin.ProjectTesterImpl
 
 javac.source = 1.5
-javac.target = jsr14
+javac.target = 1.5
 Private-Package: aQute.junit.*,\
 	junit.*,\
 	org.junit.*
diff -ru bnd-2.1.0.REL.old/biz.aQute.junit/.settings/org.eclipse.jdt.core.prefs bnd-2.1.0.REL/biz.aQute.junit/.settings/org.eclipse.jdt.core.prefs
--- bnd-2.1.0.REL.old/biz.aQute.junit/.settings/org.eclipse.jdt.core.prefs	2018-02-05 17:33:29.709000000 -0500
+++ bnd-2.1.0.REL/biz.aQute.junit/.settings/org.eclipse.jdt.core.prefs	2018-02-05 17:34:58.619000000 -0500
@@ -1,7 +1,7 @@
 #Fri Aug 10 14:06:03 CEST 2012
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=jsr14
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.5
 org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
 org.eclipse.jdt.core.compiler.compliance=1.5
 org.eclipse.jdt.core.compiler.debug.lineNumber=generate
diff -ru bnd-2.1.0.REL.old/biz.aQute.launcher/bnd.bnd bnd-2.1.0.REL/biz.aQute.launcher/bnd.bnd
--- bnd-2.1.0.REL.old/biz.aQute.launcher/bnd.bnd	2018-02-05 17:33:29.713000000 -0500
+++ bnd-2.1.0.REL/biz.aQute.launcher/bnd.bnd	2018-02-05 17:36:28.737000000 -0500
@@ -12,4 +12,4 @@
 	
 Bundle-Version: 1.0.6
 javac.source = 1.5
-javac.target = jsr14
+javac.target = 1.5
diff -ru bnd-2.1.0.REL.old/biz.aQute.launcher/.settings/org.eclipse.jdt.core.prefs bnd-2.1.0.REL/biz.aQute.launcher/.settings/org.eclipse.jdt.core.prefs
--- bnd-2.1.0.REL.old/biz.aQute.launcher/.settings/org.eclipse.jdt.core.prefs	2018-02-05 17:33:29.713000000 -0500
+++ bnd-2.1.0.REL/biz.aQute.launcher/.settings/org.eclipse.jdt.core.prefs	2018-02-05 17:36:25.253000000 -0500
@@ -1,7 +1,7 @@
 #Fri Aug 10 13:46:22 CEST 2012
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=jsr14
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.5
 org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
 org.eclipse.jdt.core.compiler.compliance=1.5
 org.eclipse.jdt.core.compiler.debug.lineNumber=generate
diff -ru bnd-2.1.0.REL.old/cnf/findbugs.include.xml bnd-2.1.0.REL/cnf/findbugs.include.xml
--- bnd-2.1.0.REL.old/cnf/findbugs.include.xml	2018-02-05 17:33:28.802000000 -0500
+++ bnd-2.1.0.REL/cnf/findbugs.include.xml	2018-02-05 17:36:20.101000000 -0500
@@ -321,9 +321,6 @@
        <Package name="eclipse_1_6" />
      </Match>
      <Match>
-       <Package name="eclipse_jsr14" />
-     </Match>
-     <Match>
        <Package name="pkg2" />
      </Match>
      <Match>
@@ -360,9 +357,6 @@
        <Package name="sun_1_6" />
      </Match>
      <Match>
-       <Package name="sun_jsr14" />
-     </Match>
-     <Match>
        <Package name="test" />
      </Match>
      <Match>
